{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Homology Modelling Tutorial\n",
    "Max Epstein and Phil Biggin (based on previous work from Phill Stansfeld)\n",
    "\n",
    "\n",
    "## Objectives\n",
    "\n",
    "In this tutorial we will be building an homology model of the human adult muscle nAChR (nicotinic acetylcholine receptor). There are quite a few homology modelling tutorials available online - most of them guide you through a very simple case of a single sequence with a known good alignment to a target with no gaps.   Often though we are not so lucky - this tutorial is designed to take you through a more typical scenario, where you might want to first generate an initial multiple sequence alignment to improve the pair-wise alignment that you will use for the structure generation, what to do about missing bits of the template, how to generate multiple chain models, dealing with disufides, loop refinement and finally how to score and select a good model.\n",
    "\n",
    "In general though its also worth thinking about some of the following points before one starts making models:\n",
    "\n",
    "\n",
    "- In what state do we want to model our protein (i.e open, closed or desensitized)?\n",
    "\n",
    "- Does a structure of a phylogenetically related protein in that particular state exist?\n",
    "\n",
    "- If so, are there multiple structures of this state?\n",
    "\n",
    "- What structure gives the best sequence alignment?\n",
    "\n",
    "- What if you have high sequence alignment but low resolution? Perhaps it's worth considering a different template...\n",
    "\n",
    "- Are there any other reasons why you might not want to select a particular structure? I.e. inaccurate assignemnt of EM density - missing loops or regions. \n",
    "\n",
    "We will use this jupyter notebook to explore this.  The idea of this tutorial is that you should be able to follow it through at your own pace.  If you are of a timetabled course, then there will be some demonstrator help available, but this tutorial has been designed such that you should be able to complete it without help, even if you get stuck.\n",
    "\n",
    "**BEFORE YOU START - Please ensure you installed all the necessary software (modeller, jalview, pymol - via conda install)**\n",
    "\n",
    "- Most of software necessary can be very simply installed under the conda framework. Conda is available on windows, linux and mac.\n",
    "\n",
    "- This is the first step if you have not already installed conda (or miniconda) on your laptop - Rather than repeating the instructions here, simply head over to https://docs.conda.io/projects/conda/en/latest/user-guide/install/index.html and pick up the relevant instructions for your operating system.\n",
    "\n",
    "- Next, install the following packages with conda\n",
    "\n",
    "  `% conda install -c salilab modeller`    <-- Note - you will need to get a (free) license key - do this in advnance of commencing the tutorial!\n",
    "  \n",
    "  `% conda install -c bioconda jalview`   <-- if this gives problems - go to installers at http://www.jalview.org/builds/nextrel/Web_Installers/install.htm\n",
    " \n",
    " `% conda install -c schrodinger pymol`  <-- Note - this is the schrodinger version - a freeware version is available depending on your OS.\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### 1. Perform a mulitple sequence alignment\n",
    "\n",
    "- Make a directory entitled 'nAChR_model' where you can save outputs to.  This will be relative to the location of this notebook."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "! mkdir nAChR_model"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Get Sequences\n",
    "\n",
    "- Go to [uniprot.org](https://www.uniprot.org/) to search for the sequences of interest.\n",
    "\n",
    "- The gene encoding the muscle nicotinic acetylcholine receptor alpha1 subunit is 'CHRNA', so type that in to the search bar and look for ACHA_HUMAN.\n",
    "\n",
    "- Click on the entry number P02708 and locate the FASTA sequence.\n",
    "\n",
    "- Two isoforms of this subunit exist. Click on the 'Isoform 1' (Isoform 2 is listed first) FASTA file download. This will redirect you to a page with the amino acid sequence.  Keep the tab open or copy the tab because we are going to cut and paste from this.  Or you can save to a file in your nAChR_model directory and cut and paste from there.\n",
    "\n",
    "- Do this for CHRNB, CHRND and CHRNE sequences as well. These are the other subunits of the adult muscle nAChR.\n",
    "\n",
    "- We'll now do a blast search to look for a suitable template sequence for modelling the human alpha muscle subunit. This will be a sequence for which a 3D structure exists that we can use as a template to base our homology model on.\n",
    "\n",
    "- Go to the [Blast](https://blast.ncbi.nlm.nih.gov/Blast.cgi?PAGE=Proteins) wesbite\n",
    "\n",
    "- In the 'choose Search set', change the Database to PDB.\n",
    "\n",
    "- Copy and paste the CHRNA amino acid seqeunce into the FASTA box and click BLAST.\n",
    "\n",
    "- 6PV7 is a structure for a recently solved human nAChR with alpha3 and beta4 subunits.\n",
    "\n",
    "- Now locate the respective FASTA sequences for these subunits in uniprot and open each FASTA sequence in a new tab.\n",
    "\n",
    "- Finally, save all the sequences in your nAChR_model directory you created."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Generate the alignment\n",
    "\n",
    "- An accurate MSA will have a good balance of related sequences. As we are trying to model a cationic channel- it makes sense to include sequences of mostly cationic channels. \n",
    "\n",
    "- If we include too many anionic channels (lower sequence identity) then the accuracy of our alignment may decrease.\n",
    "\n",
    "- Once you have all your sequences in invividual tabs, go to the [MUSCLE](https://www.ebi.ac.uk/Tools/msa/muscle) multiple sequence alignment website:\n",
    "\n",
    "- Copy and paste in all your FASTA sequences. Remember to include the first line of each respective FASTA sequence that contains the metadata. If you want to include your email address to send the resulting file to you can, although this is not necessary.\n",
    "\n",
    "- When the alignment is ready, right click on the 'Download Alignment File' tab and click 'Save link as'.\n",
    "\n",
    "- Save the resulting .clw file as 'nachr_alignment.clw' to your 'nAChR_model' directory.\n",
    "\n",
    "Next we will visualize the alignment..."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Visualize the alignment\n",
    "\n",
    "- Open up Jalview"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "! jalview"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "- An example alignment should open automatically so just close them after the programme has loaded up.\n",
    "\n",
    "- Drag and drop your .clw file on to the JalView GUI.  (alternatively rename nachr_alignment.clw as nachr_alignment.aln and use the file tab and the open file dialogue).\n",
    "\n",
    "- Next, click on the 'Colour' tab and select 'Clustal'. This will visualise your alignment by amino acid properties. \n",
    "\n",
    "- Scroll through your alignment. If there are any insertions or deletions ('indels') denoted by dashes in regions of high sequence identity then you should scrutinize your alignemnt carefully and if needed, move them at the point of generating your final alignment. Indels commonly show up in regions of low sequence identity where flexible loops exist. These are usually less problematic but should be analysed before model generation. \n",
    "\n",
    "- Jalview can also interface to a secondary structure prediction.\n",
    "\n",
    "- Click on the 'Web Service' tab and then find 'JPred secondary structure prediction'.\n",
    "You should be able to see red lines where alpha-helices have been predicted and green lines for beta-sheets.\n",
    "\n",
    "- Congratulations- you've completed the MSA part of this tutorial!   Its worth at this stage just picking out the alpha subunit and looking at how similar they are.  \n",
    "\n",
    "- Before you quit jalview, use to save a new format that is easier to manipulate in Modeller.  Go to the tab and save as nachr_alignment.pir\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "\n",
    "### 2.  Build a model of a single subunit (the human muscle alpha subunit) first\n",
    "\n",
    "- For large heteromeric proteins like the nAChR, it may help to just try and build a single subunit in the first instance.  Or you may be wanting to model a protein that is comprised of a single chain only anyway.\n",
    "\n",
    "- Homology modelling at its simplest just needs a template and the sequence to be modelled.   We could have made an alignment of just the alpha4 template sequence with our muscle alpha sequence.  However,  if you have other sequences from related proteins, the quality of the alignment is usually much better, which is why did the above multiple sequence alignment. \n",
    "\n",
    "- We could use the whole alignment, but to make things as simple as possible lets just extract the two alpha sequences.\n",
    "\n",
    "- Make a copy of your .pir file and save it with .ali at the end of the file name. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "! cp nachr_alignment.pir alpha_pairwise_alignment.ali"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "- Although we copied the file and called it alpha_pairwise_alignment.ali it currently does not have the .ali format. First remove all subunits that you wont include in your homology model. \n",
    "    \n",
    "- A quick way of doing this is within VIM. Open your .ali file and hold 'shift :'. Then type\n",
    "\n",
    "    `g/Sequence_to_be_deleted/d`\n",
    "\n",
    "\n",
    "- and press enter. This will delete all lines containing a particular string.\n",
    "    \n",
    "- You will also need to replace the line starting with SP under the ACHA3_HUMAN entry with the following string :- \n",
    "\n",
    "    `structureX:alpha3.pdb:START:A:END:A::: 0.00: 0.00`\n",
    "  \n",
    "  \n",
    "- Similarly replace the line starting with SP under the ACHA_HUMAN with this:-\n",
    "\n",
    "    `sequence:alpha_seq::::::: 0.00: 0.00`\n",
    "  \n",
    "  \n",
    "- These lines contain information about whether the sequence is from the template (Structure) or from the sequence we want to model.  : characters separate bits of information (full details can be found [here](https://salilab.org/modeller/8v2/manual/node176.html)).  The key ones for us today to worry about are fields 3,4 5 and  6 for the template sequence - these fields are the starting residue,the chain letter (A in our case), the last residue and the chain letter (again just A for us).  Here, at the moment we have place holders (START, END) for the residue mumbers, which we have to replace later.  You might think we already know the start and last residue of the template from the uniprot sequence but as we will see, this will often not work directly because the crystal structure has some bits missing.\n",
    "\n",
    "    \n",
    "- Take time with this step as it's the most important. Mistakes made at this point carry through to the final model.\n",
    "    \n",
    "- If you are worried about this (or want to cheat!) you can look at the format of one we prepared earlier in the backup directory (note that this quite match - we will take you through how to work how long the alignment is in the section below - ie how to work out what to put in place of END).\n",
    "    \n",
    "- We are going to have edit this file further later."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "! vim alpha_pairwise_alignment.ali"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "! cat backup/single_subunit/alpha_pairwise_alignment.ali"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "- Once you are satisfied with your alignment you can focus on generating a 3D model.  Go to [rcsb](http://www.rcsb.org) and search for 6pv7.\n",
    "    \n",
    "- Download a .pdb file of 6pv7 to your current working directory, then open 6pv7.pdb in pymol.\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "! pymol 6pv7.pdb"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "- Within pymol, type:\n",
    "\n",
    "`select A, chain A and not HETATM`\n",
    "\n",
    "\n",
    "- Then on the right menu click 'A' and then 'copy to object'.\n",
    "\n",
    "- Save the resulting obj01 as alpha3.pdb (careful to select .pdb output)  \n",
    "\n",
    "- You also use the followin command to get the sequence to cut and paste\n",
    "\n",
    "`print(cmd.get_fastastr('obj01'))`  \n",
    "\n",
    "- You can cut and paste that (just the sequence -not the line begining with >) to a file; lets call that alpha3_structure_sequence.fasta.   Before you quit pymol take a look at just Obj01 (Chain A of the protein) - you might notice that in fact there is an apparent break in the chain (between the 3rd and 4th transmembrane helix) - this is a problem that we will need to do with and is quite a common problem in homology modelling.  \n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "\n",
    "- We now need to update the .ali file to contain the correct information for the template you are using.  \n",
    "    \n",
    "- When proteins are solved experimentally (e.g. using X-ray crytallography or Cryo-electron microscopy), often, certain flexible motifs or loops are removed as these may make structure determination difficult to impossible. As such, the uniprot sequence that you have for the alpha3 nAChR subunit will contain residues that have not been included in the experimentally solved template structure that we have. \n",
    "    \n",
    "- We will therefore need to correct this residue mismatch by removing those residues that are not in the alpha3.pdb stucture from the template structure section of our nachr_alignment.ali alignment file.\n",
    "       \n",
    "- If you compare nachr_alignment.ali with the alpha3_structure_sequence.fasta, you will see that the structure starts at Serine 32 (this is the start of the mature protein - the uniprot derived sequences contain the whole sequence).  The easiest thing at this stage is to delete residues and dashes from both sequences until that serine.  Then you can simply replace START in alpha_pairwise_alignment.ali with 1.\n",
    "       \n",
    "- Now we have to replace END with the last residue - this is simply the number of residues we have in the structure file.  You could of course simple count them, but below is a useful bit command line unix that can do this for you (assuming that alpha3_structure_sequence.fasta contains ONLY the sequence (the answer should be 387)."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    " tr -d '\\n\\r' < alpha3_structure_sequence.fasta |wc -m"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "- If you haven't do so, make those edits to nachr_alignment.ali now using your favourite editor or vim"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "! vim alpha_pairwise_alignment.ali"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "- Now there is one final edit to make before we can actually run modeller.  Remember this structure is not complete - there is a missing section between TM3 and 4.  If you look at the sequence you will see the break between residue 327 to 410.  We need to tell Modeller about that break - to do that we just insert a / character in both sequence after the R327 and before I410 in the sequence of the template and add the / at the exact equivalent position in the sequence to be modelled.  Use your editor to make these changes.  You can then check at this stage, by comparing to one I made earlier.\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "! vim alpha_pairwise_alignment.ali"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "! cat backup/single_subunit/alpha_pairwise_alignment.ali"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "- Now we should be ready to run modeller."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "from modeller import *\n",
    "from modeller.automodel import *    # Load the automodel class\n",
    "\n",
    "log.verbose()\n",
    "env = environ()\n",
    "\n",
    "# directories for input atom files\n",
    "env.io.atom_files_directory = ['.']\n",
    "\n",
    "a = automodel(env,\n",
    "              alnfile  = 'alpha_pairwise_alignment.ali',     # alignment filename\n",
    "              knowns   = 'alpha3',              # codes of the templates\n",
    "              sequence = 'alpha_seq')              # code of the target - we can call this what we like\n",
    "\n",
    "a.starting_model= 1                 # index of the first model\n",
    "a.ending_model  = 1                # index of the last model as this is 1 and the start is 1 - we will just generated 1 model\n",
    "                                    # (determines how many models to calculate)\n",
    "a.make()                            # do comparative modeling\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "- You can either view this model directly as below or fire up pymol to check the structure looks OK - visual inspection is super important!"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "import nglview\n",
    "view = nglview.show_structure_file('alpha_seq.B99990001.pdb')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "view"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "view.render_image()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "view._display_image()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "That concludes this part of the tutorial.  "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### 3.  Build a model of the full heteromeric receptor\n",
    "\n",
    "Next we will investigate how to make a model of the complete receptor.  The instructions will deliberately get less verbose, but you should be able to refer to the part 2 above if you need to.\n",
    "\n",
    "- First lets create a new directory to do this in - lets call that full-model"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "! mkdir full-model"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "- Copy your .clw, .ali and 6pv7.pdb files to this directory.\n",
    "\n",
    "- Make a copy of 6pv7.pdb and open this in pymol.  Have a look at the structure you will see that there are various other entities in this structure including antibodies that we don't really want in our template model.\n",
    "\n",
    "- In pymol type:\n",
    "\n",
    "     `remove not polymer or chain F or chain G or chain H or chain I` \n",
    "     \n",
    "     into the GUI to remove all non protein residues and 4 antibody chains.\n",
    "     \n",
    "-  You should also be able to clearly now see the pentameric nature of the a4b3 structure template comprised of five chains A-->E.  If you imagine looking down the central axis with the large beta sheet domain nearest you then we will have to model our human muscle nAChR such that the order of the subunits going anticlockwise is alpha-delta-beta-alpha-epsilon.      \n",
    "\n",
    "- In the header of 6pv7 you can see that it tells us that chains A and D are the alpha3 subunits, and B,C and E are the beta4 subunits.  This matches the positions of where we expect the alpha subunits of our human muscle to be (phew!).  The beta4 subunits will act as the template for all non-alpha subunits.   Thus the order of the sequences in the alignment file we need to make should indeed be alpha-delta-beta-alpha-epsilon.\n",
    "\n",
    "- Save the resulting structure of just the receptor protein as a new .pdb file  (simply file->export molecule then select all as we have removed all the atoms we don't want from this structure).  If you want to check this then there is one we made earlier in ./backup/full-model/6pv7_clean.pdb\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "- Modeller will expect one long protein essentially with '/' defining breaks in the chain between TM3 and 4 again but also between the ends of chains.  The end of the whole pentamer is marked with an *. Remember its essentially just a pairwise aligment - so just two long lines, but because that is hard to read we make it over several lines.\n",
    "\n",
    "- Make a copy of the original nachr_alignment.ali and call it something sensible.  You will then need to edit this into the same format you did for the single subunit. \n",
    " "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "- If you get completely stuck with editing the alignment, then there is one we made earlier in the backup/full-model directory called pentamer-alignment.ali, but try to make work this out on your own as first!."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "- Modify the Modeller script below to accomodate your naming"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "openf___224_> Open           $(LIB)/restyp.lib\n",
      "openf___224_> Open           ${MODINSTALL9v25}/modlib/resgrp.lib\n",
      "rdresgr_266_> Number of residue groups:        2\n",
      "openf___224_> Open           ${MODINSTALL9v25}/modlib/sstruc.lib\n",
      "\n",
      "Dynamically allocated memory at   amaxlibraries [B,KiB,MiB]:       455936     445.250     0.435\n",
      "\n",
      "Dynamically allocated memory at   amaxlibraries [B,KiB,MiB]:       456464     445.766     0.435\n",
      "openf___224_> Open           ${MODINSTALL9v25}/modlib/resdih.lib\n",
      "\n",
      "Dynamically allocated memory at   amaxlibraries [B,KiB,MiB]:       505064     493.227     0.482\n",
      "rdrdih__263_> Number of dihedral angle types         :        9\n",
      "              Maximal number of dihedral angle optima:        3\n",
      "              Dihedral angle names                   :  Alph Phi Psi Omeg chi1 chi2 chi3 chi4 chi5\n",
      "openf___224_> Open           ${MODINSTALL9v25}/modlib/radii.lib\n",
      "\n",
      "Dynamically allocated memory at   amaxlibraries [B,KiB,MiB]:       518364     506.215     0.494\n",
      "openf___224_> Open           ${MODINSTALL9v25}/modlib/radii14.lib\n",
      "openf___224_> Open           ${MODINSTALL9v25}/modlib/af_mnchdef.lib\n",
      "rdwilmo_274_> Mainchain residue conformation classes:  APBLE\n",
      "openf___224_> Open           ${MODINSTALL9v25}/modlib/mnch.lib\n",
      "rdclass_257_> Number of classes:        5\n",
      "openf___224_> Open           ${MODINSTALL9v25}/modlib/mnch1.lib\n",
      "openf___224_> Open           ${MODINSTALL9v25}/modlib/mnch2.lib\n",
      "openf___224_> Open           ${MODINSTALL9v25}/modlib/mnch3.lib\n",
      "openf___224_> Open           ${MODINSTALL9v25}/modlib/xs4.mat\n",
      "rdrrwgh_268_> Number of residue types:       21\n",
      "openf___224_> Open           pentamer_alignment.ali\n",
      "\n",
      "Dynamically allocated memory at   amaxalignment [B,KiB,MiB]:       529977     517.556     0.505\n",
      "\n",
      "Dynamically allocated memory at   amaxalignment [B,KiB,MiB]:       531427     518.972     0.507\n",
      "\n",
      "Dynamically allocated memory at   amaxalignment [B,KiB,MiB]:       534327     521.804     0.510\n",
      "\n",
      "Dynamically allocated memory at   amaxalignment [B,KiB,MiB]:       540127     527.468     0.515\n",
      "\n",
      "Dynamically allocated memory at   amaxalignment [B,KiB,MiB]:       551727     538.796     0.526\n",
      "\n",
      "Dynamically allocated memory at   amaxalignment [B,KiB,MiB]:       574927     561.452     0.548\n",
      "\n",
      "Dynamically allocated memory at   amaxalignment [B,KiB,MiB]:       621327     606.765     0.593\n",
      "\n",
      "Dynamically allocated memory at   amaxalignment [B,KiB,MiB]:      1527829    1492.021     1.457\n",
      "\n",
      "Dynamically allocated memory at    amaxsequence [B,KiB,MiB]:      1527857    1492.048     1.457\n",
      "\n",
      "Dynamically allocated memory at    amaxsequence [B,KiB,MiB]:      1527885    1492.075     1.457\n",
      "\n",
      "Dynamically allocated memory at    amaxsequence [B,KiB,MiB]:      1527913    1492.103     1.457\n",
      "\n",
      "Dynamically allocated memory at    amaxsequence [B,KiB,MiB]:      1527941    1492.130     1.457\n",
      "\n",
      "Dynamically allocated memory at    amaxsequence [B,KiB,MiB]:      1527969    1492.157     1.457\n",
      "\n",
      "Dynamically allocated memory at    amaxsequence [B,KiB,MiB]:      1527997    1492.185     1.457\n",
      "\n",
      "Dynamically allocated memory at    amaxsequence [B,KiB,MiB]:      1528025    1492.212     1.457\n",
      "\n",
      "Dynamically allocated memory at    amaxsequence [B,KiB,MiB]:      1528053    1492.239     1.457\n",
      "\n",
      "Dynamically allocated memory at    amaxsequence [B,KiB,MiB]:      1528081    1492.267     1.457\n",
      "\n",
      "Dynamically allocated memory at    amaxsequence [B,KiB,MiB]:      1528109    1492.294     1.457\n",
      "\n",
      "Dynamically allocated memory at    amaxsequence [B,KiB,MiB]:      1528137    1492.321     1.457\n",
      "\n",
      "Dynamically allocated memory at    amaxsequence [B,KiB,MiB]:      1528165    1492.349     1.457\n",
      "\n",
      "Dynamically allocated memory at    amaxsequence [B,KiB,MiB]:      1528193    1492.376     1.457\n",
      "\n",
      "Dynamically allocated memory at    amaxsequence [B,KiB,MiB]:      1528221    1492.403     1.457\n",
      "\n",
      "Dynamically allocated memory at    amaxsequence [B,KiB,MiB]:      1528249    1492.431     1.457\n",
      "\n",
      "Dynamically allocated memory at    amaxsequence [B,KiB,MiB]:      1528277    1492.458     1.457\n",
      "\n",
      "Dynamically allocated memory at    amaxsequence [B,KiB,MiB]:      1528305    1492.485     1.458\n",
      "\n",
      "Dynamically allocated memory at    amaxsequence [B,KiB,MiB]:      1528333    1492.513     1.458\n",
      "\n",
      "Dynamically allocated memory at    amaxsequence [B,KiB,MiB]:      1528361    1492.540     1.458\n",
      "\n",
      "Dynamically allocated memory at    amaxsequence [B,KiB,MiB]:      1528389    1492.567     1.458\n",
      "\n",
      "Dynamically allocated memory at    amaxsequence [B,KiB,MiB]:      1536125    1500.122     1.465\n",
      "\n",
      "Dynamically allocated memory at    amaxsequence [B,KiB,MiB]:      1544141    1507.950     1.473\n",
      "\n",
      "Read the alignment from file       : pentamer_alignment.ali\n",
      "\n",
      "Total number of alignment positions:  2014\n",
      "\n",
      "  #  Code        #_Res #_Segm PDB_code    Name\n",
      "-------------------------------------------------------------------------------\n",
      "  1  6pv7_clea    1935     10 6pv7_clean.\n",
      "  2  muscle_nA    2005     10 muscle_nACh\n",
      "check_a_343_> >> BEGINNING OF COMMAND\n",
      "openf___224_> Open           6pv7_clean.pdb\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:      2126366    2076.529     2.028\n",
      "\n",
      "Dynamically allocated memory at    amaxsequence [B,KiB,MiB]:      2126366    2076.529     2.028\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:      2126537    2076.696     2.028\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:      2127387    2077.526     2.029\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:      2128645    2078.755     2.030\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:      2130549    2080.614     2.032\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:      2133405    2083.403     2.035\n",
      "\n",
      "Dynamically allocated memory at    amaxsequence [B,KiB,MiB]:      2133405    2083.403     2.035\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:      2133495    2083.491     2.035\n",
      "\n",
      "Dynamically allocated memory at    amaxsequence [B,KiB,MiB]:      2133495    2083.491     2.035\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:      2133630    2083.623     2.035\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:      2137914    2087.807     2.039\n",
      "\n",
      "Dynamically allocated memory at    amaxsequence [B,KiB,MiB]:      2137914    2087.807     2.039\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:      2138112    2088.000     2.039\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:      2144538    2094.275     2.045\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:      2154160    2103.672     2.054\n",
      "\n",
      "Dynamically allocated memory at    amaxsequence [B,KiB,MiB]:      2154160    2103.672     2.054\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:      2154457    2103.962     2.055\n",
      "\n",
      "Dynamically allocated memory at    amaxsequence [B,KiB,MiB]:      2154457    2103.962     2.055\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:      2154907    2104.401     2.055\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:      2169357    2118.513     2.069\n",
      "\n",
      "Dynamically allocated memory at    amaxsequence [B,KiB,MiB]:      2169357    2118.513     2.069\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:      2170032    2119.172     2.070\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:      2191690    2140.322     2.090\n",
      "\n",
      "Dynamically allocated memory at    amaxsequence [B,KiB,MiB]:      2191690    2140.322     2.090\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:      2192698    2141.307     2.091\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:      2225202    2173.049     2.122\n",
      "\n",
      "Dynamically allocated memory at    amaxsequence [B,KiB,MiB]:      2225202    2173.049     2.122\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:      2226714    2174.525     2.124\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:      2275470    2222.139     2.170\n",
      "\n",
      "Dynamically allocated memory at    amaxsequence [B,KiB,MiB]:      2275470    2222.139     2.170\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:      2275470    2222.139     2.170\n",
      "\n",
      "Dynamically allocated memory at    amaxsequence [B,KiB,MiB]:      2275470    2222.139     2.170\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:      2277738    2224.354     2.172\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:      2350872    2295.773     2.242\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Dynamically allocated memory at    amaxsequence [B,KiB,MiB]:      2350872    2295.773     2.242\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:      2354274    2299.096     2.245\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:      2463958    2406.209     2.350\n",
      "\n",
      "Dynamically allocated memory at    amaxsequence [B,KiB,MiB]:      2463958    2406.209     2.350\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:      2469061    2411.192     2.355\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:      2633587    2571.862     2.512\n",
      "\n",
      "Dynamically allocated memory at    amaxsequence [B,KiB,MiB]:      2633587    2571.862     2.512\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:      2633587    2571.862     2.512\n",
      "\n",
      "Dynamically allocated memory at    amaxsequence [B,KiB,MiB]:      2633587    2571.862     2.512\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:      2641246    2579.342     2.519\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:      2888052    2820.363     2.754\n",
      "\n",
      "Dynamically allocated memory at   amaxstructure [B,KiB,MiB]:      2882490    2814.932     2.749\n",
      "\n",
      "check_ali___> Checking the sequence-structure alignment. \n",
      "\n",
      "Implied intrachain target CA(i)-CA(i+1) distances longer than  8.0 angstroms:\n",
      "\n",
      "ALN_POS  TMPL  RID1  RID2  NAM1  NAM2     DIST\n",
      "----------------------------------------------\n",
      "END OF TABLE\n",
      "check_a_344_> << END OF COMMAND\n",
      "openf___224_> Open           ${LIB}/top_heav.lib\n",
      "read_to_681_> topology.submodel read from topology file:        3\n",
      "openf___224_> Open           ${LIB}/par.lib\n",
      "\n",
      "Dynamically allocated memory at  amaxparameters [B,KiB,MiB]:      3021742    2950.920     2.882\n",
      "\n",
      "Dynamically allocated memory at  amaxparameters [B,KiB,MiB]:      3023342    2952.482     2.883\n",
      "\n",
      "Dynamically allocated memory at  amaxparameters [B,KiB,MiB]:      3027322    2956.369     2.887\n",
      "\n",
      "Dynamically allocated memory at  amaxparameters [B,KiB,MiB]:      3029322    2958.322     2.889\n",
      "\n",
      "Dynamically allocated memory at  amaxparameters [B,KiB,MiB]:      3032322    2961.252     2.892\n",
      "\n",
      "Dynamically allocated memory at  amaxparameters [B,KiB,MiB]:      3036822    2965.646     2.896\n",
      "\n",
      "Dynamically allocated memory at  amaxparameters [B,KiB,MiB]:      3042394    2971.088     2.901\n",
      "\n",
      "Dynamically allocated memory at  amaxparameters [B,KiB,MiB]:      3045194    2973.822     2.904\n",
      "\n",
      "Dynamically allocated memory at  amaxparameters [B,KiB,MiB]:      3049394    2977.924     2.908\n",
      "\n",
      "Dynamically allocated memory at  amaxparameters [B,KiB,MiB]:      3055694    2984.076     2.914\n",
      "\n",
      "Dynamically allocated memory at  amaxparameters [B,KiB,MiB]:      3058466    2986.783     2.917\n",
      "\n",
      "Dynamically allocated memory at  amaxparameters [B,KiB,MiB]:      3059866    2988.150     2.918\n",
      "read_pa_232_> parameters    BONDS   ANGLS  DIHEDS IMPROPS    MODE\n",
      "                              227     561     661     112       0\n",
      "\n",
      "Dynamically allocated memory at       amaxmodel [B,KiB,MiB]:      3112775    3039.819     2.969\n",
      "\n",
      "Dynamically allocated memory at       amaxmodel [B,KiB,MiB]:      3384529    3305.204     3.228\n",
      "\n",
      "Dynamically allocated memory at       amaxmodel [B,KiB,MiB]:      3426889    3346.571     3.268\n",
      "\n",
      "mkilst______> segment topology constructed from sequence and RTF:\n",
      "              segments residues  atoms  bonds angles dihedrals impropers:\n",
      "                     1      338   2754   2825      0         0      1235\n",
      "\n",
      "Dynamically allocated memory at       amaxmodel [B,KiB,MiB]:      3546937    3463.806     3.383\n",
      "\n",
      "patch_______> segment topology patched using RTF:     1  ;  SER  ;  NTER\n",
      "              segments residues  atoms  bonds angles dihedrals impropers:\n",
      "                     1      338   2754   2825   3861      4609      1235\n",
      "\n",
      "\n",
      "Dynamically allocated memory at       amaxmodel [B,KiB,MiB]:      3679513    3593.274     3.509\n",
      "patch_______> segment topology patched using RTF:   338  ;  ARG  ;  CTER\n",
      "              segments residues  atoms  bonds angles dihedrals impropers:\n",
      "                     1      338   2755   2826   3863      4611      1236\n",
      "\n",
      "genseg______> segment topology constructed from sequence and RTF:\n",
      "              segments residues  atoms  bonds angles dihedrals impropers:\n",
      "                     1      338   2755   2826   3863      4611      1236\n",
      "\n",
      "Dynamically allocated memory at    amaxsequence [B,KiB,MiB]:      3679541    3593.302     3.509\n",
      "\n",
      "Dynamically allocated memory at       amaxmodel [B,KiB,MiB]:      3689432    3602.961     3.519\n",
      "\n",
      "Dynamically allocated memory at       amaxmodel [B,KiB,MiB]:      3740098    3652.439     3.567\n",
      "\n",
      "Dynamically allocated memory at       amaxmodel [B,KiB,MiB]:      3748570    3660.713     3.575\n",
      "\n",
      "mkilst______> segment topology constructed from sequence and RTF:\n",
      "              segments residues  atoms  bonds angles dihedrals impropers:\n",
      "                     2      401   3271   3352   3863      4611      1454\n",
      "\n",
      "Dynamically allocated memory at       amaxmodel [B,KiB,MiB]:      3770602    3682.229     3.596\n",
      "\n",
      "patch_______> segment topology patched using RTF:   339  ;  VAL  ;  NTER\n",
      "              segments residues  atoms  bonds angles dihedrals impropers:\n",
      "                     2      401   3271   3352   4575      5454      1454\n",
      "\n",
      "\n",
      "Dynamically allocated memory at       amaxmodel [B,KiB,MiB]:      3770658    3682.283     3.596\n",
      "patch_______> segment topology patched using RTF:   401  ;  SER  ;  CTER\n",
      "              segments residues  atoms  bonds angles dihedrals impropers:\n",
      "                     2      401   3272   3353   4577      5456      1455\n",
      "\n",
      "genseg______> segment topology constructed from sequence and RTF:\n",
      "              segments residues  atoms  bonds angles dihedrals impropers:\n",
      "                     2      401   3272   3353   4577      5456      1455\n",
      "\n",
      "Dynamically allocated memory at    amaxsequence [B,KiB,MiB]:      3770686    3682.311     3.596\n",
      "\n",
      "Dynamically allocated memory at       amaxmodel [B,KiB,MiB]:      3823752    3734.133     3.647\n",
      "\n",
      "Dynamically allocated memory at       amaxmodel [B,KiB,MiB]:      4093742    3997.795     3.904\n",
      "\n",
      "Dynamically allocated memory at       amaxmodel [B,KiB,MiB]:      4135374    4038.451     3.944\n",
      "\n",
      "mkilst______> segment topology constructed from sequence and RTF:\n",
      "              segments residues  atoms  bonds angles dihedrals impropers:\n",
      "                     3      739   6026   6178   4577      5456      2690\n",
      "\n",
      "Dynamically allocated memory at       amaxmodel [B,KiB,MiB]:      4255450    4155.713     4.058\n",
      "\n",
      "patch_______> segment topology patched using RTF:   402  ;  SER  ;  NTER\n",
      "              segments residues  atoms  bonds angles dihedrals impropers:\n",
      "                     3      739   6026   6178   8438     10065      2690\n",
      "\n",
      "\n",
      "Dynamically allocated memory at       amaxmodel [B,KiB,MiB]:      4397706    4294.635     4.194\n",
      "patch_______> segment topology patched using RTF:   739  ;  ARG  ;  CTER\n",
      "              segments residues  atoms  bonds angles dihedrals impropers:\n",
      "                     3      739   6027   6179   8440     10067      2691\n",
      "\n",
      "genseg______> segment topology constructed from sequence and RTF:\n",
      "              segments residues  atoms  bonds angles dihedrals impropers:\n",
      "                     3      739   6027   6179   8440     10067      2691\n",
      "\n",
      "Dynamically allocated memory at    amaxsequence [B,KiB,MiB]:      4397734    4294.662     4.194\n",
      "\n",
      "Dynamically allocated memory at       amaxmodel [B,KiB,MiB]:      4407625    4304.321     4.203\n",
      "\n",
      "Dynamically allocated memory at       amaxmodel [B,KiB,MiB]:      4458291    4353.800     4.252\n",
      "\n",
      "Dynamically allocated memory at       amaxmodel [B,KiB,MiB]:      4476779    4371.854     4.269\n",
      "\n",
      "mkilst______> segment topology constructed from sequence and RTF:\n",
      "              segments residues  atoms  bonds angles dihedrals impropers:\n",
      "                     4      802   6543   6705   8440     10067      2909\n",
      "\n",
      "Dynamically allocated memory at       amaxmodel [B,KiB,MiB]:      4498811    4393.370     4.290\n",
      "\n",
      "patch_______> segment topology patched using RTF:   740  ;  VAL  ;  NTER\n",
      "              segments residues  atoms  bonds angles dihedrals impropers:\n",
      "                     4      802   6543   6705   9152     10910      2909\n",
      "\n",
      "\n",
      "Dynamically allocated memory at       amaxmodel [B,KiB,MiB]:      4498867    4393.425     4.290\n",
      "patch_______> segment topology patched using RTF:   802  ;  SER  ;  CTER\n",
      "              segments residues  atoms  bonds angles dihedrals impropers:\n",
      "                     4      802   6544   6706   9154     10912      2910\n",
      "\n",
      "genseg______> segment topology constructed from sequence and RTF:\n",
      "              segments residues  atoms  bonds angles dihedrals impropers:\n",
      "                     4      802   6544   6706   9154     10912      2910\n",
      "\n",
      "Dynamically allocated memory at    amaxsequence [B,KiB,MiB]:      4498895    4393.452     4.290\n",
      "\n",
      "Dynamically allocated memory at       amaxmodel [B,KiB,MiB]:      4551961    4445.274     4.341\n",
      "\n",
      "Dynamically allocated memory at       amaxmodel [B,KiB,MiB]:      4821951    4708.937     4.599\n",
      "\n",
      "Dynamically allocated memory at       amaxmodel [B,KiB,MiB]:      4853567    4739.812     4.629\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "mkilst______> segment topology constructed from sequence and RTF:\n",
      "              segments residues  atoms  bonds angles dihedrals impropers:\n",
      "                     5     1140   9298   9531   9154     10912      4145\n",
      "\n",
      "Dynamically allocated memory at       amaxmodel [B,KiB,MiB]:      4973643    4857.073     4.743\n",
      "\n",
      "patch_______> segment topology patched using RTF:   803  ;  SER  ;  NTER\n",
      "              segments residues  atoms  bonds angles dihedrals impropers:\n",
      "                     5     1140   9298   9531  13015     15521      4145\n",
      "\n",
      "\n",
      "Dynamically allocated memory at       amaxmodel [B,KiB,MiB]:      5094539    4975.136     4.859\n",
      "patch_______> segment topology patched using RTF:  1140  ;  ARG  ;  CTER\n",
      "              segments residues  atoms  bonds angles dihedrals impropers:\n",
      "                     5     1140   9299   9532  13017     15523      4146\n",
      "\n",
      "genseg______> segment topology constructed from sequence and RTF:\n",
      "              segments residues  atoms  bonds angles dihedrals impropers:\n",
      "                     5     1140   9299   9532  13017     15523      4146\n",
      "\n",
      "Dynamically allocated memory at    amaxsequence [B,KiB,MiB]:      5094567    4975.163     4.859\n",
      "\n",
      "Dynamically allocated memory at       amaxmodel [B,KiB,MiB]:      5104458    4984.822     4.868\n",
      "\n",
      "Dynamically allocated memory at       amaxmodel [B,KiB,MiB]:      5155124    5034.301     4.916\n",
      "\n",
      "Dynamically allocated memory at       amaxmodel [B,KiB,MiB]:      5183636    5062.145     4.944\n",
      "\n",
      "mkilst______> segment topology constructed from sequence and RTF:\n",
      "              segments residues  atoms  bonds angles dihedrals impropers:\n",
      "                     6     1203   9815  10058  13017     15523      4364\n",
      "\n",
      "Dynamically allocated memory at       amaxmodel [B,KiB,MiB]:      5205668    5083.660     4.965\n",
      "\n",
      "patch_______> segment topology patched using RTF:  1141  ;  VAL  ;  NTER\n",
      "              segments residues  atoms  bonds angles dihedrals impropers:\n",
      "                     6     1203   9815  10058  13729     16366      4364\n",
      "\n",
      "\n",
      "Dynamically allocated memory at       amaxmodel [B,KiB,MiB]:      5284804    5160.941     5.040\n",
      "patch_______> segment topology patched using RTF:  1203  ;  SER  ;  CTER\n",
      "              segments residues  atoms  bonds angles dihedrals impropers:\n",
      "                     6     1203   9816  10059  13731     16368      4365\n",
      "\n",
      "genseg______> segment topology constructed from sequence and RTF:\n",
      "              segments residues  atoms  bonds angles dihedrals impropers:\n",
      "                     6     1203   9816  10059  13731     16368      4365\n",
      "\n",
      "Dynamically allocated memory at    amaxsequence [B,KiB,MiB]:      5284832    5160.969     5.040\n",
      "\n",
      "Dynamically allocated memory at       amaxmodel [B,KiB,MiB]:      5337898    5212.791     5.091\n",
      "\n",
      "Dynamically allocated memory at       amaxmodel [B,KiB,MiB]:      5607888    5476.453     5.348\n",
      "\n",
      "Dynamically allocated memory at       amaxmodel [B,KiB,MiB]:      5642104    5509.867     5.381\n",
      "\n",
      "mkilst______> segment topology constructed from sequence and RTF:\n",
      "              segments residues  atoms  bonds angles dihedrals impropers:\n",
      "                     7     1541  12570  12884  13731     16368      5600\n",
      "\n",
      "Dynamically allocated memory at       amaxmodel [B,KiB,MiB]:      5762180    5627.129     5.495\n",
      "\n",
      "patch_______> segment topology patched using RTF:  1204  ;  SER  ;  NTER\n",
      "              segments residues  atoms  bonds angles dihedrals impropers:\n",
      "                     7     1541  12570  12884  17592     20977      5600\n",
      "\n",
      "\n",
      "Dynamically allocated memory at       amaxmodel [B,KiB,MiB]:      5857156    5719.879     5.586\n",
      "patch_______> segment topology patched using RTF:  1541  ;  ARG  ;  CTER\n",
      "              segments residues  atoms  bonds angles dihedrals impropers:\n",
      "                     7     1541  12571  12885  17594     20979      5601\n",
      "\n",
      "genseg______> segment topology constructed from sequence and RTF:\n",
      "              segments residues  atoms  bonds angles dihedrals impropers:\n",
      "                     7     1541  12571  12885  17594     20979      5601\n",
      "\n",
      "Dynamically allocated memory at    amaxsequence [B,KiB,MiB]:      5857184    5719.906     5.586\n",
      "\n",
      "Dynamically allocated memory at       amaxmodel [B,KiB,MiB]:      5867075    5729.565     5.595\n",
      "\n",
      "Dynamically allocated memory at       amaxmodel [B,KiB,MiB]:      5917741    5779.044     5.644\n",
      "\n",
      "mkilst______> segment topology constructed from sequence and RTF:\n",
      "              segments residues  atoms  bonds angles dihedrals impropers:\n",
      "                     8     1604  13087  13411  17594     20979      5819\n",
      "\n",
      "Dynamically allocated memory at       amaxmodel [B,KiB,MiB]:      5939773    5800.560     5.665\n",
      "\n",
      "patch_______> segment topology patched using RTF:  1542  ;  VAL  ;  NTER\n",
      "              segments residues  atoms  bonds angles dihedrals impropers:\n",
      "                     8     1604  13087  13411  18306     21822      5819\n",
      "\n",
      "\n",
      "Dynamically allocated memory at       amaxmodel [B,KiB,MiB]:      5939829    5800.614     5.665\n",
      "patch_______> segment topology patched using RTF:  1604  ;  SER  ;  CTER\n",
      "              segments residues  atoms  bonds angles dihedrals impropers:\n",
      "                     8     1604  13088  13412  18308     21824      5820\n",
      "\n",
      "genseg______> segment topology constructed from sequence and RTF:\n",
      "              segments residues  atoms  bonds angles dihedrals impropers:\n",
      "                     8     1604  13088  13412  18308     21824      5820\n",
      "\n",
      "Dynamically allocated memory at    amaxsequence [B,KiB,MiB]:      5939857    5800.642     5.665\n",
      "\n",
      "Dynamically allocated memory at       amaxmodel [B,KiB,MiB]:      5992923    5852.464     5.715\n",
      "\n",
      "Dynamically allocated memory at       amaxmodel [B,KiB,MiB]:      6262913    6116.126     5.973\n",
      "\n",
      "Dynamically allocated memory at       amaxmodel [B,KiB,MiB]:      6303977    6156.228     6.012\n",
      "\n",
      "mkilst______> segment topology constructed from sequence and RTF:\n",
      "              segments residues  atoms  bonds angles dihedrals impropers:\n",
      "                     9     1942  15842  16237  18308     21824      7055\n",
      "\n",
      "Dynamically allocated memory at       amaxmodel [B,KiB,MiB]:      6424053    6273.489     6.126\n",
      "\n",
      "patch_______> segment topology patched using RTF:  1605  ;  SER  ;  NTER\n",
      "              segments residues  atoms  bonds angles dihedrals impropers:\n",
      "                     9     1942  15842  16237  22169     26433      7055\n",
      "\n",
      "\n",
      "Dynamically allocated memory at       amaxmodel [B,KiB,MiB]:      6537989    6384.755     6.235\n",
      "patch_______> segment topology patched using RTF:  1942  ;  ARG  ;  CTER\n",
      "              segments residues  atoms  bonds angles dihedrals impropers:\n",
      "                     9     1942  15843  16238  22171     26435      7056\n",
      "\n",
      "genseg______> segment topology constructed from sequence and RTF:\n",
      "              segments residues  atoms  bonds angles dihedrals impropers:\n",
      "                     9     1942  15843  16238  22171     26435      7056\n",
      "\n",
      "Dynamically allocated memory at    amaxsequence [B,KiB,MiB]:      6538017    6384.782     6.235\n",
      "\n",
      "Dynamically allocated memory at       amaxmodel [B,KiB,MiB]:      6547908    6394.441     6.245\n",
      "\n",
      "Dynamically allocated memory at       amaxmodel [B,KiB,MiB]:      6598574    6443.920     6.293\n",
      "\n",
      "Dynamically allocated memory at       amaxmodel [B,KiB,MiB]:      6647838    6492.029     6.340\n",
      "\n",
      "mkilst______> segment topology constructed from sequence and RTF:\n",
      "              segments residues  atoms  bonds angles dihedrals impropers:\n",
      "                    10     2005  16359  16764  22171     26435      7274\n",
      "\n",
      "Dynamically allocated memory at       amaxmodel [B,KiB,MiB]:      6669870    6513.545     6.361\n",
      "\n",
      "patch_______> segment topology patched using RTF:  1943  ;  VAL  ;  NTER\n",
      "              segments residues  atoms  bonds angles dihedrals impropers:\n",
      "                    10     2005  16359  16764  22883     27278      7274\n",
      "\n",
      "\n",
      "Dynamically allocated memory at       amaxmodel [B,KiB,MiB]:      6669926    6513.600     6.361\n",
      "patch_______> segment topology patched using RTF:  2005  ;  SER  ;  CTER\n",
      "              segments residues  atoms  bonds angles dihedrals impropers:\n",
      "                    10     2005  16360  16765  22885     27280      7275\n",
      "\n",
      "genseg______> segment topology constructed from sequence and RTF:\n",
      "              segments residues  atoms  bonds angles dihedrals impropers:\n",
      "                    10     2005  16360  16765  22885     27280      7275\n",
      "\n",
      "Dynamically allocated memory at       amaxmodel [B,KiB,MiB]:      6812146    6652.486     6.497\n",
      "patch_______> segment topology patched using RTF:   130   144  ;  CYS  CYS  ;  DISU\n",
      "              segments residues  atoms  bonds angles dihedrals impropers:\n",
      "                    10     2005  16360  16766  22887     27283      7275\n",
      "patch_______> segment topology patched using RTF:   203   204  ;  CYS  CYS  ;  DISU\n",
      "              segments residues  atoms  bonds angles dihedrals impropers:\n",
      "                    10     2005  16360  16767  22889     27286      7275\n",
      "\n",
      "Dynamically allocated memory at       amaxmodel [B,KiB,MiB]:      6812394    6652.729     6.497\n",
      "patch_______> segment topology patched using RTF:   531   545  ;  CYS  CYS  ;  DISU\n",
      "              segments residues  atoms  bonds angles dihedrals impropers:\n",
      "                    10     2005  16360  16768  22891     27289      7275\n",
      "patch_______> segment topology patched using RTF:   932   946  ;  CYS  CYS  ;  DISU\n",
      "              segments residues  atoms  bonds angles dihedrals impropers:\n",
      "                    10     2005  16360  16769  22893     27292      7275\n",
      "patch_______> segment topology patched using RTF:  1333  1347  ;  CYS  CYS  ;  DISU\n",
      "              segments residues  atoms  bonds angles dihedrals impropers:\n",
      "                    10     2005  16360  16770  22895     27295      7275\n",
      "patch_______> segment topology patched using RTF:  1406  1407  ;  CYS  CYS  ;  DISU\n",
      "              segments residues  atoms  bonds angles dihedrals impropers:\n",
      "                    10     2005  16360  16771  22897     27298      7275\n",
      "patch_______> segment topology patched using RTF:  1734  1748  ;  CYS  CYS  ;  DISU\n",
      "              segments residues  atoms  bonds angles dihedrals impropers:\n",
      "                    10     2005  16360  16772  22899     27301      7275\n",
      "patch_s_522_> Number of disulfides patched in MODEL:        7\n",
      "transfe_506_> MODEL is an average of all templates.\n",
      "transfe_511_> Number of templates for coordinate transfer:        1\n",
      "              After transferring coordinates of the equivalent template atoms,\n",
      "              there are defined, undefined atoms in MODEL:    13893     2467\n",
      "The following 5 residues contain 6-membered rings with poor geometries\n",
      "after transfer from templates. Rebuilding rings from internal coordinates:\n",
      "   <Residue 182:A (type TYR)>\n",
      "   <Residue 711:C (type PHE)>\n",
      "   <Residue 1112:E (type PHE)>\n",
      "   <Residue 1385:G (type TYR)>\n",
      "   <Residue 1914:I (type PHE)>\n",
      "openf___224_> Open           muscle_nAChR.ini\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "wrpdb___568_> Residues, atoms, selected atoms:     2005    16360    16360\n",
      "make_re_417_> Restraint type to be calculated:  stereo\n",
      "\n",
      "Dynamically allocated memory at  amaxrestraints [B,KiB,MiB]:      6919709    6757.528     6.599\n",
      "\n",
      "Dynamically allocated memory at  amaxrestraints [B,KiB,MiB]:      6927901    6765.528     6.607\n",
      "\n",
      "Dynamically allocated memory at  amaxrestraints [B,KiB,MiB]:      6944285    6781.528     6.623\n",
      "\n",
      "Dynamically allocated memory at  amaxrestraints [B,KiB,MiB]:      6977053    6813.528     6.654\n",
      "\n",
      "Dynamically allocated memory at  amaxrestraints [B,KiB,MiB]:      7009821    6845.528     6.685\n",
      "\n",
      "Dynamically allocated memory at  amaxrestraints [B,KiB,MiB]:      7075357    6909.528     6.748\n",
      "\n",
      "Dynamically allocated memory at  amaxrestraints [B,KiB,MiB]:      7108125    6941.528     6.779\n",
      "\n",
      "Dynamically allocated memory at  amaxrestraints [B,KiB,MiB]:      7173661    7005.528     6.841\n",
      "\n",
      "Dynamically allocated memory at  amaxrestraints [B,KiB,MiB]:      7206429    7037.528     6.873\n",
      "\n",
      "Dynamically allocated memory at  amaxrestraints [B,KiB,MiB]:      7337501    7165.528     6.998\n",
      "\n",
      "Dynamically allocated memory at  amaxrestraints [B,KiB,MiB]:      7403037    7229.528     7.060\n",
      "\n",
      "Dynamically allocated memory at  amaxrestraints [B,KiB,MiB]:      7534109    7357.528     7.185\n",
      "\n",
      "Dynamically allocated memory at  amaxrestraints [B,KiB,MiB]:      7599645    7421.528     7.248\n",
      "\n",
      "Dynamically allocated memory at  amaxrestraints [B,KiB,MiB]:      7861789    7677.528     7.498\n",
      "\n",
      "Dynamically allocated memory at  amaxrestraints [B,KiB,MiB]:      7992861    7805.528     7.623\n",
      "\n",
      "Dynamically allocated memory at  amaxrestraints [B,KiB,MiB]:      8255005    8061.528     7.873\n",
      "\n",
      "Dynamically allocated memory at  amaxrestraints [B,KiB,MiB]:      8386077    8189.528     7.998\n",
      "\n",
      "Dynamically allocated memory at  amaxrestraints [B,KiB,MiB]:      8910365    8701.528     8.498\n",
      "\n",
      "Dynamically allocated memory at  amaxrestraints [B,KiB,MiB]:      9172509    8957.528     8.748\n",
      "\n",
      "Dynamically allocated memory at  amaxrestraints [B,KiB,MiB]:      9434653    9213.528     8.998\n",
      "\n",
      "Dynamically allocated memory at  amaxrestraints [B,KiB,MiB]:      9958941    9725.528     9.498\n",
      "\n",
      "Dynamically allocated memory at  amaxrestraints [B,KiB,MiB]:     11007517   10749.528    10.498\n",
      "\n",
      "Dynamically allocated memory at  amaxrestraints [B,KiB,MiB]:     11531805   11261.528    10.998\n",
      "\n",
      "Dynamically allocated memory at  amaxrestraints [B,KiB,MiB]:     12056093   11773.528    11.498\n",
      "\n",
      "Dynamically allocated memory at  amaxrestraints [B,KiB,MiB]:     13104669   12797.528    12.498\n",
      "r_stere_606_> Stereochemical restraints were constructed from RTF & PRMF.\n",
      "              Added bond,angle,dihedral,improper restraints  :    16772   22899   25995    7185\n",
      "              Total number of restraints before, now         :        0    72851\n",
      "make_re_422_> Number of previous, current restraints         :        0    72851\n",
      "make_re_423_> Number of previous, current selected restraints:        0    72851\n",
      "make_re_417_> Restraint type to be calculated:  phi-psi_binormal\n",
      "\n",
      "Dynamically allocated memory at   amaxstructure [B,KiB,MiB]:     14279623   13944.944    13.618\n",
      "openf___224_> Open           ${MODINSTALL9v25}/modlib/mnch1.bin\n",
      "openf___224_> Open           ${MODINSTALL9v25}/modlib/mnch1.mdt\n",
      "getdata_643_> Protein accepted:  6pv7_clean\n",
      "getdata_289_> Proteins (all/accepted):        1        1\n",
      "\n",
      "Dynamically allocated memory at  amaxrestraints [B,KiB,MiB]:     16376791   15992.960    15.618\n",
      "make_re_422_> Number of previous, current restraints         :    72851    74836\n",
      "make_re_423_> Number of previous, current selected restraints:    72851    74836\n",
      "make_re_417_> Restraint type to be calculated:  omega_dihedral\n",
      "\n",
      "Dynamically allocated memory at   amaxstructure [B,KiB,MiB]:     16376791   15992.960    15.618\n",
      "openf___224_> Open           ${MODINSTALL9v25}/modlib/omega.bin\n",
      "openf___224_> Open           ${MODINSTALL9v25}/modlib/omega.mdt\n",
      "getdata_643_> Protein accepted:  6pv7_clean\n",
      "getdata_289_> Proteins (all/accepted):        1        1\n",
      "omgdel__425_> Unselected all O C +N +CA dihedrals:     2085\n",
      "              (This is to avoid clashes between STEREO\n",
      "              and OMEGA_DIHEDRAL restraints)\n",
      "make_re_422_> Number of previous, current restraints         :    74836    76831\n",
      "make_re_423_> Number of previous, current selected restraints:    74836    74746\n",
      "make_re_417_> Restraint type to be calculated:  chi1_dihedral\n",
      "\n",
      "Dynamically allocated memory at   amaxstructure [B,KiB,MiB]:     16376791   15992.960    15.618\n",
      "openf___224_> Open           ${MODINSTALL9v25}/modlib/chi1234.bin\n",
      "openf___224_> Open           ${MODINSTALL9v25}/modlib/chi1.mdt\n",
      "getdata_643_> Protein accepted:  6pv7_clean\n",
      "getdata_289_> Proteins (all/accepted):        1        1\n",
      "\n",
      "Dynamically allocated memory at  amaxrestraints [B,KiB,MiB]:     17425367   17016.961    16.618\n",
      "make_re_422_> Number of previous, current restraints         :    76831    78686\n",
      "make_re_423_> Number of previous, current selected restraints:    74746    76601\n",
      "make_re_417_> Restraint type to be calculated:  chi2_dihedral\n",
      "\n",
      "Dynamically allocated memory at   amaxstructure [B,KiB,MiB]:     17425367   17016.961    16.618\n",
      "openf___224_> Open           ${MODINSTALL9v25}/modlib/chi1234.bin\n",
      "openf___224_> Open           ${MODINSTALL9v25}/modlib/chi2.mdt\n",
      "getdata_643_> Protein accepted:  6pv7_clean\n",
      "getdata_289_> Proteins (all/accepted):        1        1\n",
      "\n",
      "Dynamically allocated memory at  amaxrestraints [B,KiB,MiB]:     18473943   18040.961    17.618\n",
      "make_re_422_> Number of previous, current restraints         :    78686    80051\n",
      "make_re_423_> Number of previous, current selected restraints:    76601    77966\n",
      "make_re_417_> Restraint type to be calculated:  chi3_dihedral\n",
      "\n",
      "Dynamically allocated memory at   amaxstructure [B,KiB,MiB]:     18473943   18040.961    17.618\n",
      "openf___224_> Open           ${MODINSTALL9v25}/modlib/chi1234.bin\n",
      "openf___224_> Open           ${MODINSTALL9v25}/modlib/chi3.mdt\n",
      "getdata_643_> Protein accepted:  6pv7_clean\n",
      "getdata_289_> Proteins (all/accepted):        1        1\n",
      "make_re_422_> Number of previous, current restraints         :    80051    80516\n",
      "make_re_423_> Number of previous, current selected restraints:    77966    78431\n",
      "make_re_417_> Restraint type to be calculated:  chi4_dihedral\n",
      "\n",
      "Dynamically allocated memory at   amaxstructure [B,KiB,MiB]:     18473943   18040.961    17.618\n",
      "openf___224_> Open           ${MODINSTALL9v25}/modlib/chi1234.bin\n",
      "openf___224_> Open           ${MODINSTALL9v25}/modlib/chi4.mdt\n",
      "mdtrsr__446W> A potential that relies on one protein is used, yet you have at\n",
      "              least one known structure available. MDT, not library, potential is used.\n",
      "getdata_643_> Protein accepted:  6pv7_clean\n",
      "getdata_289_> Proteins (all/accepted):        1        1\n",
      "make_re_422_> Number of previous, current restraints         :    80516    80701\n",
      "make_re_423_> Number of previous, current selected restraints:    78431    78616\n",
      "make_re_417_> Restraint type to be calculated:  DISTANCE\n",
      "\n",
      "Dynamically allocated memory at amaxhash_contac [B,KiB,MiB]:     23185587   22642.176    22.111\n",
      "make_re_422_> Number of previous, current restraints         :    80701   120782\n",
      "make_re_423_> Number of previous, current selected restraints:    78616   118697\n",
      "make_re_417_> Restraint type to be calculated:  DISTANCE\n",
      "\n",
      "Dynamically allocated memory at  amaxrestraints [B,KiB,MiB]:     25282739   24690.176    24.111\n",
      "\n",
      "Dynamically allocated memory at  amaxrestraints [B,KiB,MiB]:     29477043   28786.176    28.111\n",
      "make_re_422_> Number of previous, current restraints         :   120782   158939\n",
      "make_re_423_> Number of previous, current selected restraints:   118697   156854\n",
      "make_re_417_> Restraint type to be calculated:  DISTANCE\n",
      "\n",
      "Dynamically allocated memory at  amaxrestraints [B,KiB,MiB]:     31574195   30834.176    30.111\n",
      "make_re_422_> Number of previous, current restraints         :   158939   183459\n",
      "make_re_423_> Number of previous, current selected restraints:   156854   181374\n",
      "make_re_417_> Restraint type to be calculated:  DISTANCE\n",
      "make_re_422_> Number of previous, current restraints         :   183459   193152\n",
      "make_re_423_> Number of previous, current selected restraints:   181374   191067\n",
      "0 atoms in HETATM/BLK residues constrained\n",
      "to protein atoms within 2.30 angstroms\n",
      "and protein CA atoms within 10.00 angstroms\n",
      "make_re_417_> Restraint type to be calculated:  DISTANCE\n",
      "make_re_422_> Number of previous, current restraints         :   193152   193152\n",
      "make_re_423_> Number of previous, current selected restraints:   191067   191067\n",
      "make_re_417_> Restraint type to be calculated:  DISTANCE\n",
      "make_re_422_> Number of previous, current restraints         :   193152   193152\n",
      "make_re_423_> Number of previous, current selected restraints:   191067   191067\n",
      "make_re_417_> Restraint type to be calculated:  DISTANCE\n",
      "make_re_422_> Number of previous, current restraints         :   193152   193152\n",
      "make_re_423_> Number of previous, current selected restraints:   191067   191067\n",
      "make_re_417_> Restraint type to be calculated:  DISTANCE\n",
      "make_re_422_> Number of previous, current restraints         :   193152   193152\n",
      "make_re_423_> Number of previous, current selected restraints:   191067   191067\n",
      "0 atoms in residues without defined topology\n",
      "constrained to be rigid bodies\n",
      "make_re_417_> Restraint type to be calculated:  DISTANCE\n",
      "make_re_422_> Number of previous, current restraints         :   193152   193152\n",
      "make_re_423_> Number of previous, current selected restraints:   191067   191067\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "rmdupl__427_>    12274 redundant cosine dihedral restraints were unselected.\n",
      "condens_443_> Restraints marked for deletion were removed.\n",
      "              Total number of restraints before, now:   193152   178793\n",
      "openf___224_> Open           muscle_nAChR.rsr\n",
      "openf___224_> Open           muscle_nAChR.rsr\n",
      "\n",
      "Dynamically allocated memory at  amaxrestraints [B,KiB,MiB]:     32237901   31482.324    30.744\n",
      "rdcsr2__307_> Number of restraints read    :   178793\n",
      "              Number of excluded pairs read:        0\n",
      "              Number of pseudo atoms read  :        0\n",
      "rdcsrs__304_> Restraints in memory, selected restraints:   178793   178793\n",
      "              Explicitly excluded atom pairs in memory :        0\n",
      "              Pseudo atoms in memory                   :        0\n",
      "openf___224_> Open           muscle_nAChR.ini\n",
      "\n",
      "Dynamically allocated memory at       amaxmodel [B,KiB,MiB]:     32237925   31482.348    30.744\n",
      "read_mo_297_> Segments, residues, atoms:       10     2005    16360\n",
      "read_mo_298_> Segment:        1     1 A   338 A     2755\n",
      "read_mo_298_> Segment:        2   339 B   401 B      517\n",
      "read_mo_298_> Segment:        3   402 C   739 C     2755\n",
      "read_mo_298_> Segment:        4   740 D   802 D      517\n",
      "read_mo_298_> Segment:        5   803 E  1140 E     2755\n",
      "read_mo_298_> Segment:        6  1141 F  1203 F      517\n",
      "read_mo_298_> Segment:        7  1204 G  1541 G     2755\n",
      "read_mo_298_> Segment:        8  1542 H  1604 H      517\n",
      "read_mo_298_> Segment:        9  1605 I  1942 I     2755\n",
      "read_mo_298_> Segment:       10  1943 J  2005 J      517\n",
      "randomi_498_> Atoms,selected atoms,random_seed,amplitude:    16360    16360        1        4.0000\n",
      "randomi_496_> Amplitude is > 0; randomization is done.\n",
      "pick_re_612_> Number of MODEL atoms, selected restraints;    16360    97310\n",
      "pick_re_612_> Number of MODEL atoms, selected restraints;    16360   108036\n",
      "pick_re_612_> Number of MODEL atoms, selected restraints;    16360   113583\n",
      "pick_re_612_> Number of MODEL atoms, selected restraints;    16360   119124\n",
      "pick_re_612_> Number of MODEL atoms, selected restraints;    16360   123454\n",
      "pick_re_612_> Number of MODEL atoms, selected restraints;    16360   127910\n",
      "pick_re_612_> Number of MODEL atoms, selected restraints;    16360   138169\n",
      "pick_re_612_> Number of MODEL atoms, selected restraints;    16360   150793\n",
      "pick_re_612_> Number of MODEL atoms, selected restraints;    16360   155971\n",
      "pick_re_612_> Number of MODEL atoms, selected restraints;    16360   163177\n",
      "pick_re_612_> Number of MODEL atoms, selected restraints;    16360   164131\n",
      "pick_re_612_> Number of MODEL atoms, selected restraints;    16360   175723\n",
      "\n",
      "Dynamically allocated memory at  amaxrestraints [B,KiB,MiB]:     32923145   32151.508    31.398\n",
      "pick_re_612_> Number of MODEL atoms, selected restraints;    16360   175813\n",
      "pick_re_612_> Number of MODEL atoms, selected restraints;    16360   175874\n",
      "pick_re_612_> Number of MODEL atoms, selected restraints;    16360   178793\n",
      "iupac_m_487_> NH1/2 swapped:      178.1285        8     8\n",
      "iupac_m_485_> OD1/2 will be swapped:     -179.3071       12    12\n",
      "iupac_m_487_> NH1/2 swapped:      179.7710       19    19\n",
      "iupac_m_487_> NH1/2 swapped:     -174.8329       22    22\n",
      "iupac_m_485_> OD1/2 will be swapped:     -121.4969       29    29\n",
      "iupac_m_487_> NH1/2 swapped:     -175.7505       38    38\n",
      "iupac_m_486_> OE1/2 will be swapped:     -173.6051       61    61\n",
      "iupac_m_485_> OD1/2 will be swapped:     -159.9812       72    72\n",
      "iupac_m_485_> OD1/2 will be swapped:      148.8574       78    78\n",
      "iupac_m_486_> OE1/2 will be swapped:     -175.8397       85    85\n",
      "iupac_m_486_> OE1/2 will be swapped:       92.8851      103   103\n",
      "iupac_m_486_> OE1/2 will be swapped:      166.7984      135   135\n",
      "iupac_m_487_> NH1/2 swapped:     -179.9802      161   161\n",
      "iupac_m_486_> OE1/2 will be swapped:     -169.4280      167   167\n",
      "iupac_m_485_> OD1/2 will be swapped:      127.7164      168   168\n",
      "iupac_m_486_> OE1/2 will be swapped:      169.4186      173   173\n",
      "iupac_m_485_> OD1/2 will be swapped:      176.9832      188   188\n",
      "iupac_m_487_> NH1/2 swapped:     -179.1352      196   196\n",
      "iupac_m_486_> OE1/2 will be swapped:      -95.7396      205   205\n",
      "iupac_m_487_> NH1/2 swapped:     -177.9583      218   218\n",
      "iupac_m_487_> NH1/2 swapped:      177.3210      219   219\n",
      "iupac_m_487_> NH1/2 swapped:     -178.7485      317   317\n",
      "iupac_m_487_> NH1/2 swapped:     -179.6310      320   320\n",
      "iupac_m_486_> OE1/2 will be swapped:     -172.6263      341   341\n",
      "iupac_m_485_> OD1/2 will be swapped:     -125.4194      347   347\n",
      "iupac_m_486_> OE1/2 will be swapped:     -163.5797      363   363\n",
      "iupac_m_486_> OE1/2 will be swapped:     -176.4429      367   367\n",
      "iupac_m_483_> CD1/2 CE1/2 swapped:     -120.3393      418   418\n",
      "iupac_m_397W> Atoms were not swapped because of the uncertainty of how to handle the H atom.\n",
      "iupac_m_487_> NH1/2 swapped:     -178.8911      439   439\n",
      "iupac_m_486_> OE1/2 will be swapped:      144.7227      460   460\n",
      "iupac_m_485_> OD1/2 will be swapped:      139.3439      474   474\n",
      "iupac_m_483_> CD1/2 CE1/2 swapped:       90.7168      557   557\n",
      "iupac_m_397W> Atoms were not swapped because of the uncertainty of how to handle the H atom.\n",
      "iupac_m_487_> NH1/2 swapped:     -178.3229      562   562\n",
      "iupac_m_486_> OE1/2 will be swapped:      162.0980      587   587\n",
      "iupac_m_487_> NH1/2 swapped:      179.5559      597   597\n",
      "iupac_m_483_> CD1/2 CE1/2 swapped:       90.0692      601   601\n",
      "iupac_m_397W> Atoms were not swapped because of the uncertainty of how to handle the H atom.\n",
      "iupac_m_487_> NH1/2 swapped:      171.6272      620   620\n",
      "iupac_m_486_> OE1/2 will be swapped:     -166.4726      673   673\n",
      "iupac_m_487_> NH1/2 swapped:      176.7220      718   718\n",
      "iupac_m_487_> NH1/2 swapped:      179.5248      721   721\n",
      "iupac_m_487_> NH1/2 swapped:     -179.4811      733   733\n",
      "iupac_m_486_> OE1/2 will be swapped:      179.2889      738   738\n",
      "iupac_m_487_> NH1/2 swapped:      178.1649      739   739\n",
      "iupac_m_487_> NH1/2 swapped:      178.5714      741   741\n",
      "iupac_m_485_> OD1/2 will be swapped:     -122.4607      749   749\n",
      "iupac_m_487_> NH1/2 swapped:     -179.5450      810   810\n",
      "iupac_m_485_> OD1/2 will be swapped:      121.3207      813   813\n",
      "iupac_m_487_> NH1/2 swapped:     -177.0615      821   821\n",
      "iupac_m_487_> NH1/2 swapped:     -179.9515      832   832\n",
      "iupac_m_487_> NH1/2 swapped:     -178.2451      840   840\n",
      "iupac_m_485_> OD1/2 will be swapped:     -175.6588      846   846\n",
      "iupac_m_485_> OD1/2 will be swapped:      109.2814      866   866\n",
      "iupac_m_486_> OE1/2 will be swapped:     -129.4059      887   887\n",
      "iupac_m_483_> CD1/2 CE1/2 swapped:      -91.8151      941   941\n",
      "iupac_m_397W> Atoms were not swapped because of the uncertainty of how to handle the H atom.\n",
      "iupac_m_486_> OE1/2 will be swapped:     -140.7607      969   969\n",
      "iupac_m_486_> OE1/2 will be swapped:      146.3187      975   975\n",
      "iupac_m_485_> OD1/2 will be swapped:      118.5565      982   982\n",
      "iupac_m_483_> CD1/2 CE1/2 swapped:      101.3386      983   983\n",
      "iupac_m_397W> Atoms were not swapped because of the uncertainty of how to handle the H atom.\n",
      "iupac_m_486_> OE1/2 will be swapped:     -160.9536      988   988\n",
      "iupac_m_487_> NH1/2 swapped:     -178.1873      998   998\n",
      "iupac_m_486_> OE1/2 will be swapped:     -121.7939     1000  1000\n",
      "iupac_m_483_> CD1/2 CE1/2 swapped:       95.2310     1003  1003\n",
      "iupac_m_397W> Atoms were not swapped because of the uncertainty of how to handle the H atom.\n",
      "iupac_m_486_> OE1/2 will be swapped:     -162.9409     1008  1008\n",
      "iupac_m_483_> CD1/2 CE1/2 swapped:       90.6544     1010  1010\n",
      "iupac_m_397W> Atoms were not swapped because of the uncertainty of how to handle the H atom.\n",
      "iupac_m_487_> NH1/2 swapped:      179.4195     1020  1020\n",
      "iupac_m_486_> OE1/2 will be swapped:     -177.6629     1074  1074\n",
      "iupac_m_487_> NH1/2 swapped:      179.3915     1134  1134\n",
      "iupac_m_487_> NH1/2 swapped:     -179.7490     1140  1140\n",
      "iupac_m_487_> NH1/2 swapped:     -179.4633     1142  1142\n",
      "iupac_m_485_> OD1/2 will be swapped:     -155.2131     1150  1150\n",
      "iupac_m_486_> OE1/2 will be swapped:     -171.2066     1165  1165\n",
      "iupac_m_486_> OE1/2 will be swapped:     -111.2496     1208  1208\n",
      "iupac_m_487_> NH1/2 swapped:      167.1221     1211  1211\n",
      "iupac_m_485_> OD1/2 will be swapped:     -117.9906     1214  1214\n",
      "iupac_m_487_> NH1/2 swapped:     -178.6918     1222  1222\n",
      "iupac_m_487_> NH1/2 swapped:      176.8353     1225  1225\n",
      "iupac_m_487_> NH1/2 swapped:      177.2565     1233  1233\n",
      "iupac_m_487_> NH1/2 swapped:      179.2287     1241  1241\n",
      "iupac_m_486_> OE1/2 will be swapped:      176.6292     1264  1264\n",
      "iupac_m_485_> OD1/2 will be swapped:      178.3042     1275  1275\n",
      "iupac_m_485_> OD1/2 will be swapped:      179.1851     1276  1276\n",
      "iupac_m_485_> OD1/2 will be swapped:      136.5284     1281  1281\n",
      "iupac_m_485_> OD1/2 will be swapped:      113.1801     1302  1302\n",
      "iupac_m_483_> CD1/2 CE1/2 swapped:      112.2791     1332  1332\n",
      "iupac_m_397W> Atoms were not swapped because of the uncertainty of how to handle the H atom.\n",
      "iupac_m_486_> OE1/2 will be swapped:     -118.0697     1338  1338\n",
      "iupac_m_487_> NH1/2 swapped:      179.1734     1364  1364\n",
      "iupac_m_485_> OD1/2 will be swapped:      173.7364     1371  1371\n",
      "iupac_m_486_> OE1/2 will be swapped:     -154.0714     1376  1376\n",
      "iupac_m_485_> OD1/2 will be swapped:      169.4288     1380  1380\n",
      "iupac_m_485_> OD1/2 will be swapped:      117.6048     1383  1383\n",
      "iupac_m_486_> OE1/2 will be swapped:     -174.7315     1401  1401\n",
      "iupac_m_486_> OE1/2 will be swapped:      122.8406     1408  1408\n",
      "iupac_m_487_> NH1/2 swapped:      179.7693     1421  1421\n",
      "iupac_m_487_> NH1/2 swapped:     -179.9657     1523  1523\n",
      "iupac_m_487_> NH1/2 swapped:     -179.2854     1526  1526\n",
      "iupac_m_486_> OE1/2 will be swapped:     -160.9043     1540  1540\n",
      "iupac_m_487_> NH1/2 swapped:     -178.3926     1541  1541\n",
      "iupac_m_486_> OE1/2 will be swapped:      179.9315     1544  1544\n",
      "iupac_m_486_> OE1/2 will be swapped:     -168.6456     1546  1546\n",
      "iupac_m_485_> OD1/2 will be swapped:     -158.1827     1551  1551\n",
      "iupac_m_487_> NH1/2 swapped:      178.7623     1553  1553\n",
      "iupac_m_485_> OD1/2 will be swapped:     -162.7517     1563  1563\n",
      "iupac_m_486_> OE1/2 will be swapped:     -117.6072     1569  1569\n",
      "iupac_m_487_> NH1/2 swapped:     -178.6271     1581  1581\n",
      "iupac_m_487_> NH1/2 swapped:     -179.8847     1612  1612\n",
      "iupac_m_487_> NH1/2 swapped:      179.9374     1623  1623\n",
      "iupac_m_487_> NH1/2 swapped:      177.8643     1626  1626\n",
      "iupac_m_485_> OD1/2 will be swapped:      125.3831     1633  1633\n",
      "iupac_m_487_> NH1/2 swapped:      179.9802     1642  1642\n",
      "iupac_m_486_> OE1/2 will be swapped:       99.9546     1663  1663\n",
      "iupac_m_485_> OD1/2 will be swapped:     -161.7828     1677  1677\n",
      "iupac_m_485_> OD1/2 will be swapped:      140.9501     1737  1737\n",
      "iupac_m_486_> OE1/2 will be swapped:      178.4177     1745  1745\n",
      "iupac_m_487_> NH1/2 swapped:     -178.3660     1765  1765\n",
      "iupac_m_485_> OD1/2 will be swapped:     -168.0374     1781  1781\n",
      "iupac_m_483_> CD1/2 CE1/2 swapped:       90.6662     1786  1786\n",
      "iupac_m_397W> Atoms were not swapped because of the uncertainty of how to handle the H atom.\n",
      "iupac_m_486_> OE1/2 will be swapped:     -162.4927     1809  1809\n",
      "iupac_m_487_> NH1/2 swapped:     -176.5371     1822  1822\n",
      "iupac_m_486_> OE1/2 will be swapped:      176.4820     1876  1876\n",
      "iupac_m_487_> NH1/2 swapped:     -176.8458     1915  1915\n",
      "iupac_m_487_> NH1/2 swapped:      179.4976     1921  1921\n",
      "iupac_m_487_> NH1/2 swapped:      179.9011     1924  1924\n",
      "iupac_m_487_> NH1/2 swapped:     -179.2102     1927  1927\n",
      "iupac_m_487_> NH1/2 swapped:      177.3717     1942  1942\n",
      "iupac_m_486_> OE1/2 will be swapped:      177.9670     1945  1945\n",
      "iupac_m_485_> OD1/2 will be swapped:      165.9637     1951  1951\n",
      "iupac_m_485_> OD1/2 will be swapped:      125.8501     1952  1952\n",
      "iupac_m_483_> CD1/2 CE1/2 swapped:      101.2996     1966  1966\n",
      "iupac_m_397W> Atoms were not swapped because of the uncertainty of how to handle the H atom.\n",
      "iupac_m_486_> OE1/2 will be swapped:      101.1067     1971  1971\n",
      "\n",
      "\n",
      ">> ENERGY; Differences between the model's features and restraints:\n",
      "Number of all residues in MODEL                   :     2005\n",
      "Number of all, selected real atoms                :    16360   16360\n",
      "Number of all, selected pseudo atoms              :        0       0\n",
      "Number of all static, selected restraints         :   178793  178793\n",
      "COVALENT_CYS                                      :        F\n",
      "NONBONDED_SEL_ATOMS                               :        1\n",
      "Number of non-bonded pairs (excluding 1-2,1-3,1-4):    33729\n",
      "Dynamic pairs routine                             : 2, NATM x NATM cell sorting\n",
      "Atomic shift for contacts update (UPDATE_DYNAMIC) :    0.390\n",
      "LENNARD_JONES_SWITCH                              :    6.500   7.500\n",
      "COULOMB_JONES_SWITCH                              :    6.500   7.500\n",
      "RESIDUE_SPAN_RANGE                                :        0   99999\n",
      "NLOGN_USE                                         :       15\n",
      "CONTACT_SHELL                                     :    4.000\n",
      "DYNAMIC_PAIRS,_SPHERE,_COULOMB,_LENNARD,_MODELLER :        T       T       F       F       F\n",
      "SPHERE_STDV                                       :    0.050\n",
      "RADII_FACTOR                                      :    0.820\n",
      "Current energy                                    :       12277.3945\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Summary of the restraint violations: \n",
      "\n",
      "   NUM     ... number of restraints.\n",
      "   NUMVI   ... number of restraints with RVIOL > VIOL_REPORT_CUT[i].\n",
      "   RVIOL   ... relative difference from the best value.\n",
      "   NUMVP   ... number of restraints with -Ln(pdf) > VIOL_REPORT_CUT2[i].\n",
      "   RMS_1   ... RMS(feature, minimally_violated_basis_restraint, NUMB).\n",
      "   RMS_2   ... RMS(feature, best_value, NUMB).\n",
      "   MOL.PDF ... scaled contribution to -Ln(Molecular pdf).\n",
      "\n",
      " #                     RESTRAINT_GROUP      NUM   NUMVI  NUMVP   RMS_1   RMS_2         MOL.PDF     S_i\n",
      "------------------------------------------------------------------------------------------------------\n",
      " 1 Bond length potential              :   16765       0      1   0.006   0.006      171.33       1.000\n",
      " 2 Bond angle potential               :   22885       3    120   2.140   2.140      2054.4       1.000\n",
      " 3 Stereochemical cosine torsion poten:   11629       0    310  45.778  45.778      3807.8       1.000\n",
      " 4 Stereochemical improper torsion pot:    7185       1      9   1.429   1.429      345.76       1.000\n",
      " 5 Soft-sphere overlap restraints     :   33729       0      6   0.005   0.005      93.022       1.000\n",
      " 6 Lennard-Jones 6-12 potential       :       0       0      0   0.000   0.000      0.0000       1.000\n",
      " 7 Coulomb point-point electrostatic p:       0       0      0   0.000   0.000      0.0000       1.000\n",
      " 8 H-bonding potential                :       0       0      0   0.000   0.000      0.0000       1.000\n",
      " 9 Distance restraints 1 (CA-CA)      :   40081       0      9   0.143   0.143      683.28       1.000\n",
      "10 Distance restraints 2 (N-O)        :   38157       7     59   0.218   0.218      1614.6       1.000\n",
      "11 Mainchain Phi dihedral restraints  :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "12 Mainchain Psi dihedral restraints  :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "13 Mainchain Omega dihedral restraints:    1995       0     18   4.344   4.344      440.37       1.000\n",
      "14 Sidechain Chi_1 dihedral restraints:    1855       0     20  68.217  68.217      349.87       1.000\n",
      "15 Sidechain Chi_2 dihedral restraints:    1365       0      6  62.611  62.611      386.87       1.000\n",
      "16 Sidechain Chi_3 dihedral restraints:     465       0      0  69.893  69.893      271.08       1.000\n",
      "17 Sidechain Chi_4 dihedral restraints:     185       0      0  95.291  95.291      118.74       1.000\n",
      "18 Disulfide distance restraints      :       7       0      0   0.015   0.015     0.26545       1.000\n",
      "19 Disulfide angle restraints         :      14       0      0   1.849   1.849      1.0573       1.000\n",
      "20 Disulfide dihedral angle restraints:       7       0      0  27.443  27.443      3.8605       1.000\n",
      "21 Lower bound distance restraints    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "22 Upper bound distance restraints    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "23 Distance restraints 3 (SDCH-MNCH)  :   24520       0      0   0.327   0.327      506.25       1.000\n",
      "24 Sidechain Chi_5 dihedral restraints:       0       0      0   0.000   0.000      0.0000       1.000\n",
      "25 Phi/Psi pair of dihedral restraints:    1985      79    242  26.862  40.962      724.31       1.000\n",
      "26 Distance restraints 4 (SDCH-SDCH)  :    9693       0     18   0.683   0.683      704.52       1.000\n",
      "27 Distance restraints 5 (X-Y)        :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "28 NMR distance restraints 6 (X-Y)    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "29 NMR distance restraints 7 (X-Y)    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "30 Minimal distance restraints        :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "31 Non-bonded restraints              :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "32 Atomic accessibility restraints    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "33 Atomic density restraints          :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "34 Absolute position restraints       :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "35 Dihedral angle difference restraint:       0       0      0   0.000   0.000      0.0000       1.000\n",
      "36 GBSA implicit solvent potential    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "37 EM density fitting potential       :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "38 SAXS restraints                    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "39 Symmetry restraints                :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "\n",
      "\n",
      "\n",
      "openf___224_> Open           muscle_nAChR.V99990001\n",
      "# Heavy relative violation of each residue is written to: muscle_nAChR.V99990001\n",
      "# The profile is NOT normalized by the number of restraints.\n",
      "# The profiles are smoothed over a window of residues:    1\n",
      "# The sum of all numbers in the file:  188220.9531\n",
      "\n",
      "\n",
      "\n",
      "List of the violated restraints:\n",
      "   A restraint is violated when the relative difference\n",
      "   from the best value (RVIOL) is larger than CUTOFF.\n",
      "\n",
      "   ICSR   ... index of a restraint in the current set.\n",
      "   RESNO  ... residue numbers of the first two atoms.\n",
      "   ATM    ... IUPAC atom names of the first two atoms.\n",
      "   FEAT   ... the value of the feature in the model.\n",
      "   restr  ... the mean of the basis restraint with the smallest\n",
      "              difference from the model (local minimum).\n",
      "   viol   ... difference from the local minimum.\n",
      "   rviol  ... relative difference from the local minimum.\n",
      "   RESTR  ... the best value (global minimum).\n",
      "   VIOL   ... difference from the best value.\n",
      "   RVIOL  ... relative difference from the best value.\n",
      "\n",
      "\n",
      "-------------------------------------------------------------------------------------------------\n",
      "\n",
      "Feature  2                           : Bond angle potential                    \n",
      "List of the RVIOL violations larger than   :       4.5000\n",
      "\n",
      "    #   ICSR  RESNO1/2 ATM1/2   INDATM1/2    FEAT   restr    viol   rviol   RESTR    VIOL   RVIOL\n",
      "    1  23324 572Q 572Q CB  CG   4675 4676  130.02  108.00   22.01    5.10  108.00   22.01    5.10\n",
      "    2  371281781D1782L C   N   1454014542  140.04  120.00   20.04    4.56  120.00   20.04    4.56\n",
      "\n",
      "-------------------------------------------------------------------------------------------------\n",
      "\n",
      "Feature 10                           : Distance restraints 2 (N-O)             \n",
      "List of the RVIOL violations larger than   :       4.5000\n",
      "\n",
      "    #   ICSR  RESNO1/2 ATM1/2   INDATM1/2    FEAT   restr    viol   rviol   RESTR    VIOL   RVIOL\n",
      "    1 109894 176I 192V N   O    1438 1584    5.30    3.30    2.00    7.51    3.30    2.00    7.51\n",
      "    2 117528 577I 593V N   O    4710 4856    5.04    3.22    1.82    5.19    3.22    1.82    5.19\n",
      "    3 1326591379I1395V N   O   1125411400    4.82    3.23    1.59    5.97    3.23    1.59    5.97\n",
      "\n",
      "-------------------------------------------------------------------------------------------------\n",
      "\n",
      "Feature 25                           : Phi/Psi pair of dihedral restraints     \n",
      "List of the RVIOL violations larger than   :       6.5000\n",
      "\n",
      "    #   ICSR  RESNO1/2 ATM1/2   INDATM1/2    FEAT   restr    viol   rviol   RESTR    VIOL   RVIOL\n",
      "    1  58493   1S   2A C   N       5    7  -68.40  -68.20    8.42    0.67  -62.50  165.48   27.41\n",
      "    1          2A   2A N   CA      7    8  153.72  145.30                  -40.90\n",
      "    2  58514  22R  23P C   N     182  184  -79.34  -58.70   71.10    4.46  -64.50  110.66    9.06\n",
      "    2         23P  23P N   CA    184  185   37.54  -30.50                  147.20\n",
      "    3  58658 166T 167E C   N    1368 1370 -107.97 -117.80   88.94    4.28  -63.60   99.18   11.51\n",
      "    3        167E 167E N   CA   1370 1371   48.40  136.80                  -40.30\n",
      "    4  58659 167E 168D C   N    1377 1379  -85.38  -96.50   25.74    1.08  -63.30  132.84   15.19\n",
      "    4        168D 168D N   CA   1379 1380   90.99  114.20                  -40.00\n",
      "    5  58660 168D 169S C   N    1385 1387 -148.56 -136.60   63.74    3.45  -64.10  149.69   10.39\n",
      "    5        169S 169S N   CA   1387 1388   88.59  151.20                  -35.00\n",
      "    6  58664 172I 173E C   N    1416 1418   69.78  -69.30  141.80    9.74  -63.60 -155.37   34.29\n",
      "    6        173E 173E N   CA   1418 1419  114.89  142.50                  -40.30\n",
      "    7  58666 174V 175G C   N    1432 1434 -152.86 -167.20   47.17    1.61   82.20  174.04   13.38\n",
      "    7        175G 175G N   CA   1434 1435  129.66  174.60                    8.50\n",
      "    8  58667 175G 176I C   N    1436 1438  -31.92  -97.30  168.39   16.05 -120.60  176.05   13.17\n",
      "    8        176I 176I N   CA   1438 1439  -77.62  127.20                  130.30\n",
      "    9  58676 184S 185V C   N    1513 1515   86.18   55.90   61.68    2.96 -125.40 -143.56   12.33\n",
      "    9        185V 185V N   CA   1515 1516  -14.23   39.50                  143.30\n",
      "   10  58829 339V 340R C   N    2761 2763  -77.92  -72.10    5.84    0.44  -63.00  177.13   24.95\n",
      "   10        340R 340R N   CA   2763 2764  142.40  141.90                  -41.10\n",
      "   11  58830 340R 341E C   N    2772 2774 -119.83 -117.80    9.66    0.44  -63.60 -177.66   27.49\n",
      "   11        341E 341E N   CA   2774 2775  146.25  136.80                  -40.30\n",
      "   12  58903 415L 416S C   N    3382 3384  -78.47  -72.40   86.82    5.32  -64.10  101.81    6.66\n",
      "   12        416S 416S N   CA   3384 3385   65.79  152.40                  -35.00\n",
      "   13  58904 416S 417N C   N    3388 3390  178.95 -119.90  126.55    4.21  -63.20  137.65   22.63\n",
      "   13        417N 417N N   CA   3390 3391 -112.21  137.00                  -41.10\n",
      "   14  58916 428N 429S C   N    3488 3490   52.86   56.90   16.88    0.78  -64.10  146.25   18.80\n",
      "   14        429S 429S N   CA   3490 3491   52.79   36.40                  -35.00\n",
      "   15  58933 445D 446V C   N    3622 3624  -94.85  -62.40   67.19    7.35  -62.40   67.19    7.35\n",
      "   15        446V 446V N   CA   3624 3625   16.43  -42.40                  -42.40\n",
      "   16  59055 567T 568E C   N    4640 4642 -137.62 -117.80   22.27    0.65  -63.60 -172.06   29.22\n",
      "   16        568E 568E N   CA   4642 4643  146.95  136.80                  -40.30\n",
      "   17  59056 568E 569D C   N    4649 4651  176.45  -96.50  117.76    4.82  -63.30  141.67   17.76\n",
      "   17        569D 569D N   CA   4651 4652   34.89  114.20                  -40.00\n",
      "   18  59057 569D 570S C   N    4657 4659 -122.66 -136.60   46.40    2.02  -64.10  153.55    9.53\n",
      "   18        570S 570S N   CA   4659 4660  106.95  151.20                  -35.00\n",
      "   19  59058 570S 571N C   N    4663 4665 -172.11 -119.90   79.42    2.36  -63.20  163.59   26.20\n",
      "   19        571N 571N N   CA   4665 4666 -163.16  137.00                  -41.10\n",
      "   20  59061 573I 574E C   N    4688 4690  -51.11  -69.30   63.55    4.19  -63.60  122.55   17.04\n",
      "   20        574E 574E N   CA   4690 4691   81.61  142.50                  -40.30\n",
      "   21  59073 585S 586V C   N    4785 4787   84.06   55.90   73.25    3.55 -125.40 -131.86   13.05\n",
      "   21        586V 586V N   CA   4787 4788  -28.12   39.50                  143.30\n",
      "   22  59089 601F 602Y C   N    4918 4920 -152.00 -124.30   56.90    2.24  -63.50  158.51   29.28\n",
      "   22        602Y 602Y N   CA   4920 4921 -174.90  135.40                  -43.40\n",
      "   23  59093 605C 606E C   N    4950 4952 -150.44 -117.80   50.30    1.70  -63.60  168.69   27.27\n",
      "   23        606E 606E N   CA   4952 4953  175.07  136.80                  -40.30\n",
      "   24  59285 799Q 800A C   N    6524 6526 -148.78 -134.00   16.84    0.39  -62.50 -174.66   34.99\n",
      "   24        800A 800A N   CA   6526 6527  155.07  147.00                  -40.90\n",
      "   25  59286 800A 801P C   N    6529 6531  -70.37  -58.70   16.49    1.16  -64.50  166.14   12.71\n",
      "   25        801P 801P N   CA   6531 6532  -18.84  -30.50                  147.20\n",
      "   26  59287 803S 804A C   N    6549 6551 -118.06 -134.00   15.96    0.56  -62.50 -179.85   32.48\n",
      "   26        804A 804A N   CA   6551 6552  147.73  147.00                  -40.90\n",
      "   27  59288 804A 805N C   N    6554 6556  -93.41  -71.20   24.33    1.94  -63.20  176.57   20.55\n",
      "   27        805N 805N N   CA   6556 6557  132.87  142.80                  -41.10\n",
      "   28  59289 805N 806S C   N    6562 6564  -77.44  -72.40   10.31    0.47  -64.10  164.14   12.73\n",
      "   28        806S 806S N   CA   6564 6565  161.40  152.40                  -35.00\n",
      "   29  59301 817S 818N C   N    6660 6662   53.24   55.90    8.85    0.82  -63.20  136.98   22.52\n",
      "   29        818N 818N N   CA   6662 6663   31.06   39.50                  -41.10\n",
      "   30  59312 828N 829N C   N    6752 6754  -62.91  -63.20   14.70    1.82 -119.90  176.65    9.12\n",
      "   30        829N 829N N   CA   6754 6755  -55.80  -41.10                  137.00\n",
      "   31  59313 829N 830S C   N    6760 6762 -120.06 -136.60   69.94    3.16  -64.10  130.81    8.25\n",
      "   31        830S 830S N   CA   6762 6763   83.24  151.20                  -35.00\n",
      "   32  59325 841L 842S C   N    6855 6857 -134.44 -136.60   70.32    3.61  -64.10  125.15   14.15\n",
      "   32        842S 842S N   CA   6857 6858 -138.52  151.20                  -35.00\n",
      "   33  59330 846D 847V C   N    6894 6896  -92.70  -62.40   67.55    7.40  -62.40   67.55    7.40\n",
      "   33        847V 847V N   CA   6896 6897   17.97  -42.40                  -42.40\n",
      "   34  59342 858V 859W C   N    6988 6990  -93.17  -63.00   62.87    6.92  -63.00   62.87    6.92\n",
      "   34        859W 859W N   CA   6990 6991   10.97  -44.20                  -44.20\n",
      "   35  59352 868K 869L C   N    7087 7089  -85.95  -63.50   53.71    6.60  -63.50   53.71    6.60\n",
      "   35        869L 869L N   CA   7089 7090    7.60  -41.20                  -41.20\n",
      "   36  59452 968T 969E C   N    7912 7914  108.58 -117.80  136.85    4.36  -63.60 -129.43   27.96\n",
      "   36        969E 969E N   CA   7914 7915  166.34  136.80                  -40.30\n",
      "   37  59453 969E 970D C   N    7921 7923 -129.54  -96.50   42.92    1.79  -63.30 -169.70   27.77\n",
      "   37        970D 970D N   CA   7923 7924  141.60  114.20                  -40.00\n",
      "   38  59454 970D 971S C   N    7929 7931 -148.37 -136.60   49.53    2.24  -64.10  151.33   17.05\n",
      "   38        971S 971S N   CA   7931 7932 -160.70  151.20                  -35.00\n",
      "   39  59455 971S 972N C   N    7935 7937 -164.67 -119.90   67.22    1.99  -63.20  166.31   26.27\n",
      "   39        972N 972N N   CA   7937 7938 -172.86  137.00                  -41.10\n",
      "   40  59456 972N 973Q C   N    7943 7945  133.96 -121.10  158.08    5.28  -63.80  173.61   31.11\n",
      "   40        973Q 973Q N   CA   7945 7946 -102.08  139.70                  -40.30\n",
      "   41  59458 974I 975E C   N    7960 7962  171.40 -117.80   94.40    2.93  -63.60  173.60   29.49\n",
      "   41        975E 975E N   CA   7962 7963 -160.76  136.80                  -40.30\n",
      "   42  59461 977G 978I C   N    7980 7982  -74.40  -97.30  153.09   15.91 -120.60  161.23    7.52\n",
      "   42        978I 978I N   CA   7982 7983  -24.17  127.20                  130.30\n",
      "   43  59470 986S 987V C   N    8057 8059   79.13   55.90   53.83    2.57 -125.40 -142.32   12.33\n",
      "   43        987V 987V N   CA   8059 8060   -9.06   39.50                  143.30\n",
      "   44  594861002F1003Y C   N    8190 8192  -64.55  -98.40   34.83    1.65  -63.50  163.62   25.91\n",
      "   44       1003Y1003Y N   CA   8192 8193  120.22  128.40                  -43.40\n",
      "   45  594881004I1005C C   N    8210 8212  -61.29  -63.00   18.04    2.09 -117.90  169.57    7.35\n",
      "   45       1005C1005C N   CA   8212 8213  -59.06  -41.10                  141.10\n",
      "   46  594891005C1006C C   N    8216 8218  -67.81  -63.00    9.88    1.56 -117.90  176.42    7.57\n",
      "   46       1006C1006C N   CA   8218 8219  -49.73  -41.10                  141.10\n",
      "   47  594901006C1007E C   N    8222 8224  -86.23  -69.30   17.15    1.20  -63.60  175.95   24.73\n",
      "   47       1007E1007E N   CA   8224 8225  145.21  142.50                  -40.30\n",
      "   48  595361052G1053E C   N    8593 8595 -104.14  -63.60   56.54    6.77  -63.60   56.54    6.77\n",
      "   48       1053E1053E N   CA   8595 8596   -0.90  -40.30                  -40.30\n",
      "   49  596821200Q1201A C   N    9796 9798  177.29 -134.00   51.60    1.21  -62.50 -163.77   38.45\n",
      "   49       1201A1201A N   CA   9798 9799  164.01  147.00                  -40.90\n",
      "   50  596841204S1205A C   N    9821 9823  -99.22  -68.20   33.68    3.22  -62.50  176.94   27.44\n",
      "   50       1205A1205A N   CA   9823 9824  132.19  145.30                  -40.90\n",
      "   51  597581278G1279G C   N   1043710439  -93.07  -62.40   38.43    5.63   82.20  177.27   12.76\n",
      "   51       1279G1279G N   CA  1043910440  -18.05  -41.20                    8.50\n",
      "   52  598491369T1370E C   N   1118411186  -93.81 -117.80   49.41    1.90  -63.60  137.28   16.83\n",
      "   52       1370E1370E N   CA  1118611187   93.61  136.80                  -40.30\n",
      "   53  598501370E1371D C   N   1119311195 -116.13  -96.50   57.70    2.40  -63.30  113.05   11.99\n",
      "   53       1371D1371D N   CA  1119511196   59.95  114.20                  -40.00\n",
      "   54  598511371D1372S C   N   1120111203  -98.27  -72.40   31.24    1.62  -64.10  158.82   13.82\n",
      "   54       1372S1372S N   CA  1120311204  169.90  152.40                  -35.00\n",
      "   55  598521372S1373N C   N   1120711209  -89.67  -71.20   21.35    1.25  -63.20  167.50   22.44\n",
      "   55       1373N1373N N   CA  1120911210  153.51  142.80                  -41.10\n",
      "   56  598531373N1374Q C   N   1121511217  166.82 -121.10   78.64    2.46  -63.80 -163.01   35.18\n",
      "   56       1374Q1374Q N   CA  1121711218  171.14  139.70                  -40.30\n",
      "   57  598541374Q1375I C   N   1122411226  -66.00  -63.40   10.49    1.82 -120.60 -175.79   12.58\n",
      "   57       1375I1375I N   CA  1122611227  -53.76  -43.60                  130.30\n",
      "   58  598551375I1376E C   N   1123211234 -108.61 -117.80   48.23    2.23  -63.60  137.34   16.30\n",
      "   58       1376E1376E N   CA  1123411235   89.45  136.80                  -40.30\n",
      "   59  598571377V1378G C   N   1124811250 -178.15   82.20  105.90    8.08   82.20  105.90    8.08\n",
      "   59       1378G1378G N   CA  1125011251   44.36    8.50                    8.50\n",
      "   60  598671387S1388V C   N   1132911331   90.42   55.90   64.20    3.15 -125.40 -146.16   12.20\n",
      "   60       1388V1388V N   CA  1133111332  -14.62   39.50                  143.30\n",
      "   61  600201542V1543R C   N   1257712579  -68.66  -72.10    4.96    0.43  -63.00  173.53   23.82\n",
      "   61       1543R1543R N   CA  1257912580  145.46  141.90                  -41.10\n",
      "   62  600211543R1544E C   N   1258812590 -142.79 -117.80  107.97    4.90  -63.60  111.06   18.84\n",
      "   62       1544E1544E N   CA  1259012591 -118.17  136.80                  -40.30\n",
      "   63  600951619S1620N C   N   1320413206  -76.40 -119.90   79.63    2.53  -63.20  112.18   13.34\n",
      "   63       1620N1620N N   CA  1320613207   70.30  137.00                  -41.10\n",
      "   64  601071631N1632S C   N   1330413306   54.52   56.90   10.25    0.47  -64.10  143.84   18.68\n",
      "   64       1632S1632S N   CA  1330613307   46.37   36.40                  -35.00\n",
      "   65  601191643L1644S C   N   1339913401 -140.66  -64.10   77.11   10.14  -64.10   77.11   10.14\n",
      "   65       1644S1644S N   CA  1340113402  -44.25  -35.00                  -35.00\n",
      "   66  601751699Y1700N C   N   1387213874  -96.24 -119.90   75.14    2.82  -63.20  111.77   12.46\n",
      "   66       1700N1700N N   CA  1387413875   65.68  137.00                  -41.10\n",
      "   67  602461770T1771E C   N   1445614458  -38.47  -69.30   32.16    2.11  -63.60  175.46   24.80\n",
      "   67       1771E1771E N   CA  1445814459  133.35  142.50                  -40.30\n",
      "   68  602471771E1772D C   N   1446514467 -136.09  -63.30   95.02   16.01  -63.30   95.02   16.01\n",
      "   68       1772D1772D N   CA  1446714468 -101.07  -40.00                  -40.00\n",
      "   69  602481772D1773S C   N   1447314475 -127.77 -136.60   12.76    0.71  -64.10  176.47   17.29\n",
      "   69       1773S1773S N   CA  1447514476  160.41  151.20                  -35.00\n",
      "   70  602491773S1774N C   N   1447914481 -168.58 -119.90   54.52    1.69  -63.20 -170.61   29.47\n",
      "   70       1774N1774N N   CA  1448114482  161.54  137.00                  -41.10\n",
      "   71  602501774N1775Q C   N   1448714489 -123.18  -63.80   61.22    9.44  -63.80   61.22    9.44\n",
      "   71       1775Q1775Q N   CA  1448914490  -25.43  -40.30                  -40.30\n",
      "   72  602521776I1777E C   N   1450414506  -74.94  -69.30   39.55    2.82  -63.60  138.52   19.10\n",
      "   72       1777E1777E N   CA  1450614507 -178.35  142.50                  -40.30\n",
      "   73  602641788S1789V C   N   1460114603   81.73   55.90   59.85    2.86 -125.40 -140.31   12.49\n",
      "   73       1789V1789V N   CA  1460314604  -14.48   39.50                  143.30\n",
      "   74  602801804F1805Y C   N   1473414736  -59.47  -98.40   39.21    1.64  -63.50  167.23   26.74\n",
      "   74       1805Y1805Y N   CA  1473614737  123.78  128.40                  -43.40\n",
      "   75  602841808C1809E C   N   1476614768  -71.51  -69.30    7.71    0.51  -63.60  169.99   23.10\n",
      "   75       1809E1809E N   CA  1476814769  149.89  142.50                  -40.30\n",
      "   76  602851809E1810E C   N   1477514777  -66.65  -63.60    3.05    0.49 -117.80 -175.48    8.16\n",
      "   76       1810E1810E N   CA  1477714778  -40.48  -40.30                  136.80\n",
      "   77  602871811P1812Y C   N   1479114793  151.06 -124.30   87.77    7.14 -124.30   87.77    7.14\n",
      "   77       1812Y1812Y N   CA  1479314794  112.20  135.40                  135.40\n",
      "   78  604752001L2002Q C   N   1633116333  -66.98  -73.00   19.72    1.18  -63.80  162.26   23.69\n",
      "   78       2002Q2002Q N   CA  1633316334  121.92  140.70                  -40.30\n",
      "   79  604762002Q2003A C   N   1634016342 -117.90 -134.00   16.61    0.71  -62.50  176.91   31.94\n",
      "   79       2003A2003A N   CA  1634216343  151.09  147.00                  -40.90\n",
      "\n",
      "\n",
      "report______> Distribution of short non-bonded contacts:\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "DISTANCE1:  0.00 2.10 2.20 2.30 2.40 2.50 2.60 2.70 2.80 2.90 3.00 3.10 3.20 3.30 3.40\n",
      "DISTANCE2:  2.10 2.20 2.30 2.40 2.50 2.60 2.70 2.80 2.90 3.00 3.10 3.20 3.30 3.40 3.50\n",
      "FREQUENCY:     0    0    0    0    4  134  190  783 1038 1695 1344 1686 2223 2549 2726\n",
      "\n",
      "\n",
      "<< end of ENERGY.\n",
      "openf___224_> Open           muscle_nAChR.B99990001.pdb\n",
      "wrpdb___568_> Residues, atoms, selected atoms:     2005    16360    16360\n",
      "\n",
      ">> Summary of successfully produced models:\n",
      "Filename                          molpdf\n",
      "----------------------------------------\n",
      "muscle_nAChR.B99990001.pdb    12277.39453\n",
      "\n"
     ]
    }
   ],
   "source": [
    "from modeller import *\n",
    "from modeller.automodel import *    # Load the automodel class\n",
    "\n",
    "log.verbose()\n",
    "env = environ()\n",
    "\n",
    "# directories for input atom files\n",
    "env.io.atom_files_directory = ['.']\n",
    "\n",
    "a = automodel(env,\n",
    "              alnfile  = 'pentamer_alignment.ali',     # alignment filename\n",
    "              knowns   = '6pv7_clean', # codes of the templates - ie your \"cleaned up\" pdb above\n",
    "              sequence = 'muscle_nAChR')              # code of the target\n",
    "\n",
    "a.starting_model= 1                 # index of the first model\n",
    "a.ending_model  = 1                # index of the last model\n",
    "                                    # (determines how many models to calculate)\n",
    "a.make()                            # do comparative modeling"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "- lets take a look - modify the holding code below as appropriate"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "import nglview\n",
    "view = nglview.show_structure_file('YOUR_TARGET_NAME.B99990001.pdb')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "view"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "view._display_image()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### 4. Modifying models to include features like disufide bonds"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "- The muscle nAChR, like many other surface proteins, contains disulfide bonds in its structure.  It is of course crucially important to check that these have been correctly formed after you generate your model.\n",
    "\n",
    "- Modeller can easily deal with this problem by effectively using a model of itself and correcting the problematic bonds.\n",
    "\n",
    "- Included in the directory 'disulfide' is a previously built homology model that does NOT contain proper disulfide bonds. \n",
    "\n",
    "- Open the model in pymol and locate the cysteine residues that should be bonded.\n",
    "\n",
    "    `select. br resname CYS` \n",
    "\n",
    "    into the pymol GUI to help find the correct residues.\n",
    "\n",
    "- As you did in previous sections above, extract the fast sequence of the protein.  You can then use these sequence as both the template and the sequence to be modelled - thus the alignment is super-trivial in this case!\n",
    "\n",
    "- Construct a alignment.ali file for this.\n",
    "\n",
    "- Adapt the modeller run below (you can keep the number of models generated to 1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Comparative modeling by the automodel class\n",
    "from modeller import *\t\t\t# Load standard Modeller classes\n",
    "from modeller.automodel import *\t# Load the automodel class\n",
    "\n",
    "# Modeller can automatically build in disulfides if they are aligned with template disu\n",
    "# Redefine the special_patches routine to include the additional disulfides\n",
    "# (this routine is empty by default):\n",
    "log.verbose()\n",
    "env = environ()\n",
    "env.io.atom_files_directory = ['.']\n",
    "\n",
    "class MyModel(automodel):\n",
    "    def special_patches(self, aln):\n",
    "        # A disulfide between residues 196 and 197:\n",
    "        self.patch(residue_type='DISU', residues=(self.residues['192:A'],\n",
    "                                                  self.residues['193:A']))\n",
    "\n",
    "a = MyModel(env, alnfile='self-on-self-align.ali',#alignment filename\n",
    "              knowns='muscle_nAChR', sequence='disulfide', \t#edit the knowns to reflect the structure of one you have made earlier\n",
    "              assess_methods=(assess.DOPE,\t\n",
    "                              assess.GA341))\n",
    "a.starting_model = 1\t#index of the first model\n",
    "a.ending_model = 1\t#index of the last model (determines how many models to calculate)\n",
    "a.make()\t\t#do comparative modeling"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### 5. Refining Loops"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "- Due to the uncertainty of how to model flexible loops, you may need to try a few different loop conformations before you obtain something that makes sense.\n",
    "\n",
    "- select your model of the pentameric receptor you worked on previously.\n",
    "\n",
    "- You do not need a .ali file here, just modify the holding modeller script below.\n",
    "\n",
    "- A modeller objective function will be displayed in the pdb files as before.\n",
    "\n",
    "- Open all these structures in pymol to see the different loop conformation outputs.  In this case we are refining the loop C region of one of the subunits."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "openf___224_> Open           $(LIB)/restyp.lib\n",
      "openf___224_> Open           ${MODINSTALL9v25}/modlib/resgrp.lib\n",
      "rdresgr_266_> Number of residue groups:        2\n",
      "openf___224_> Open           ${MODINSTALL9v25}/modlib/sstruc.lib\n",
      "\n",
      "Dynamically allocated memory at   amaxlibraries [B,KiB,MiB]:     33114711   32338.586    31.581\n",
      "\n",
      "Dynamically allocated memory at   amaxlibraries [B,KiB,MiB]:     33115239   32339.102    31.581\n",
      "openf___224_> Open           ${MODINSTALL9v25}/modlib/resdih.lib\n",
      "\n",
      "Dynamically allocated memory at   amaxlibraries [B,KiB,MiB]:     33163839   32386.562    31.628\n",
      "rdrdih__263_> Number of dihedral angle types         :        9\n",
      "              Maximal number of dihedral angle optima:        3\n",
      "              Dihedral angle names                   :  Alph Phi Psi Omeg chi1 chi2 chi3 chi4 chi5\n",
      "openf___224_> Open           ${MODINSTALL9v25}/modlib/radii.lib\n",
      "\n",
      "Dynamically allocated memory at   amaxlibraries [B,KiB,MiB]:     33177139   32399.551    31.640\n",
      "openf___224_> Open           ${MODINSTALL9v25}/modlib/radii14.lib\n",
      "openf___224_> Open           ${MODINSTALL9v25}/modlib/af_mnchdef.lib\n",
      "rdwilmo_274_> Mainchain residue conformation classes:  APBLE\n",
      "openf___224_> Open           ${MODINSTALL9v25}/modlib/mnch.lib\n",
      "rdclass_257_> Number of classes:        5\n",
      "openf___224_> Open           ${MODINSTALL9v25}/modlib/mnch1.lib\n",
      "openf___224_> Open           ${MODINSTALL9v25}/modlib/mnch2.lib\n",
      "openf___224_> Open           ${MODINSTALL9v25}/modlib/mnch3.lib\n",
      "openf___224_> Open           ${MODINSTALL9v25}/modlib/xs4.mat\n",
      "rdrrwgh_268_> Number of residue types:       21\n",
      "openf___224_> Open           ${LIB}/top_heav.lib\n",
      "read_to_681_> topology.submodel read from topology file:        3\n",
      "openf___224_> Open           ${LIB}/par.lib\n",
      "\n",
      "Dynamically allocated memory at  amaxparameters [B,KiB,MiB]:     33325736   32544.664    31.782\n",
      "\n",
      "Dynamically allocated memory at  amaxparameters [B,KiB,MiB]:     33327336   32546.227    31.783\n",
      "\n",
      "Dynamically allocated memory at  amaxparameters [B,KiB,MiB]:     33331316   32550.113    31.787\n",
      "\n",
      "Dynamically allocated memory at  amaxparameters [B,KiB,MiB]:     33333316   32552.066    31.789\n",
      "\n",
      "Dynamically allocated memory at  amaxparameters [B,KiB,MiB]:     33336316   32554.996    31.792\n",
      "\n",
      "Dynamically allocated memory at  amaxparameters [B,KiB,MiB]:     33340816   32559.391    31.796\n",
      "\n",
      "Dynamically allocated memory at  amaxparameters [B,KiB,MiB]:     33346388   32564.832    31.802\n",
      "\n",
      "Dynamically allocated memory at  amaxparameters [B,KiB,MiB]:     33349188   32567.566    31.804\n",
      "\n",
      "Dynamically allocated memory at  amaxparameters [B,KiB,MiB]:     33353388   32571.668    31.808\n",
      "\n",
      "Dynamically allocated memory at  amaxparameters [B,KiB,MiB]:     33359688   32577.820    31.814\n",
      "\n",
      "Dynamically allocated memory at  amaxparameters [B,KiB,MiB]:     33362460   32580.527    31.817\n",
      "\n",
      "Dynamically allocated memory at  amaxparameters [B,KiB,MiB]:     33363860   32581.895    31.818\n",
      "read_pa_232_> parameters    BONDS   ANGLS  DIHEDS IMPROPS    MODE\n",
      "                              227     561     661     112       0\n",
      "openf___224_> Open           $(LIB)/atmcls-melo.lib\n",
      "\n",
      "Dynamically allocated memory at amaxgroup_restr [B,KiB,MiB]:     33364294   32582.318    31.819\n",
      "\n",
      "Dynamically allocated memory at amaxgroup_restr [B,KiB,MiB]:     33364358   32582.381    31.819\n",
      "\n",
      "Dynamically allocated memory at amaxgroup_restr [B,KiB,MiB]:     33364422   32582.443    31.819\n",
      "\n",
      "Dynamically allocated memory at amaxgroup_restr [B,KiB,MiB]:     33364550   32582.568    31.819\n",
      "\n",
      "Dynamically allocated memory at amaxgroup_restr [B,KiB,MiB]:     33364742   32582.756    31.819\n",
      "\n",
      "Dynamically allocated memory at amaxgroup_restr [B,KiB,MiB]:     33364998   32583.006    31.819\n",
      "\n",
      "Dynamically allocated memory at amaxgroup_restr [B,KiB,MiB]:     33365382   32583.381    31.820\n",
      "\n",
      "Dynamically allocated memory at amaxgroup_restr [B,KiB,MiB]:     33365958   32583.943    31.820\n",
      "\n",
      "Dynamically allocated memory at amaxgroup_restr [B,KiB,MiB]:     33366008   32583.992    31.820\n",
      "\n",
      "Dynamically allocated memory at amaxgroup_restr [B,KiB,MiB]:     33366066   32584.049    31.820\n",
      "\n",
      "Dynamically allocated memory at amaxgroup_restr [B,KiB,MiB]:     33366962   32584.924    31.821\n",
      "\n",
      "Dynamically allocated memory at amaxgroup_restr [B,KiB,MiB]:     33368306   32586.236    31.822\n",
      "\n",
      "Dynamically allocated memory at amaxgroup_restr [B,KiB,MiB]:     33368372   32586.301    31.823\n",
      "\n",
      "Dynamically allocated memory at amaxgroup_restr [B,KiB,MiB]:     33370356   32588.238    31.824\n",
      "\n",
      "Dynamically allocated memory at amaxgroup_restr [B,KiB,MiB]:     33370512   32588.391    31.825\n",
      "\n",
      "Dynamically allocated memory at amaxgroup_restr [B,KiB,MiB]:     33373520   32591.328    31.827\n",
      "\n",
      "Dynamically allocated memory at amaxgroup_restr [B,KiB,MiB]:     33373814   32591.615    31.828\n",
      "\n",
      "Dynamically allocated memory at amaxgroup_restr [B,KiB,MiB]:     33374318   32592.107    31.828\n",
      "\n",
      "Dynamically allocated memory at amaxgroup_restr [B,KiB,MiB]:     33375314   32593.080    31.829\n",
      "\n",
      "Dynamically allocated memory at amaxgroup_restr [B,KiB,MiB]:     33379794   32597.455    31.833\n",
      "\n",
      "Dynamically allocated memory at amaxgroup_restr [B,KiB,MiB]:     33381828   32599.441    31.835\n",
      "\n",
      "Dynamically allocated memory at amaxgroup_restr [B,KiB,MiB]:     33386280   32603.789    31.840\n",
      "openf___224_> Open           $(LIB)/melo1-dist.lib\n",
      "\n",
      "Dynamically allocated memory at amaxgroup_restr [B,KiB,MiB]:     33386676   32604.176    31.840\n",
      "\n",
      "Dynamically allocated memory at amaxgroup_restr [B,KiB,MiB]:     33387472   32604.953    31.841\n",
      "\n",
      "Dynamically allocated memory at amaxgroup_restr [B,KiB,MiB]:     33388268   32605.730    31.842\n",
      "\n",
      "Dynamically allocated memory at amaxgroup_restr [B,KiB,MiB]:     33389456   32606.891    31.843\n",
      "\n",
      "Dynamically allocated memory at amaxgroup_restr [B,KiB,MiB]:     33389856   32607.281    31.843\n",
      "\n",
      "Dynamically allocated memory at amaxgroup_restr [B,KiB,MiB]:     33390456   32607.867    31.844\n",
      "\n",
      "Dynamically allocated memory at amaxgroup_restr [B,KiB,MiB]:     33391356   32608.746    31.844\n",
      "\n",
      "Dynamically allocated memory at amaxgroup_restr [B,KiB,MiB]:     33392704   32610.062    31.846\n",
      "\n",
      "Dynamically allocated memory at amaxgroup_restr [B,KiB,MiB]:     33394728   32612.039    31.848\n",
      "\n",
      "Dynamically allocated memory at amaxgroup_restr [B,KiB,MiB]:     33394928   32612.234    31.848\n",
      "\n",
      "Dynamically allocated memory at amaxgroup_restr [B,KiB,MiB]:     33397964   32615.199    31.851\n",
      "\n",
      "Dynamically allocated memory at amaxgroup_restr [B,KiB,MiB]:     33398264   32615.492    31.851\n",
      "\n",
      "Dynamically allocated memory at amaxgroup_restr [B,KiB,MiB]:     33402816   32619.938    31.855\n",
      "\n",
      "Dynamically allocated memory at amaxgroup_restr [B,KiB,MiB]:     33403264   32620.375    31.856\n",
      "\n",
      "Dynamically allocated memory at amaxgroup_restr [B,KiB,MiB]:     33410092   32627.043    31.862\n",
      "\n",
      "Dynamically allocated memory at amaxgroup_restr [B,KiB,MiB]:     33410764   32627.699    31.863\n",
      "\n",
      "Dynamically allocated memory at amaxgroup_restr [B,KiB,MiB]:     33421008   32637.703    31.873\n",
      "\n",
      "Dynamically allocated memory at amaxgroup_restr [B,KiB,MiB]:     33422016   32638.688    31.874\n",
      "\n",
      "Dynamically allocated memory at amaxgroup_restr [B,KiB,MiB]:     33422616   32639.273    31.874\n",
      "\n",
      "Dynamically allocated memory at amaxgroup_restr [B,KiB,MiB]:     33423016   32639.664    31.875\n",
      "\n",
      "Dynamically allocated memory at amaxgroup_restr [B,KiB,MiB]:     33438380   32654.668    31.889\n",
      "\n",
      "Dynamically allocated memory at amaxgroup_restr [B,KiB,MiB]:     33439892   32656.145    31.891\n",
      "\n",
      "Dynamically allocated memory at amaxgroup_restr [B,KiB,MiB]:     33440792   32657.023    31.892\n",
      "\n",
      "Dynamically allocated memory at amaxgroup_restr [B,KiB,MiB]:     33441392   32657.609    31.892\n",
      "\n",
      "Dynamically allocated memory at amaxgroup_restr [B,KiB,MiB]:     33464440   32680.117    31.914\n",
      "\n",
      "Dynamically allocated memory at amaxgroup_restr [B,KiB,MiB]:     33466708   32682.332    31.916\n",
      "\n",
      "Dynamically allocated memory at amaxgroup_restr [B,KiB,MiB]:     33468052   32683.645    31.918\n",
      "\n",
      "Dynamically allocated memory at amaxgroup_restr [B,KiB,MiB]:     33468952   32684.523    31.918\n",
      "\n",
      "Dynamically allocated memory at amaxgroup_restr [B,KiB,MiB]:     33503524   32718.285    31.951\n",
      "\n",
      "Dynamically allocated memory at amaxgroup_restr [B,KiB,MiB]:     33506928   32721.609    31.955\n",
      "\n",
      "Dynamically allocated memory at amaxgroup_restr [B,KiB,MiB]:     33508944   32723.578    31.957\n",
      "\n",
      "Dynamically allocated memory at amaxgroup_restr [B,KiB,MiB]:     33510292   32724.895    31.958\n",
      "\n",
      "Dynamically allocated memory at amaxgroup_restr [B,KiB,MiB]:     33562148   32775.535    32.007\n",
      "\n",
      "Dynamically allocated memory at amaxgroup_restr [B,KiB,MiB]:     33567252   32780.520    32.012\n",
      "\n",
      "Dynamically allocated memory at amaxgroup_restr [B,KiB,MiB]:     33570276   32783.473    32.015\n",
      "\n",
      "Dynamically allocated memory at amaxgroup_restr [B,KiB,MiB]:     33572300   32785.449    32.017\n",
      "\n",
      "Dynamically allocated memory at amaxgroup_restr [B,KiB,MiB]:     33650084   32861.410    32.091\n",
      "\n",
      "Dynamically allocated memory at amaxgroup_restr [B,KiB,MiB]:     33657740   32868.887    32.099\n",
      "\n",
      "Dynamically allocated memory at amaxgroup_restr [B,KiB,MiB]:     33662276   32873.316    32.103\n",
      "\n",
      "Dynamically allocated memory at amaxgroup_restr [B,KiB,MiB]:     33665312   32876.281    32.106\n",
      "\n",
      "Dynamically allocated memory at amaxgroup_restr [B,KiB,MiB]:     33781988   32990.223    32.217\n",
      "read_pa_232_> parameters    BONDS   ANGLS  DIHEDS IMPROPS    MODE\n",
      "                                0       0       0       0     820\n",
      "openf___224_> Open           muscle_nachr.pdb\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:     33783654   32991.852    32.219\n",
      "\n",
      "Dynamically allocated memory at    amaxsequence [B,KiB,MiB]:     33783730   32991.922    32.219\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:     33783901   32992.090    32.219\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:     33784751   32992.922    32.220\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:     33786009   32994.148    32.221\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:     33787913   32996.008    32.223\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:     33790769   32998.797    32.225\n",
      "\n",
      "Dynamically allocated memory at    amaxsequence [B,KiB,MiB]:     33790809   32998.836    32.225\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:     33790899   32998.926    32.226\n",
      "\n",
      "Dynamically allocated memory at    amaxsequence [B,KiB,MiB]:     33790959   32998.984    32.226\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:     33791094   32999.117    32.226\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:     33795378   33003.297    32.230\n",
      "\n",
      "Dynamically allocated memory at    amaxsequence [B,KiB,MiB]:     33795466   33003.383    32.230\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:     33795664   33003.578    32.230\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:     33802090   33009.852    32.236\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:     33811712   33019.250    32.245\n",
      "\n",
      "Dynamically allocated memory at    amaxsequence [B,KiB,MiB]:     33811844   33019.379    32.245\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:     33812141   33019.668    32.246\n",
      "\n",
      "Dynamically allocated memory at    amaxsequence [B,KiB,MiB]:     33812341   33019.863    32.246\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:     33812791   33020.305    32.246\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:     33827241   33034.414    32.260\n",
      "\n",
      "Dynamically allocated memory at    amaxsequence [B,KiB,MiB]:     33827541   33034.707    32.260\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:     33828216   33035.367    32.261\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:     33849874   33056.516    32.282\n",
      "\n",
      "Dynamically allocated memory at    amaxsequence [B,KiB,MiB]:     33850322   33056.953    32.282\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:     33851330   33057.938    32.283\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:     33883834   33089.680    32.314\n",
      "\n",
      "Dynamically allocated memory at    amaxsequence [B,KiB,MiB]:     33884506   33090.336    32.315\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:     33886018   33091.812    32.316\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:     33934774   33139.430    32.363\n",
      "\n",
      "Dynamically allocated memory at    amaxsequence [B,KiB,MiB]:     33934858   33139.508    32.363\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:     33934858   33139.508    32.363\n",
      "\n",
      "Dynamically allocated memory at    amaxsequence [B,KiB,MiB]:     33935866   33140.492    32.364\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:     33938134   33142.711    32.366\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:     34011268   33214.129    32.436\n",
      "\n",
      "Dynamically allocated memory at    amaxsequence [B,KiB,MiB]:     34012780   33215.605    32.437\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:     34016182   33218.930    32.440\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:     34125866   33326.039    32.545\n",
      "\n",
      "Dynamically allocated memory at    amaxsequence [B,KiB,MiB]:     34128134   33328.258    32.547\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:     34133237   33333.238    32.552\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:     34297763   33493.910    32.709\n",
      "\n",
      "Dynamically allocated memory at    amaxsequence [B,KiB,MiB]:     34297819   33493.965    32.709\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:     34297819   33493.965    32.709\n",
      "\n",
      "Dynamically allocated memory at    amaxsequence [B,KiB,MiB]:     34301223   33497.289    32.712\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:     34308882   33504.766    32.719\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:     34555688   33745.789    32.955\n",
      "\n",
      "Dynamically allocated memory at       amaxmodel [B,KiB,MiB]:     36316840   35465.664    34.634\n",
      "read_mo_297_> Segments, residues, atoms:        5     1935    15758\n",
      "read_mo_298_> Segment:        1     1 A   469 A     3167\n",
      "read_mo_298_> Segment:        2     1 B   459 B     3149\n",
      "read_mo_298_> Segment:        3     4 C   459 C     3126\n",
      "read_mo_298_> Segment:        4     1 D   469 D     3167\n",
      "read_mo_298_> Segment:        5     1 E   459 E     3149\n",
      "\n",
      "Dynamically allocated memory at   amaxalignment [B,KiB,MiB]:     36373773   35521.262    34.689\n",
      "\n",
      "Dynamically allocated memory at   amaxstructure [B,KiB,MiB]:     37515324   36636.059    35.777\n",
      "\n",
      "Dynamically allocated memory at   amaxalignment [B,KiB,MiB]:     37533181   36653.496    35.794\n",
      "\n",
      "Dynamically allocated memory at   amaxstructure [B,KiB,MiB]:     38674732   37768.293    36.883\n",
      "\n",
      "Dynamically allocated memory at       amaxmodel [B,KiB,MiB]:     38723396   37815.816    36.930\n",
      "\n",
      "mkilst______> segment topology constructed from sequence and RTF:\n",
      "              segments residues  atoms  bonds angles dihedrals impropers:\n",
      "                     1      387   3167   3253      0         0      1415\n",
      "\n",
      "Dynamically allocated memory at       amaxmodel [B,KiB,MiB]:     38861840   37951.016    37.062\n",
      "\n",
      "patch_______> segment topology patched using RTF:     1  ;  SER  ;  NTER\n",
      "              segments residues  atoms  bonds angles dihedrals impropers:\n",
      "                     1      387   3167   3253   4434      5329      1415\n",
      "\n",
      "\n",
      "Dynamically allocated memory at       amaxmodel [B,KiB,MiB]:     38994416   38080.484    37.188\n",
      "patch_______> segment topology patched using RTF:   387  ;  MET  ;  CTER\n",
      "              segments residues  atoms  bonds angles dihedrals impropers:\n",
      "                     1      387   3168   3254   4436      5331      1416\n",
      "\n",
      "genseg______> segment topology constructed from sequence and RTF:\n",
      "              segments residues  atoms  bonds angles dihedrals impropers:\n",
      "                     1      387   3168   3254   4436      5331      1416\n",
      "\n",
      "Dynamically allocated memory at       amaxmodel [B,KiB,MiB]:     39042352   38127.297    37.234\n",
      "\n",
      "mkilst______> segment topology constructed from sequence and RTF:\n",
      "              segments residues  atoms  bonds angles dihedrals impropers:\n",
      "                     2      775   6317   6483   4436      5331      2796\n",
      "\n",
      "Dynamically allocated memory at       amaxmodel [B,KiB,MiB]:     39179532   38261.262    37.365\n",
      "\n",
      "patch_______> segment topology patched using RTF:   388  ;  ARG  ;  NTER\n",
      "              segments residues  atoms  bonds angles dihedrals impropers:\n",
      "                     2      775   6317   6483   8841     10601      2796\n",
      "\n",
      "\n",
      "Dynamically allocated memory at       amaxmodel [B,KiB,MiB]:     39321788   38400.184    37.500\n",
      "patch_______> segment topology patched using RTF:   775  ;  PRO  ;  CTER\n",
      "              segments residues  atoms  bonds angles dihedrals impropers:\n",
      "                     2      775   6318   6484   8843     10603      2797\n",
      "\n",
      "genseg______> segment topology constructed from sequence and RTF:\n",
      "              segments residues  atoms  bonds angles dihedrals impropers:\n",
      "                     2      775   6318   6484   8843     10603      2797\n",
      "\n",
      "Dynamically allocated memory at       amaxmodel [B,KiB,MiB]:     39369412   38446.691    37.546\n",
      "\n",
      "mkilst______> segment topology constructed from sequence and RTF:\n",
      "              segments residues  atoms  bonds angles dihedrals impropers:\n",
      "                     3     1160   9444   9690   8843     10603      4169\n",
      "\n",
      "Dynamically allocated memory at       amaxmodel [B,KiB,MiB]:     39505660   38579.746    37.676\n",
      "\n",
      "patch_______> segment topology patched using RTF:   776  ;  ASN  ;  NTER\n",
      "              segments residues  atoms  bonds angles dihedrals impropers:\n",
      "                     3     1160   9444   9690  13217     15838      4169\n",
      "\n",
      "\n",
      "Dynamically allocated memory at       amaxmodel [B,KiB,MiB]:     39626556   38697.809    37.791\n",
      "patch_______> segment topology patched using RTF:  1160  ;  PRO  ;  CTER\n",
      "              segments residues  atoms  bonds angles dihedrals impropers:\n",
      "                     3     1160   9445   9691  13219     15840      4170\n",
      "\n",
      "genseg______> segment topology constructed from sequence and RTF:\n",
      "              segments residues  atoms  bonds angles dihedrals impropers:\n",
      "                     3     1160   9445   9691  13219     15840      4170\n",
      "\n",
      "Dynamically allocated memory at       amaxmodel [B,KiB,MiB]:     39675244   38745.355    37.837\n",
      "\n",
      "mkilst______> segment topology constructed from sequence and RTF:\n",
      "              segments residues  atoms  bonds angles dihedrals impropers:\n",
      "                     4     1547  12612  12944  13219     15840      5585\n",
      "\n",
      "Dynamically allocated memory at       amaxmodel [B,KiB,MiB]:     39813716   38880.582    37.969\n",
      "\n",
      "patch_______> segment topology patched using RTF:  1161  ;  SER  ;  NTER\n",
      "              segments residues  atoms  bonds angles dihedrals impropers:\n",
      "                     4     1547  12612  12944  17653     21169      5585\n",
      "\n",
      "\n",
      "Dynamically allocated memory at       amaxmodel [B,KiB,MiB]:     39987772   39050.559    38.135\n",
      "patch_______> segment topology patched using RTF:  1547  ;  MET  ;  CTER\n",
      "              segments residues  atoms  bonds angles dihedrals impropers:\n",
      "                     4     1547  12613  12945  17655     21171      5586\n",
      "\n",
      "genseg______> segment topology constructed from sequence and RTF:\n",
      "              segments residues  atoms  bonds angles dihedrals impropers:\n",
      "                     4     1547  12613  12945  17655     21171      5586\n",
      "\n",
      "Dynamically allocated memory at       amaxmodel [B,KiB,MiB]:     40035708   39097.371    38.181\n",
      "\n",
      "mkilst______> segment topology constructed from sequence and RTF:\n",
      "              segments residues  atoms  bonds angles dihedrals impropers:\n",
      "                     5     1935  15762  16174  17655     21171      6966\n",
      "\n",
      "Dynamically allocated memory at       amaxmodel [B,KiB,MiB]:     40172888   39231.336    38.312\n",
      "\n",
      "patch_______> segment topology patched using RTF:  1548  ;  ARG  ;  NTER\n",
      "              segments residues  atoms  bonds angles dihedrals impropers:\n",
      "                     5     1935  15762  16174  22060     26441      6966\n",
      "\n",
      "\n",
      "Dynamically allocated memory at       amaxmodel [B,KiB,MiB]:     40286824   39342.602    38.421\n",
      "patch_______> segment topology patched using RTF:  1935  ;  PRO  ;  CTER\n",
      "              segments residues  atoms  bonds angles dihedrals impropers:\n",
      "                     5     1935  15763  16175  22062     26443      6967\n",
      "\n",
      "genseg______> segment topology constructed from sequence and RTF:\n",
      "              segments residues  atoms  bonds angles dihedrals impropers:\n",
      "                     5     1935  15763  16175  22062     26443      6967\n",
      "openf___224_> Open           muscle_nachr.pdb\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:     40297835   39353.355    38.431\n",
      "\n",
      "Dynamically allocated memory at    amaxsequence [B,KiB,MiB]:     40297911   39353.430    38.431\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:     40298082   39353.594    38.431\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:     40298932   39354.426    38.432\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:     40300190   39355.656    38.433\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:     40302094   39357.516    38.435\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:     40304950   39360.305    38.438\n",
      "\n",
      "Dynamically allocated memory at    amaxsequence [B,KiB,MiB]:     40304990   39360.344    38.438\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:     40305080   39360.430    38.438\n",
      "\n",
      "Dynamically allocated memory at    amaxsequence [B,KiB,MiB]:     40305140   39360.488    38.438\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:     40305275   39360.621    38.438\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:     40309559   39364.805    38.442\n",
      "\n",
      "Dynamically allocated memory at    amaxsequence [B,KiB,MiB]:     40309647   39364.891    38.442\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:     40309845   39365.082    38.442\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:     40316271   39371.359    38.449\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:     40325893   39380.754    38.458\n",
      "\n",
      "Dynamically allocated memory at    amaxsequence [B,KiB,MiB]:     40326025   39380.883    38.458\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:     40326322   39381.172    38.458\n",
      "\n",
      "Dynamically allocated memory at    amaxsequence [B,KiB,MiB]:     40326522   39381.367    38.458\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:     40326972   39381.809    38.459\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:     40341422   39395.922    38.473\n",
      "\n",
      "Dynamically allocated memory at    amaxsequence [B,KiB,MiB]:     40341722   39396.211    38.473\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:     40342397   39396.871    38.474\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:     40364055   39418.023    38.494\n",
      "\n",
      "Dynamically allocated memory at    amaxsequence [B,KiB,MiB]:     40364503   39418.461    38.495\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:     40365511   39419.445    38.496\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:     40398015   39451.188    38.527\n",
      "\n",
      "Dynamically allocated memory at    amaxsequence [B,KiB,MiB]:     40398687   39451.844    38.527\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:     40400199   39453.320    38.529\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:     40448955   39500.934    38.575\n",
      "\n",
      "Dynamically allocated memory at    amaxsequence [B,KiB,MiB]:     40449039   39501.016    38.575\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:     40449039   39501.016    38.575\n",
      "\n",
      "Dynamically allocated memory at    amaxsequence [B,KiB,MiB]:     40450047   39502.000    38.576\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:     40452315   39504.215    38.578\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:     40525449   39575.633    38.648\n",
      "\n",
      "Dynamically allocated memory at    amaxsequence [B,KiB,MiB]:     40526961   39577.109    38.650\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:     40530363   39580.434    38.653\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:     40640047   39687.547    38.757\n",
      "\n",
      "Dynamically allocated memory at    amaxsequence [B,KiB,MiB]:     40642315   39689.762    38.760\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:     40647418   39694.742    38.764\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:     40811944   39855.414    38.921\n",
      "\n",
      "Dynamically allocated memory at    amaxsequence [B,KiB,MiB]:     40812000   39855.469    38.921\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:     40812000   39855.469    38.921\n",
      "\n",
      "Dynamically allocated memory at    amaxsequence [B,KiB,MiB]:     40815404   39858.793    38.925\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:     40823063   39866.273    38.932\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:     41069869   40107.293    39.167\n",
      "\n",
      "Dynamically allocated memory at       amaxmodel [B,KiB,MiB]:     42831021   41827.168    40.847\n",
      "read_mo_297_> Segments, residues, atoms:        5     1935    15758\n",
      "read_mo_298_> Segment:        1     1 A   469 A     3167\n",
      "read_mo_298_> Segment:        2     1 B   459 B     3149\n",
      "read_mo_298_> Segment:        3     4 C   459 C     3126\n",
      "read_mo_298_> Segment:        4     1 D   469 D     3167\n",
      "read_mo_298_> Segment:        5     1 E   459 E     3149\n",
      "  #      ALGNMT CODE\n",
      "\n",
      "  1  muscle\n",
      "  2  muscle\n",
      "transfe_506_> MODEL is an average of all templates.\n",
      "transfe_511_> Number of templates for coordinate transfer:        1\n",
      "              After transferring coordinates of the equivalent template atoms,\n",
      "              there are defined, undefined atoms in MODEL:    15758        5\n",
      "86 atoms selected for loop refinement\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "The following 5 atoms were not found in the input model's non-loop region,\n",
      "and were added automatically by Modeller in order to determine correct\n",
      "interactions between the loop and the rest of the protein:\n",
      "<Atom OXT:469:A>, <Atom OXT:459:B>, <Atom OXT:459:C>, <Atom OXT:469:D>, <Atom OXT:459:E>\n",
      "The coordinates for these atoms will be constructed automatically using\n",
      "internal coordinates, and not relaxed, so clashes between these atoms and the\n",
      "rest of the protein may exist (note, however, that the score of the loop does\n",
      "not include protein-protein internal interactions, so will not be adversely\n",
      "affected by any clashes).\n",
      "\n",
      "Output models will contain both the atoms present in the input model and the\n",
      "newly-added atoms. If you want them to only contain the atoms from the input\n",
      "model, set loopmodel.loop.write_defined_only=True.\n",
      "\n",
      "make_re_417_> Restraint type to be calculated:  bond\n",
      "\n",
      "Dynamically allocated memory at  amaxrestraints [B,KiB,MiB]:     40393292   39446.574    38.522\n",
      "r_stere_606_> Stereochemical restraints were constructed from RTF & PRMF.\n",
      "              Added bond,angle,dihedral,improper restraints  :       90       0       0       0\n",
      "              Total number of restraints before, now         :        0       90\n",
      "make_re_422_> Number of previous, current restraints         :        0       90\n",
      "make_re_423_> Number of previous, current selected restraints:        0       90\n",
      "make_re_417_> Restraint type to be calculated:  angle\n",
      "r_stere_606_> Stereochemical restraints were constructed from RTF & PRMF.\n",
      "              Added bond,angle,dihedral,improper restraints  :        0     128       0       0\n",
      "              Total number of restraints before, now         :       90      218\n",
      "make_re_422_> Number of previous, current restraints         :       90      218\n",
      "make_re_423_> Number of previous, current selected restraints:       90      218\n",
      "make_re_417_> Restraint type to be calculated:  improper\n",
      "r_stere_606_> Stereochemical restraints were constructed from RTF & PRMF.\n",
      "              Added bond,angle,dihedral,improper restraints  :        0       0       0      40\n",
      "              Total number of restraints before, now         :      218      258\n",
      "make_re_422_> Number of previous, current restraints         :      218      258\n",
      "make_re_423_> Number of previous, current selected restraints:      218      258\n",
      "make_re_417_> Restraint type to be calculated:  dihedral\n",
      "\n",
      "Dynamically allocated memory at  amaxrestraints [B,KiB,MiB]:     40401484   39454.574    38.530\n",
      "r_stere_606_> Stereochemical restraints were constructed from RTF & PRMF.\n",
      "              Added bond,angle,dihedral,improper restraints  :        0       0     162       0\n",
      "              Total number of restraints before, now         :      258      420\n",
      "make_re_422_> Number of previous, current restraints         :      258      420\n",
      "make_re_423_> Number of previous, current selected restraints:      258      420\n",
      "make_re_417_> Restraint type to be calculated:  phi-psi_binormal\n",
      "\n",
      "Dynamically allocated memory at   amaxstructure [B,KiB,MiB]:     40401654   39454.742    38.530\n",
      "openf___224_> Open           ${MODINSTALL9v25}/modlib/mnch1.bin\n",
      "openf___224_> Open           ${MODINSTALL9v25}/modlib/mnch1.mdt\n",
      "getdata_643_> Protein accepted:  muscle\n",
      "getdata_289_> Proteins (all/accepted):        1        1\n",
      "make_re_422_> Number of previous, current restraints         :      420      433\n",
      "make_re_423_> Number of previous, current selected restraints:      420      433\n",
      "make_re_417_> Restraint type to be calculated:  omega_dihedral\n",
      "\n",
      "Dynamically allocated memory at   amaxstructure [B,KiB,MiB]:     40401670   39454.758    38.530\n",
      "openf___224_> Open           ${MODINSTALL9v25}/modlib/omega.bin\n",
      "openf___224_> Open           ${MODINSTALL9v25}/modlib/omega.mdt\n",
      "getdata_643_> Protein accepted:  muscle\n",
      "getdata_289_> Proteins (all/accepted):        1        1\n",
      "omgdel__425_> Unselected all O C +N +CA dihedrals:       14\n",
      "              (This is to avoid clashes between STEREO\n",
      "              and OMEGA_DIHEDRAL restraints)\n",
      "make_re_422_> Number of previous, current restraints         :      433      445\n",
      "make_re_423_> Number of previous, current selected restraints:      433      431\n",
      "make_re_417_> Restraint type to be calculated:  chi1_dihedral\n",
      "\n",
      "Dynamically allocated memory at   amaxstructure [B,KiB,MiB]:     40401670   39454.758    38.530\n",
      "openf___224_> Open           ${MODINSTALL9v25}/modlib/chi1234.bin\n",
      "openf___224_> Open           ${MODINSTALL9v25}/modlib/chi1.mdt\n",
      "getdata_643_> Protein accepted:  muscle\n",
      "getdata_289_> Proteins (all/accepted):        1        1\n",
      "make_re_422_> Number of previous, current restraints         :      445      456\n",
      "make_re_423_> Number of previous, current selected restraints:      431      442\n",
      "make_re_417_> Restraint type to be calculated:  chi2_dihedral\n",
      "\n",
      "Dynamically allocated memory at   amaxstructure [B,KiB,MiB]:     40401670   39454.758    38.530\n",
      "openf___224_> Open           ${MODINSTALL9v25}/modlib/chi1234.bin\n",
      "openf___224_> Open           ${MODINSTALL9v25}/modlib/chi2.mdt\n",
      "getdata_643_> Protein accepted:  muscle\n",
      "getdata_289_> Proteins (all/accepted):        1        1\n",
      "make_re_422_> Number of previous, current restraints         :      456      463\n",
      "make_re_423_> Number of previous, current selected restraints:      442      449\n",
      "make_re_417_> Restraint type to be calculated:  chi3_dihedral\n",
      "\n",
      "Dynamically allocated memory at   amaxstructure [B,KiB,MiB]:     40401670   39454.758    38.530\n",
      "openf___224_> Open           ${MODINSTALL9v25}/modlib/chi1234.bin\n",
      "openf___224_> Open           ${MODINSTALL9v25}/modlib/chi3.mdt\n",
      "getdata_643_> Protein accepted:  muscle\n",
      "getdata_289_> Proteins (all/accepted):        1        1\n",
      "make_re_422_> Number of previous, current restraints         :      463      466\n",
      "make_re_423_> Number of previous, current selected restraints:      449      452\n",
      "make_re_417_> Restraint type to be calculated:  chi4_dihedral\n",
      "\n",
      "Dynamically allocated memory at   amaxstructure [B,KiB,MiB]:     40401670   39454.758    38.530\n",
      "openf___224_> Open           ${MODINSTALL9v25}/modlib/chi1234.bin\n",
      "openf___224_> Open           ${MODINSTALL9v25}/modlib/chi4.mdt\n",
      "mdtrsr__446W> A potential that relies on one protein is used, yet you have at\n",
      "              least one known structure available. MDT, not library, potential is used.\n",
      "getdata_643_> Protein accepted:  muscle\n",
      "getdata_289_> Proteins (all/accepted):        1        1\n",
      "make_re_422_> Number of previous, current restraints         :      466      466\n",
      "make_re_423_> Number of previous, current selected restraints:      452      452\n",
      "pick_re_612_> Number of MODEL atoms, selected restraints;       86      466\n",
      "rmdupl__427_>       81 redundant cosine dihedral restraints were unselected.\n",
      "condens_443_> Restraints marked for deletion were removed.\n",
      "              Total number of restraints before, now:      466      385\n",
      "openf___224_> Open           muscle.lrsr\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "\n",
      "Dynamically allocated memory at amaxhash_contac [B,KiB,MiB]:     44941378   43888.062    42.859\n",
      "\n",
      "Dynamically allocated memory at  amaxrestraints [B,KiB,MiB]:     45612294   44543.258    43.499\n",
      "\n",
      "\n",
      ">> ENERGY; Differences between the model's features and restraints:\n",
      "Number of all residues in MODEL                   :     1935\n",
      "Number of all, selected real atoms                :    15763      86\n",
      "Number of all, selected pseudo atoms              :        0       0\n",
      "Number of all static, selected restraints         :      385     385\n",
      "COVALENT_CYS                                      :        F\n",
      "NONBONDED_SEL_ATOMS                               :        1\n",
      "Number of non-bonded pairs (excluding 1-2,1-3,1-4):     2340\n",
      "Dynamic pairs routine                             : 5, NATM_SEL x NATM double loop\n",
      "Atomic shift for contacts update (UPDATE_DYNAMIC) :    0.390\n",
      "LENNARD_JONES_SWITCH                              :    6.500   7.500\n",
      "COULOMB_JONES_SWITCH                              :    6.500   7.500\n",
      "RESIDUE_SPAN_RANGE                                :        0   99999\n",
      "NLOGN_USE                                         :       15\n",
      "CONTACT_SHELL                                     :    7.000\n",
      "DYNAMIC_PAIRS,_SPHERE,_COULOMB,_LENNARD,_MODELLER :        T       T       F       F       T\n",
      "SPHERE_STDV                                       :    0.050\n",
      "RADII_FACTOR                                      :    0.820\n",
      "Current energy                                    :         138.2753\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Summary of the restraint violations: \n",
      "\n",
      "   NUM     ... number of restraints.\n",
      "   NUMVI   ... number of restraints with RVIOL > VIOL_REPORT_CUT[i].\n",
      "   RVIOL   ... relative difference from the best value.\n",
      "   NUMVP   ... number of restraints with -Ln(pdf) > VIOL_REPORT_CUT2[i].\n",
      "   RMS_1   ... RMS(feature, minimally_violated_basis_restraint, NUMB).\n",
      "   RMS_2   ... RMS(feature, best_value, NUMB).\n",
      "   MOL.PDF ... scaled contribution to -Ln(Molecular pdf).\n",
      "\n",
      " #                     RESTRAINT_GROUP      NUM   NUMVI  NUMVP   RMS_1   RMS_2         MOL.PDF     S_i\n",
      "------------------------------------------------------------------------------------------------------\n",
      " 1 Bond length potential              :      90       0      0   0.023   0.023      13.366       1.000\n",
      " 2 Bond angle potential               :     128       0      2   3.047   3.047      21.218       1.000\n",
      " 3 Stereochemical cosine torsion poten:      81       0     10  58.540  58.540      41.475       1.000\n",
      " 4 Stereochemical improper torsion pot:      40       0      0   1.365   1.365      1.3640       1.000\n",
      " 5 Soft-sphere overlap restraints     :    2340       0      0   0.000   0.000     0.51052E-02   1.000\n",
      " 6 Lennard-Jones 6-12 potential       :       0       0      0   0.000   0.000      0.0000       1.000\n",
      " 7 Coulomb point-point electrostatic p:       0       0      0   0.000   0.000      0.0000       1.000\n",
      " 8 H-bonding potential                :       0       0      0   0.000   0.000      0.0000       1.000\n",
      " 9 Distance restraints 1 (CA-CA)      :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "10 Distance restraints 2 (N-O)        :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "11 Mainchain Phi dihedral restraints  :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "12 Mainchain Psi dihedral restraints  :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "13 Mainchain Omega dihedral restraints:      12       0      2   8.563   8.563      10.386       1.000\n",
      "14 Sidechain Chi_1 dihedral restraints:      11       0      0  73.170  73.170      2.4478       1.000\n",
      "15 Sidechain Chi_2 dihedral restraints:       7       0      0  27.348  27.348      1.6174       1.000\n",
      "16 Sidechain Chi_3 dihedral restraints:       3       0      0  37.454  37.454     0.61724       1.000\n",
      "17 Sidechain Chi_4 dihedral restraints:       0       0      0   0.000   0.000      0.0000       1.000\n",
      "18 Disulfide distance restraints      :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "19 Disulfide angle restraints         :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "20 Disulfide dihedral angle restraints:       0       0      0   0.000   0.000      0.0000       1.000\n",
      "21 Lower bound distance restraints    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "22 Upper bound distance restraints    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "23 Distance restraints 3 (SDCH-MNCH)  :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "24 Sidechain Chi_5 dihedral restraints:       0       0      0   0.000   0.000      0.0000       1.000\n",
      "25 Phi/Psi pair of dihedral restraints:      13       8      5  68.295 133.800      48.103       1.000\n",
      "26 Distance restraints 4 (SDCH-SDCH)  :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "27 Distance restraints 5 (X-Y)        :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "28 NMR distance restraints 6 (X-Y)    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "29 NMR distance restraints 7 (X-Y)    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "30 Minimal distance restraints        :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "31 Non-bonded restraints              :    2340       0      2   1.890   1.890     -2.3239       1.000\n",
      "32 Atomic accessibility restraints    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "33 Atomic density restraints          :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "34 Absolute position restraints       :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "35 Dihedral angle difference restraint:       0       0      0   0.000   0.000      0.0000       1.000\n",
      "36 GBSA implicit solvent potential    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "37 EM density fitting potential       :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "38 SAXS restraints                    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "39 Symmetry restraints                :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "\n",
      "\n",
      "\n",
      "List of the violated restraints:\n",
      "   A restraint is violated when the relative difference\n",
      "   from the best value (RVIOL) is larger than CUTOFF.\n",
      "\n",
      "   ICSR   ... index of a restraint in the current set.\n",
      "   RESNO  ... residue numbers of the first two atoms.\n",
      "   ATM    ... IUPAC atom names of the first two atoms.\n",
      "   FEAT   ... the value of the feature in the model.\n",
      "   restr  ... the mean of the basis restraint with the smallest\n",
      "              difference from the model (local minimum).\n",
      "   viol   ... difference from the local minimum.\n",
      "   rviol  ... relative difference from the local minimum.\n",
      "   RESTR  ... the best value (global minimum).\n",
      "   VIOL   ... difference from the best value.\n",
      "   RVIOL  ... relative difference from the best value.\n",
      "\n",
      "\n",
      "-------------------------------------------------------------------------------------------------\n",
      "\n",
      "Feature 25                           : Phi/Psi pair of dihedral restraints     \n",
      "List of the RVIOL violations larger than   :       6.5000\n",
      "\n",
      "    #   ICSR  RESNO1/2 ATM1/2   INDATM1/2    FEAT   restr    viol   rviol   RESTR    VIOL   RVIOL\n",
      "    1    340 960R 961R C   N    7830 7832 -101.95 -125.20   24.62    0.80  -63.00  177.94   21.69\n",
      "    1        961R 961R N   CA   7832 7833  132.52  140.60                  -41.10\n",
      "    2    341 961R 962T C   N    7841 7843 -118.81 -124.80   19.70    0.68  -63.20  175.86   19.43\n",
      "    2        962T 962T N   CA   7843 7844  124.74  143.50                  -42.10\n",
      "    3    343 963V 964N C   N    7855 7857 -103.92 -119.90   25.18    0.76  -63.20  163.78   18.52\n",
      "    3        964N 964N N   CA   7857 7858  117.53  137.00                  -41.10\n",
      "    4    345 965P 966Q C   N    7870 7872   59.41  -63.80  140.08   25.38  -63.80  140.08   25.38\n",
      "    4        966Q 966Q N   CA   7872 7873   26.36  -40.30                  -40.30\n",
      "    5    346 966Q 967D C   N    7879 7881 -161.24  -96.50   70.94    2.89  -63.30  158.95   17.26\n",
      "    5        967D 967D N   CA   7881 7882   85.19  114.20                  -40.00\n",
      "    6    348 968P 969S C   N    7894 7896   55.91  -64.10  152.65   19.51  -64.10  152.65   19.51\n",
      "    6        969S 969S N   CA   7896 7897   59.34  -35.00                  -35.00\n",
      "    7    349 969S 970Y C   N    7900 7902 -154.46 -124.30   46.51    3.67  -63.50  169.81   23.50\n",
      "    7        970Y 970Y N   CA   7902 7903  100.00  135.40                  -43.40\n",
      "    8    351 971V 972D C   N    7919 7921 -168.29  -96.50   76.08    3.14  -63.30 -152.16   22.08\n",
      "    8        972D 972D N   CA   7921 7922  139.37  114.20                  -40.00\n",
      "\n",
      "\n",
      "report______> Distribution of short non-bonded contacts:\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "  serious non-bonded atom clash:  1055 1178   2.040\n",
      "  serious non-bonded atom clash:  1976 5131   1.952\n",
      "  serious non-bonded atom clash:  4241 4364   2.034\n",
      "  serious non-bonded atom clash:  7368 7491   2.036\n",
      "  serious non-bonded atom clash:  7778 8016   2.297\n",
      "  serious non-bonded atom clash: 1050010623   2.039\n",
      "  serious non-bonded atom clash: 1103111037   2.051\n",
      "  serious non-bonded atom clash: 1142114576   2.100\n",
      "  serious non-bonded atom clash: 1368613809   2.037\n",
      "  serious non-bonded atom clash: 1414614278   2.265\n",
      "\n",
      "DISTANCE1:  0.00 2.10 2.20 2.30 2.40 2.50 2.60 2.70 2.80 2.90 3.00 3.10 3.20 3.30 3.40\n",
      "DISTANCE2:  2.10 2.20 2.30 2.40 2.50 2.60 2.70 2.80 2.90 3.00 3.10 3.20 3.30 3.40 3.50\n",
      "FREQUENCY:     7    1    2   13   11   34   77   79  249  691  983  898 1797 2409 2304\n",
      "\n",
      "\n",
      "<< end of ENERGY.\n",
      "openf___224_> Open           muscle.IL00000001.pdb\n",
      "wrpdb___568_> Residues, atoms, selected atoms:     1935    15763    15763\n",
      "openf___224_> Open           muscle.IL00000001.pdb\n",
      "\n",
      "Dynamically allocated memory at       amaxmodel [B,KiB,MiB]:     45612318   44543.281    43.499\n",
      "read_mo_297_> Segments, residues, atoms:        5     1935    15763\n",
      "read_mo_298_> Segment:        1     1 A   469 A     3168\n",
      "read_mo_298_> Segment:        2     1 B   459 B     3150\n",
      "read_mo_298_> Segment:        3     4 C   459 C     3127\n",
      "read_mo_298_> Segment:        4     1 D   469 D     3168\n",
      "read_mo_298_> Segment:        5     1 E   459 E     3150\n",
      "randomi_498_> Atoms,selected atoms,random_seed,amplitude:    15763       86        1        5.0000\n",
      "randomi_496_> Amplitude is > 0; randomization is done.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "\n",
      "\n",
      ">> ENERGY; Differences between the model's features and restraints:\n",
      "Number of all residues in MODEL                   :     1935\n",
      "Number of all, selected real atoms                :    15763      86\n",
      "Number of all, selected pseudo atoms              :        0       0\n",
      "Number of all static, selected restraints         :      385     385\n",
      "COVALENT_CYS                                      :        F\n",
      "NONBONDED_SEL_ATOMS                               :        1\n",
      "Number of non-bonded pairs (excluding 1-2,1-3,1-4):     1937\n",
      "Dynamic pairs routine                             : 5, NATM_SEL x NATM double loop\n",
      "Atomic shift for contacts update (UPDATE_DYNAMIC) :    0.390\n",
      "LENNARD_JONES_SWITCH                              :    6.500   7.500\n",
      "COULOMB_JONES_SWITCH                              :    6.500   7.500\n",
      "RESIDUE_SPAN_RANGE                                :        0   99999\n",
      "NLOGN_USE                                         :       15\n",
      "CONTACT_SHELL                                     :    7.000\n",
      "DYNAMIC_PAIRS,_SPHERE,_COULOMB,_LENNARD,_MODELLER :        T       T       F       F       T\n",
      "SPHERE_STDV                                       :    0.050\n",
      "RADII_FACTOR                                      :    0.820\n",
      "Current energy                                    :          76.5386\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Summary of the restraint violations: \n",
      "\n",
      "   NUM     ... number of restraints.\n",
      "   NUMVI   ... number of restraints with RVIOL > VIOL_REPORT_CUT[i].\n",
      "   RVIOL   ... relative difference from the best value.\n",
      "   NUMVP   ... number of restraints with -Ln(pdf) > VIOL_REPORT_CUT2[i].\n",
      "   RMS_1   ... RMS(feature, minimally_violated_basis_restraint, NUMB).\n",
      "   RMS_2   ... RMS(feature, best_value, NUMB).\n",
      "   MOL.PDF ... scaled contribution to -Ln(Molecular pdf).\n",
      "\n",
      " #                     RESTRAINT_GROUP      NUM   NUMVI  NUMVP   RMS_1   RMS_2         MOL.PDF     S_i\n",
      "------------------------------------------------------------------------------------------------------\n",
      " 1 Bond length potential              :      90       0      0   0.008   0.008      1.6891       1.000\n",
      " 2 Bond angle potential               :     128       0      0   3.176   3.176      23.367       1.000\n",
      " 3 Stereochemical cosine torsion poten:      81       0      1  43.758  43.758      24.980       1.000\n",
      " 4 Stereochemical improper torsion pot:      40       0      0   1.279   1.279      1.4446       1.000\n",
      " 5 Soft-sphere overlap restraints     :    1937       0      0   0.000   0.000      0.0000       1.000\n",
      " 6 Lennard-Jones 6-12 potential       :       0       0      0   0.000   0.000      0.0000       1.000\n",
      " 7 Coulomb point-point electrostatic p:       0       0      0   0.000   0.000      0.0000       1.000\n",
      " 8 H-bonding potential                :       0       0      0   0.000   0.000      0.0000       1.000\n",
      " 9 Distance restraints 1 (CA-CA)      :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "10 Distance restraints 2 (N-O)        :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "11 Mainchain Phi dihedral restraints  :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "12 Mainchain Psi dihedral restraints  :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "13 Mainchain Omega dihedral restraints:      12       0      0   4.581   4.581      2.9723       1.000\n",
      "14 Sidechain Chi_1 dihedral restraints:      11       0      1  67.137  67.137      8.6616       1.000\n",
      "15 Sidechain Chi_2 dihedral restraints:       7       0      0  93.358  93.358      4.8773       1.000\n",
      "16 Sidechain Chi_3 dihedral restraints:       3       0      0 106.187 106.187      1.2363       1.000\n",
      "17 Sidechain Chi_4 dihedral restraints:       0       0      0   0.000   0.000      0.0000       1.000\n",
      "18 Disulfide distance restraints      :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "19 Disulfide angle restraints         :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "20 Disulfide dihedral angle restraints:       0       0      0   0.000   0.000      0.0000       1.000\n",
      "21 Lower bound distance restraints    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "22 Upper bound distance restraints    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "23 Distance restraints 3 (SDCH-MNCH)  :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "24 Sidechain Chi_5 dihedral restraints:       0       0      0   0.000   0.000      0.0000       1.000\n",
      "25 Phi/Psi pair of dihedral restraints:      13       9      2  36.003 134.507      19.504       1.000\n",
      "26 Distance restraints 4 (SDCH-SDCH)  :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "27 Distance restraints 5 (X-Y)        :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "28 NMR distance restraints 6 (X-Y)    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "29 NMR distance restraints 7 (X-Y)    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "30 Minimal distance restraints        :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "31 Non-bonded restraints              :    1937       0      2   1.935   1.935     -12.194       1.000\n",
      "32 Atomic accessibility restraints    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "33 Atomic density restraints          :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "34 Absolute position restraints       :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "35 Dihedral angle difference restraint:       0       0      0   0.000   0.000      0.0000       1.000\n",
      "36 GBSA implicit solvent potential    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "37 EM density fitting potential       :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "38 SAXS restraints                    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "39 Symmetry restraints                :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "\n",
      "\n",
      "\n",
      "List of the violated restraints:\n",
      "   A restraint is violated when the relative difference\n",
      "   from the best value (RVIOL) is larger than CUTOFF.\n",
      "\n",
      "   ICSR   ... index of a restraint in the current set.\n",
      "   RESNO  ... residue numbers of the first two atoms.\n",
      "   ATM    ... IUPAC atom names of the first two atoms.\n",
      "   FEAT   ... the value of the feature in the model.\n",
      "   restr  ... the mean of the basis restraint with the smallest\n",
      "              difference from the model (local minimum).\n",
      "   viol   ... difference from the local minimum.\n",
      "   rviol  ... relative difference from the local minimum.\n",
      "   RESTR  ... the best value (global minimum).\n",
      "   VIOL   ... difference from the best value.\n",
      "   RVIOL  ... relative difference from the best value.\n",
      "\n",
      "\n",
      "-------------------------------------------------------------------------------------------------\n",
      "\n",
      "Feature 25                           : Phi/Psi pair of dihedral restraints     \n",
      "List of the RVIOL violations larger than   :       6.5000\n",
      "\n",
      "    #   ICSR  RESNO1/2 ATM1/2   INDATM1/2    FEAT   restr    viol   rviol   RESTR    VIOL   RVIOL\n",
      "    1    340 960R 961R C   N    7830 7832 -101.95 -125.20   24.55    0.80  -63.00  178.12   21.71\n",
      "    1        961R 961R N   CA   7832 7833  132.71  140.60                  -41.10\n",
      "    2    341 961R 962T C   N    7841 7843  -98.24  -78.10   23.03    1.53  -63.20 -177.35   25.76\n",
      "    2        962T 962T N   CA   7843 7844  138.64  149.80                  -42.10\n",
      "    3    343 963V 964N C   N    7855 7857 -119.82 -119.90   20.64    0.94  -63.20  167.33   18.42\n",
      "    3        964N 964N N   CA   7857 7858  116.36  137.00                  -41.10\n",
      "    4    344 964N 965P C   N    7863 7865  -78.25  -58.70   28.05    1.95  -64.50  158.19   12.54\n",
      "    4        965P 965P N   CA   7865 7866  -10.39  -30.50                  147.20\n",
      "    5    345 965P 966Q C   N    7870 7872 -103.95 -121.10   71.74    3.65  -63.80  117.42   19.40\n",
      "    5        966Q 966Q N   CA   7872 7873 -150.64  139.70                  -40.30\n",
      "    6    346 966Q 967D C   N    7879 7881  -75.11  -70.90    7.92    0.27  -63.30  163.42   20.91\n",
      "    6        967D 967D N   CA   7881 7882  157.01  150.30                  -40.00\n",
      "    7    348 968P 969S C   N    7894 7896 -141.31 -136.60   16.10    0.89  -64.10 -172.55   11.75\n",
      "    7        969S 969S N   CA   7896 7897  135.81  151.20                  -35.00\n",
      "    8    349 969S 970Y C   N    7900 7902 -129.23 -124.30   12.47    0.87  -63.50  179.79   25.60\n",
      "    8        970Y 970Y N   CA   7902 7903  123.94  135.40                  -43.40\n",
      "    9    351 971V 972D C   N    7919 7921 -109.50  -63.30   82.43   13.11  -63.30   82.43   13.11\n",
      "    9        972D 972D N   CA   7921 7922 -108.26  -40.00                  -40.00\n",
      "\n",
      "\n",
      "report______> Distribution of short non-bonded contacts:\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "  serious non-bonded atom clash:  1055 1178   2.040\n",
      "  serious non-bonded atom clash:  1976 5131   1.952\n",
      "  serious non-bonded atom clash:  4241 4364   2.034\n",
      "  serious non-bonded atom clash:  7368 7491   2.036\n",
      "  serious non-bonded atom clash:  7778 8016   2.297\n",
      "  serious non-bonded atom clash: 1050010623   2.039\n",
      "  serious non-bonded atom clash: 1103111037   2.051\n",
      "  serious non-bonded atom clash: 1142114576   2.100\n",
      "  serious non-bonded atom clash: 1368613809   2.037\n",
      "  serious non-bonded atom clash: 1414614278   2.265\n",
      "\n",
      "DISTANCE1:  0.00 2.10 2.20 2.30 2.40 2.50 2.60 2.70 2.80 2.90 3.00 3.10 3.20 3.30 3.40\n",
      "DISTANCE2:  2.10 2.20 2.30 2.40 2.50 2.60 2.70 2.80 2.90 3.00 3.10 3.20 3.30 3.40 3.50\n",
      "FREQUENCY:     7    1    2   13   11   33   76   81  252  691  982  897 1797 2406 2304\n",
      "\n",
      "\n",
      "<< end of ENERGY.\n",
      "iupac_m_487_> NH1/2 swapped:     -179.8813       10    10\n",
      "iupac_m_487_> NH1/2 swapped:      179.9234       20    20\n",
      "iupac_m_487_> NH1/2 swapped:      179.8847      207   207\n",
      "iupac_m_487_> NH1/2 swapped:     -179.7626      327   327\n",
      "iupac_m_487_> NH1/2 swapped:      180.0000      405    18\n",
      "iupac_m_487_> NH1/2 swapped:     -179.7577      411    24\n",
      "iupac_m_487_> NH1/2 swapped:      179.8703      462    75\n",
      "iupac_m_487_> NH1/2 swapped:      179.7842      470    83\n",
      "iupac_m_487_> NH1/2 swapped:     -179.7762      475    88\n",
      "iupac_m_487_> NH1/2 swapped:      179.8481      502   115\n",
      "iupac_m_487_> NH1/2 swapped:     -179.9093      701   314\n",
      "iupac_m_487_> NH1/2 swapped:     -179.7168      715   328\n",
      "iupac_m_487_> NH1/2 swapped:      179.8615      716   400\n",
      "iupac_m_487_> NH1/2 swapped:     -179.9657      840    68\n",
      "iupac_m_487_> NH1/2 swapped:     -179.9209      847    75\n",
      "iupac_m_487_> NH1/2 swapped:      179.7175      860    88\n",
      "iupac_m_487_> NH1/2 swapped:     -180.0000      887   115\n",
      "iupac_m_487_> NH1/2 swapped:     -178.9152      923   151\n",
      "iupac_m_487_> NH1/2 swapped:      179.6326      960   188\n",
      "iupac_m_483_> CD1/2 CE1/2 swapped:      -92.6622      970   198\n",
      "iupac_m_397W> Atoms were not swapped because of the uncertainty of how to handle the H atom.\n",
      "iupac_m_485_> OD1/2 will be swapped:     -130.3766      972   200\n",
      "iupac_m_487_> NH1/2 swapped:      179.9558     1086   314\n",
      "iupac_m_487_> NH1/2 swapped:     -179.5980     1100   328\n",
      "iupac_m_487_> NH1/2 swapped:     -179.6961     1170    10\n",
      "iupac_m_487_> NH1/2 swapped:     -179.8275     1180    20\n",
      "iupac_m_486_> OE1/2 will be swapped:     -109.6868     1332   172\n",
      "iupac_m_487_> NH1/2 swapped:      179.9477     1367   207\n",
      "iupac_m_487_> NH1/2 swapped:      179.9115     1487   327\n",
      "iupac_m_487_> NH1/2 swapped:     -179.5082     1571    24\n",
      "iupac_m_487_> NH1/2 swapped:      179.9720     1622    75\n",
      "iupac_m_487_> NH1/2 swapped:      180.0000     1630    83\n",
      "iupac_m_487_> NH1/2 swapped:     -179.8864     1635    88\n",
      "iupac_m_487_> NH1/2 swapped:      179.5804     1662   115\n",
      "iupac_m_487_> NH1/2 swapped:      179.8072     1861   314\n",
      "iupac_m_487_> NH1/2 swapped:      179.7521     1875   328\n",
      "iupac_m_487_> NH1/2 swapped:      179.8587     1876   400\n",
      "openf___224_> Open           muscle.BL00010001.pdb\n",
      "wrpdb___568_> Residues, atoms, selected atoms:     1935    15763    15763\n",
      "openf___224_> Open           muscle.IL00000001.pdb\n",
      "\n",
      "Dynamically allocated memory at       amaxmodel [B,KiB,MiB]:     45612318   44543.281    43.499\n",
      "read_mo_297_> Segments, residues, atoms:        5     1935    15763\n",
      "read_mo_298_> Segment:        1     1 A   469 A     3168\n",
      "read_mo_298_> Segment:        2     1 B   459 B     3150\n",
      "read_mo_298_> Segment:        3     4 C   459 C     3127\n",
      "read_mo_298_> Segment:        4     1 D   469 D     3168\n",
      "read_mo_298_> Segment:        5     1 E   459 E     3150\n",
      "randomi_498_> Atoms,selected atoms,random_seed,amplitude:    15763       86        1        5.0000\n",
      "randomi_496_> Amplitude is > 0; randomization is done.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "\n",
      "\n",
      ">> ENERGY; Differences between the model's features and restraints:\n",
      "Number of all residues in MODEL                   :     1935\n",
      "Number of all, selected real atoms                :    15763      86\n",
      "Number of all, selected pseudo atoms              :        0       0\n",
      "Number of all static, selected restraints         :      385     385\n",
      "COVALENT_CYS                                      :        F\n",
      "NONBONDED_SEL_ATOMS                               :        1\n",
      "Number of non-bonded pairs (excluding 1-2,1-3,1-4):     1819\n",
      "Dynamic pairs routine                             : 5, NATM_SEL x NATM double loop\n",
      "Atomic shift for contacts update (UPDATE_DYNAMIC) :    0.390\n",
      "LENNARD_JONES_SWITCH                              :    6.500   7.500\n",
      "COULOMB_JONES_SWITCH                              :    6.500   7.500\n",
      "RESIDUE_SPAN_RANGE                                :        0   99999\n",
      "NLOGN_USE                                         :       15\n",
      "CONTACT_SHELL                                     :    7.000\n",
      "DYNAMIC_PAIRS,_SPHERE,_COULOMB,_LENNARD,_MODELLER :        T       T       F       F       T\n",
      "SPHERE_STDV                                       :    0.050\n",
      "RADII_FACTOR                                      :    0.820\n",
      "Current energy                                    :          40.1934\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Summary of the restraint violations: \n",
      "\n",
      "   NUM     ... number of restraints.\n",
      "   NUMVI   ... number of restraints with RVIOL > VIOL_REPORT_CUT[i].\n",
      "   RVIOL   ... relative difference from the best value.\n",
      "   NUMVP   ... number of restraints with -Ln(pdf) > VIOL_REPORT_CUT2[i].\n",
      "   RMS_1   ... RMS(feature, minimally_violated_basis_restraint, NUMB).\n",
      "   RMS_2   ... RMS(feature, best_value, NUMB).\n",
      "   MOL.PDF ... scaled contribution to -Ln(Molecular pdf).\n",
      "\n",
      " #                     RESTRAINT_GROUP      NUM   NUMVI  NUMVP   RMS_1   RMS_2         MOL.PDF     S_i\n",
      "------------------------------------------------------------------------------------------------------\n",
      " 1 Bond length potential              :      90       0      0   0.009   0.009      2.1154       1.000\n",
      " 2 Bond angle potential               :     128       0      1   3.023   3.023      19.618       1.000\n",
      " 3 Stereochemical cosine torsion poten:      81       0      3  45.468  45.468      27.471       1.000\n",
      " 4 Stereochemical improper torsion pot:      40       0      0   1.600   1.600      2.0806       1.000\n",
      " 5 Soft-sphere overlap restraints     :    1819       0      0   0.000   0.000     0.75521E-05   1.000\n",
      " 6 Lennard-Jones 6-12 potential       :       0       0      0   0.000   0.000      0.0000       1.000\n",
      " 7 Coulomb point-point electrostatic p:       0       0      0   0.000   0.000      0.0000       1.000\n",
      " 8 H-bonding potential                :       0       0      0   0.000   0.000      0.0000       1.000\n",
      " 9 Distance restraints 1 (CA-CA)      :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "10 Distance restraints 2 (N-O)        :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "11 Mainchain Phi dihedral restraints  :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "12 Mainchain Psi dihedral restraints  :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "13 Mainchain Omega dihedral restraints:      12       0      0   3.804   3.804      2.0498       1.000\n",
      "14 Sidechain Chi_1 dihedral restraints:      11       0      1  62.156  62.156      7.0916       1.000\n",
      "15 Sidechain Chi_2 dihedral restraints:       7       0      1  49.989  49.989      6.6142       1.000\n",
      "16 Sidechain Chi_3 dihedral restraints:       3       0      0  83.059  83.059     0.35823       1.000\n",
      "17 Sidechain Chi_4 dihedral restraints:       0       0      0   0.000   0.000      0.0000       1.000\n",
      "18 Disulfide distance restraints      :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "19 Disulfide angle restraints         :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "20 Disulfide dihedral angle restraints:       0       0      0   0.000   0.000      0.0000       1.000\n",
      "21 Lower bound distance restraints    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "22 Upper bound distance restraints    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "23 Distance restraints 3 (SDCH-MNCH)  :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "24 Sidechain Chi_5 dihedral restraints:       0       0      0   0.000   0.000      0.0000       1.000\n",
      "25 Phi/Psi pair of dihedral restraints:      13       8      2  39.410 137.014      17.948       1.000\n",
      "26 Distance restraints 4 (SDCH-SDCH)  :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "27 Distance restraints 5 (X-Y)        :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "28 NMR distance restraints 6 (X-Y)    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "29 NMR distance restraints 7 (X-Y)    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "30 Minimal distance restraints        :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "31 Non-bonded restraints              :    1819       0      0   1.849   1.849     -45.154       1.000\n",
      "32 Atomic accessibility restraints    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "33 Atomic density restraints          :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "34 Absolute position restraints       :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "35 Dihedral angle difference restraint:       0       0      0   0.000   0.000      0.0000       1.000\n",
      "36 GBSA implicit solvent potential    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "37 EM density fitting potential       :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "38 SAXS restraints                    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "39 Symmetry restraints                :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "\n",
      "\n",
      "\n",
      "List of the violated restraints:\n",
      "   A restraint is violated when the relative difference\n",
      "   from the best value (RVIOL) is larger than CUTOFF.\n",
      "\n",
      "   ICSR   ... index of a restraint in the current set.\n",
      "   RESNO  ... residue numbers of the first two atoms.\n",
      "   ATM    ... IUPAC atom names of the first two atoms.\n",
      "   FEAT   ... the value of the feature in the model.\n",
      "   restr  ... the mean of the basis restraint with the smallest\n",
      "              difference from the model (local minimum).\n",
      "   viol   ... difference from the local minimum.\n",
      "   rviol  ... relative difference from the local minimum.\n",
      "   RESTR  ... the best value (global minimum).\n",
      "   VIOL   ... difference from the best value.\n",
      "   RVIOL  ... relative difference from the best value.\n",
      "\n",
      "\n",
      "-------------------------------------------------------------------------------------------------\n",
      "\n",
      "Feature 25                           : Phi/Psi pair of dihedral restraints     \n",
      "List of the RVIOL violations larger than   :       6.5000\n",
      "\n",
      "    #   ICSR  RESNO1/2 ATM1/2   INDATM1/2    FEAT   restr    viol   rviol   RESTR    VIOL   RVIOL\n",
      "    1    340 960R 961R C   N    7830 7832 -101.95 -125.20   24.64    0.80  -63.00  177.86   21.68\n",
      "    1        961R 961R N   CA   7832 7833  132.45  140.60                  -41.10\n",
      "    2    341 961R 962T C   N    7841 7843  -64.46  -78.10   36.23    1.21  -63.20  158.33   20.22\n",
      "    2        962T 962T N   CA   7843 7844  116.23  149.80                  -42.10\n",
      "    3    343 963V 964N C   N    7855 7857 -128.32 -119.90   21.25    1.12  -63.20  171.44   18.69\n",
      "    3        964N 964N N   CA   7857 7858  117.49  137.00                  -41.10\n",
      "    4    345 965P 966Q C   N    7870 7872  153.75 -121.10  108.47    3.40  -63.80 -178.30   32.95\n",
      "    4        966Q 966Q N   CA   7872 7873 -153.10  139.70                  -40.30\n",
      "    5    346 966Q 967D C   N    7879 7881  -76.38  -70.90    7.88    0.31  -63.30  164.55   21.13\n",
      "    5        967D 967D N   CA   7881 7882  155.97  150.30                  -40.00\n",
      "    6    348 968P 969S C   N    7894 7896 -135.01 -136.60   28.72    1.49  -64.10  161.52   16.78\n",
      "    6        969S 969S N   CA   7896 7897  179.87  151.20                  -35.00\n",
      "    7    349 969S 970Y C   N    7900 7902  -75.80  -98.40   23.00    0.93  -63.50  176.47   27.39\n",
      "    7        970Y 970Y N   CA   7902 7903  132.64  128.40                  -43.40\n",
      "    8    351 971V 972D C   N    7919 7921 -132.28  -96.50   43.26    1.80  -63.30 -168.63   20.54\n",
      "    8        972D 972D N   CA   7921 7922  138.51  114.20                  -40.00\n",
      "\n",
      "\n",
      "report______> Distribution of short non-bonded contacts:\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "  serious non-bonded atom clash:  1055 1178   2.040\n",
      "  serious non-bonded atom clash:  1976 5131   1.952\n",
      "  serious non-bonded atom clash:  4241 4364   2.034\n",
      "  serious non-bonded atom clash:  7368 7491   2.036\n",
      "  serious non-bonded atom clash:  7778 8016   2.297\n",
      "  serious non-bonded atom clash: 1050010623   2.039\n",
      "  serious non-bonded atom clash: 1103111037   2.051\n",
      "  serious non-bonded atom clash: 1142114576   2.100\n",
      "  serious non-bonded atom clash: 1368613809   2.037\n",
      "  serious non-bonded atom clash: 1414614278   2.265\n",
      "\n",
      "DISTANCE1:  0.00 2.10 2.20 2.30 2.40 2.50 2.60 2.70 2.80 2.90 3.00 3.10 3.20 3.30 3.40\n",
      "DISTANCE2:  2.10 2.20 2.30 2.40 2.50 2.60 2.70 2.80 2.90 3.00 3.10 3.20 3.30 3.40 3.50\n",
      "FREQUENCY:     7    1    2   13   11   35   76   84  252  693  981  893 1797 2413 2308\n",
      "\n",
      "\n",
      "<< end of ENERGY.\n",
      "iupac_m_487_> NH1/2 swapped:     -179.8813       10    10\n",
      "iupac_m_487_> NH1/2 swapped:      179.9234       20    20\n",
      "iupac_m_487_> NH1/2 swapped:      179.8847      207   207\n",
      "iupac_m_487_> NH1/2 swapped:     -179.7626      327   327\n",
      "iupac_m_487_> NH1/2 swapped:      180.0000      405    18\n",
      "iupac_m_487_> NH1/2 swapped:     -179.7577      411    24\n",
      "iupac_m_487_> NH1/2 swapped:      179.8703      462    75\n",
      "iupac_m_487_> NH1/2 swapped:      179.7842      470    83\n",
      "iupac_m_487_> NH1/2 swapped:     -179.7762      475    88\n",
      "iupac_m_487_> NH1/2 swapped:      179.8481      502   115\n",
      "iupac_m_487_> NH1/2 swapped:     -179.9093      701   314\n",
      "iupac_m_487_> NH1/2 swapped:     -179.7168      715   328\n",
      "iupac_m_487_> NH1/2 swapped:      179.8615      716   400\n",
      "iupac_m_487_> NH1/2 swapped:     -179.9657      840    68\n",
      "iupac_m_487_> NH1/2 swapped:     -179.9209      847    75\n",
      "iupac_m_487_> NH1/2 swapped:      179.7175      860    88\n",
      "iupac_m_487_> NH1/2 swapped:     -180.0000      887   115\n",
      "iupac_m_487_> NH1/2 swapped:     -178.9152      923   151\n",
      "iupac_m_487_> NH1/2 swapped:      179.6326      960   188\n",
      "iupac_m_487_> NH1/2 swapped:      179.9558     1086   314\n",
      "iupac_m_487_> NH1/2 swapped:     -179.5980     1100   328\n",
      "iupac_m_487_> NH1/2 swapped:     -179.6961     1170    10\n",
      "iupac_m_487_> NH1/2 swapped:     -179.8275     1180    20\n",
      "iupac_m_486_> OE1/2 will be swapped:     -109.6868     1332   172\n",
      "iupac_m_487_> NH1/2 swapped:      179.9477     1367   207\n",
      "iupac_m_487_> NH1/2 swapped:      179.9115     1487   327\n",
      "iupac_m_487_> NH1/2 swapped:     -179.5082     1571    24\n",
      "iupac_m_487_> NH1/2 swapped:      179.9720     1622    75\n",
      "iupac_m_487_> NH1/2 swapped:      180.0000     1630    83\n",
      "iupac_m_487_> NH1/2 swapped:     -179.8864     1635    88\n",
      "iupac_m_487_> NH1/2 swapped:      179.5804     1662   115\n",
      "iupac_m_487_> NH1/2 swapped:      179.8072     1861   314\n",
      "iupac_m_487_> NH1/2 swapped:      179.7521     1875   328\n",
      "iupac_m_487_> NH1/2 swapped:      179.8587     1876   400\n",
      "openf___224_> Open           muscle.BL00020001.pdb\n",
      "wrpdb___568_> Residues, atoms, selected atoms:     1935    15763    15763\n",
      "openf___224_> Open           muscle.IL00000001.pdb\n",
      "\n",
      "Dynamically allocated memory at       amaxmodel [B,KiB,MiB]:     45612318   44543.281    43.499\n",
      "read_mo_297_> Segments, residues, atoms:        5     1935    15763\n",
      "read_mo_298_> Segment:        1     1 A   469 A     3168\n",
      "read_mo_298_> Segment:        2     1 B   459 B     3150\n",
      "read_mo_298_> Segment:        3     4 C   459 C     3127\n",
      "read_mo_298_> Segment:        4     1 D   469 D     3168\n",
      "read_mo_298_> Segment:        5     1 E   459 E     3150\n",
      "randomi_498_> Atoms,selected atoms,random_seed,amplitude:    15763       86        1        5.0000\n",
      "randomi_496_> Amplitude is > 0; randomization is done.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "\n",
      "\n",
      ">> ENERGY; Differences between the model's features and restraints:\n",
      "Number of all residues in MODEL                   :     1935\n",
      "Number of all, selected real atoms                :    15763      86\n",
      "Number of all, selected pseudo atoms              :        0       0\n",
      "Number of all static, selected restraints         :      385     385\n",
      "COVALENT_CYS                                      :        F\n",
      "NONBONDED_SEL_ATOMS                               :        1\n",
      "Number of non-bonded pairs (excluding 1-2,1-3,1-4):     2060\n",
      "Dynamic pairs routine                             : 5, NATM_SEL x NATM double loop\n",
      "Atomic shift for contacts update (UPDATE_DYNAMIC) :    0.390\n",
      "LENNARD_JONES_SWITCH                              :    6.500   7.500\n",
      "COULOMB_JONES_SWITCH                              :    6.500   7.500\n",
      "RESIDUE_SPAN_RANGE                                :        0   99999\n",
      "NLOGN_USE                                         :       15\n",
      "CONTACT_SHELL                                     :    7.000\n",
      "DYNAMIC_PAIRS,_SPHERE,_COULOMB,_LENNARD,_MODELLER :        T       T       F       F       T\n",
      "SPHERE_STDV                                       :    0.050\n",
      "RADII_FACTOR                                      :    0.820\n",
      "Current energy                                    :          61.4125\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Summary of the restraint violations: \n",
      "\n",
      "   NUM     ... number of restraints.\n",
      "   NUMVI   ... number of restraints with RVIOL > VIOL_REPORT_CUT[i].\n",
      "   RVIOL   ... relative difference from the best value.\n",
      "   NUMVP   ... number of restraints with -Ln(pdf) > VIOL_REPORT_CUT2[i].\n",
      "   RMS_1   ... RMS(feature, minimally_violated_basis_restraint, NUMB).\n",
      "   RMS_2   ... RMS(feature, best_value, NUMB).\n",
      "   MOL.PDF ... scaled contribution to -Ln(Molecular pdf).\n",
      "\n",
      " #                     RESTRAINT_GROUP      NUM   NUMVI  NUMVP   RMS_1   RMS_2         MOL.PDF     S_i\n",
      "------------------------------------------------------------------------------------------------------\n",
      " 1 Bond length potential              :      90       0      0   0.009   0.009      2.1753       1.000\n",
      " 2 Bond angle potential               :     128       1      2   3.649   3.649      34.068       1.000\n",
      " 3 Stereochemical cosine torsion poten:      81       0      0  45.399  45.399      26.932       1.000\n",
      " 4 Stereochemical improper torsion pot:      40       0      0   2.038   2.038      3.3253       1.000\n",
      " 5 Soft-sphere overlap restraints     :    2060       0      0   0.000   0.000     0.25695E-01   1.000\n",
      " 6 Lennard-Jones 6-12 potential       :       0       0      0   0.000   0.000      0.0000       1.000\n",
      " 7 Coulomb point-point electrostatic p:       0       0      0   0.000   0.000      0.0000       1.000\n",
      " 8 H-bonding potential                :       0       0      0   0.000   0.000      0.0000       1.000\n",
      " 9 Distance restraints 1 (CA-CA)      :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "10 Distance restraints 2 (N-O)        :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "11 Mainchain Phi dihedral restraints  :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "12 Mainchain Psi dihedral restraints  :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "13 Mainchain Omega dihedral restraints:      12       0      0   5.861   5.861      4.8643       1.000\n",
      "14 Sidechain Chi_1 dihedral restraints:      11       0      0  83.587  83.587      4.2256       1.000\n",
      "15 Sidechain Chi_2 dihedral restraints:       7       0      0  53.517  53.517      4.1824       1.000\n",
      "16 Sidechain Chi_3 dihedral restraints:       3       0      0  54.372  54.372      1.3100       1.000\n",
      "17 Sidechain Chi_4 dihedral restraints:       0       0      0   0.000   0.000      0.0000       1.000\n",
      "18 Disulfide distance restraints      :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "19 Disulfide angle restraints         :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "20 Disulfide dihedral angle restraints:       0       0      0   0.000   0.000      0.0000       1.000\n",
      "21 Lower bound distance restraints    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "22 Upper bound distance restraints    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "23 Distance restraints 3 (SDCH-MNCH)  :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "24 Sidechain Chi_5 dihedral restraints:       0       0      0   0.000   0.000      0.0000       1.000\n",
      "25 Phi/Psi pair of dihedral restraints:      13       8      2  30.694 133.113      18.157       1.000\n",
      "26 Distance restraints 4 (SDCH-SDCH)  :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "27 Distance restraints 5 (X-Y)        :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "28 NMR distance restraints 6 (X-Y)    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "29 NMR distance restraints 7 (X-Y)    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "30 Minimal distance restraints        :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "31 Non-bonded restraints              :    2060       0      2   1.862   1.862     -37.854       1.000\n",
      "32 Atomic accessibility restraints    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "33 Atomic density restraints          :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "34 Absolute position restraints       :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "35 Dihedral angle difference restraint:       0       0      0   0.000   0.000      0.0000       1.000\n",
      "36 GBSA implicit solvent potential    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "37 EM density fitting potential       :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "38 SAXS restraints                    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "39 Symmetry restraints                :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "\n",
      "\n",
      "\n",
      "List of the violated restraints:\n",
      "   A restraint is violated when the relative difference\n",
      "   from the best value (RVIOL) is larger than CUTOFF.\n",
      "\n",
      "   ICSR   ... index of a restraint in the current set.\n",
      "   RESNO  ... residue numbers of the first two atoms.\n",
      "   ATM    ... IUPAC atom names of the first two atoms.\n",
      "   FEAT   ... the value of the feature in the model.\n",
      "   restr  ... the mean of the basis restraint with the smallest\n",
      "              difference from the model (local minimum).\n",
      "   viol   ... difference from the local minimum.\n",
      "   rviol  ... relative difference from the local minimum.\n",
      "   RESTR  ... the best value (global minimum).\n",
      "   VIOL   ... difference from the best value.\n",
      "   RVIOL  ... relative difference from the best value.\n",
      "\n",
      "\n",
      "-------------------------------------------------------------------------------------------------\n",
      "\n",
      "Feature  2                           : Bond angle potential                    \n",
      "List of the RVIOL violations larger than   :       4.5000\n",
      "\n",
      "    #   ICSR  RESNO1/2 ATM1/2   INDATM1/2    FEAT   restr    viol   rviol   RESTR    VIOL   RVIOL\n",
      "    1    209 972D 972D N   CA   7921 7922  123.16  107.00   16.16    4.64  107.00   16.16    4.64\n",
      "\n",
      "-------------------------------------------------------------------------------------------------\n",
      "\n",
      "Feature 25                           : Phi/Psi pair of dihedral restraints     \n",
      "List of the RVIOL violations larger than   :       6.5000\n",
      "\n",
      "    #   ICSR  RESNO1/2 ATM1/2   INDATM1/2    FEAT   restr    viol   rviol   RESTR    VIOL   RVIOL\n",
      "    1    340 960R 961R C   N    7830 7832 -101.95 -125.20   23.92    0.82  -63.00 -179.67   22.01\n",
      "    1        961R 961R N   CA   7832 7833  134.98  140.60                  -41.10\n",
      "    2    341 961R 962T C   N    7841 7843  -70.68  -78.10   28.38    1.00  -63.20  164.68   20.63\n",
      "    2        962T 962T N   CA   7843 7844  122.41  149.80                  -42.10\n",
      "    3    343 963V 964N C   N    7855 7857 -146.33 -119.90   31.94    0.94  -63.20 -176.16   27.65\n",
      "    3        964N 964N N   CA   7857 7858  154.93  137.00                  -41.10\n",
      "    4    344 964N 965P C   N    7863 7865  -60.02  -58.70   16.82    1.50  -64.50  165.59   12.60\n",
      "    4        965P 965P N   CA   7865 7866  -47.27  -30.50                  147.20\n",
      "    5    345 965P 966Q C   N    7870 7872 -110.62 -121.10   17.32    0.60  -63.80  172.68   23.21\n",
      "    5        966Q 966Q N   CA   7872 7873  125.91  139.70                  -40.30\n",
      "    6    346 966Q 967D C   N    7879 7881 -121.02  -96.50   25.42    1.04  -63.30  170.94   18.44\n",
      "    6        967D 967D N   CA   7881 7882  120.90  114.20                  -40.00\n",
      "    7    348 968P 969S C   N    7894 7896  -72.49  -72.40   12.58    0.74  -64.10  175.02   12.21\n",
      "    7        969S 969S N   CA   7896 7897  139.82  152.40                  -35.00\n",
      "    8    349 969S 970Y C   N    7900 7902 -133.23 -124.30   73.00    3.49  -63.50  129.18   23.78\n",
      "    8        970Y 970Y N   CA   7902 7903 -152.15  135.40                  -43.40\n",
      "\n",
      "\n",
      "report______> Distribution of short non-bonded contacts:\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "  serious non-bonded atom clash:  1055 1178   2.040\n",
      "  serious non-bonded atom clash:  1976 5131   1.952\n",
      "  serious non-bonded atom clash:  4241 4364   2.034\n",
      "  serious non-bonded atom clash:  7368 7491   2.036\n",
      "  serious non-bonded atom clash:  7778 8016   2.297\n",
      "  serious non-bonded atom clash: 1050010623   2.039\n",
      "  serious non-bonded atom clash: 1103111037   2.051\n",
      "  serious non-bonded atom clash: 1142114576   2.100\n",
      "  serious non-bonded atom clash: 1368613809   2.037\n",
      "  serious non-bonded atom clash: 1414614278   2.265\n",
      "\n",
      "DISTANCE1:  0.00 2.10 2.20 2.30 2.40 2.50 2.60 2.70 2.80 2.90 3.00 3.10 3.20 3.30 3.40\n",
      "DISTANCE2:  2.10 2.20 2.30 2.40 2.50 2.60 2.70 2.80 2.90 3.00 3.10 3.20 3.30 3.40 3.50\n",
      "FREQUENCY:     7    1    2   13   11   37   76   84  256  692  988  900 1798 2411 2306\n",
      "\n",
      "\n",
      "<< end of ENERGY.\n",
      "iupac_m_487_> NH1/2 swapped:     -179.8813       10    10\n",
      "iupac_m_487_> NH1/2 swapped:      179.9234       20    20\n",
      "iupac_m_487_> NH1/2 swapped:      179.8847      207   207\n",
      "iupac_m_487_> NH1/2 swapped:     -179.7626      327   327\n",
      "iupac_m_487_> NH1/2 swapped:      180.0000      405    18\n",
      "iupac_m_487_> NH1/2 swapped:     -179.7577      411    24\n",
      "iupac_m_487_> NH1/2 swapped:      179.8703      462    75\n",
      "iupac_m_487_> NH1/2 swapped:      179.7842      470    83\n",
      "iupac_m_487_> NH1/2 swapped:     -179.7762      475    88\n",
      "iupac_m_487_> NH1/2 swapped:      179.8481      502   115\n",
      "iupac_m_487_> NH1/2 swapped:     -179.9093      701   314\n",
      "iupac_m_487_> NH1/2 swapped:     -179.7168      715   328\n",
      "iupac_m_487_> NH1/2 swapped:      179.8615      716   400\n",
      "iupac_m_487_> NH1/2 swapped:     -179.9657      840    68\n",
      "iupac_m_487_> NH1/2 swapped:     -179.9209      847    75\n",
      "iupac_m_487_> NH1/2 swapped:      179.7175      860    88\n",
      "iupac_m_487_> NH1/2 swapped:     -180.0000      887   115\n",
      "iupac_m_487_> NH1/2 swapped:     -178.9152      923   151\n",
      "iupac_m_487_> NH1/2 swapped:      179.6326      960   188\n",
      "iupac_m_483_> CD1/2 CE1/2 swapped:       98.7895      970   198\n",
      "iupac_m_397W> Atoms were not swapped because of the uncertainty of how to handle the H atom.\n",
      "iupac_m_487_> NH1/2 swapped:      179.9558     1086   314\n",
      "iupac_m_487_> NH1/2 swapped:     -179.5980     1100   328\n",
      "iupac_m_487_> NH1/2 swapped:     -179.6961     1170    10\n",
      "iupac_m_487_> NH1/2 swapped:     -179.8275     1180    20\n",
      "iupac_m_486_> OE1/2 will be swapped:     -109.6868     1332   172\n",
      "iupac_m_487_> NH1/2 swapped:      179.9477     1367   207\n",
      "iupac_m_487_> NH1/2 swapped:      179.9115     1487   327\n",
      "iupac_m_487_> NH1/2 swapped:     -179.5082     1571    24\n",
      "iupac_m_487_> NH1/2 swapped:      179.9720     1622    75\n",
      "iupac_m_487_> NH1/2 swapped:      180.0000     1630    83\n",
      "iupac_m_487_> NH1/2 swapped:     -179.8864     1635    88\n",
      "iupac_m_487_> NH1/2 swapped:      179.5804     1662   115\n",
      "iupac_m_487_> NH1/2 swapped:      179.8072     1861   314\n",
      "iupac_m_487_> NH1/2 swapped:      179.7521     1875   328\n",
      "iupac_m_487_> NH1/2 swapped:      179.8587     1876   400\n",
      "openf___224_> Open           muscle.BL00030001.pdb\n",
      "wrpdb___568_> Residues, atoms, selected atoms:     1935    15763    15763\n",
      "openf___224_> Open           muscle.IL00000001.pdb\n",
      "\n",
      "Dynamically allocated memory at       amaxmodel [B,KiB,MiB]:     45612318   44543.281    43.499\n",
      "read_mo_297_> Segments, residues, atoms:        5     1935    15763\n",
      "read_mo_298_> Segment:        1     1 A   469 A     3168\n",
      "read_mo_298_> Segment:        2     1 B   459 B     3150\n",
      "read_mo_298_> Segment:        3     4 C   459 C     3127\n",
      "read_mo_298_> Segment:        4     1 D   469 D     3168\n",
      "read_mo_298_> Segment:        5     1 E   459 E     3150\n",
      "randomi_498_> Atoms,selected atoms,random_seed,amplitude:    15763       86        1        5.0000\n",
      "randomi_496_> Amplitude is > 0; randomization is done.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "\n",
      "\n",
      ">> ENERGY; Differences between the model's features and restraints:\n",
      "Number of all residues in MODEL                   :     1935\n",
      "Number of all, selected real atoms                :    15763      86\n",
      "Number of all, selected pseudo atoms              :        0       0\n",
      "Number of all static, selected restraints         :      385     385\n",
      "COVALENT_CYS                                      :        F\n",
      "NONBONDED_SEL_ATOMS                               :        1\n",
      "Number of non-bonded pairs (excluding 1-2,1-3,1-4):     3029\n",
      "Dynamic pairs routine                             : 5, NATM_SEL x NATM double loop\n",
      "Atomic shift for contacts update (UPDATE_DYNAMIC) :    0.390\n",
      "LENNARD_JONES_SWITCH                              :    6.500   7.500\n",
      "COULOMB_JONES_SWITCH                              :    6.500   7.500\n",
      "RESIDUE_SPAN_RANGE                                :        0   99999\n",
      "NLOGN_USE                                         :       15\n",
      "CONTACT_SHELL                                     :    7.000\n",
      "DYNAMIC_PAIRS,_SPHERE,_COULOMB,_LENNARD,_MODELLER :        T       T       F       F       T\n",
      "SPHERE_STDV                                       :    0.050\n",
      "RADII_FACTOR                                      :    0.820\n",
      "Current energy                                    :         482.2588\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Summary of the restraint violations: \n",
      "\n",
      "   NUM     ... number of restraints.\n",
      "   NUMVI   ... number of restraints with RVIOL > VIOL_REPORT_CUT[i].\n",
      "   RVIOL   ... relative difference from the best value.\n",
      "   NUMVP   ... number of restraints with -Ln(pdf) > VIOL_REPORT_CUT2[i].\n",
      "   RMS_1   ... RMS(feature, minimally_violated_basis_restraint, NUMB).\n",
      "   RMS_2   ... RMS(feature, best_value, NUMB).\n",
      "   MOL.PDF ... scaled contribution to -Ln(Molecular pdf).\n",
      "\n",
      " #                     RESTRAINT_GROUP      NUM   NUMVI  NUMVP   RMS_1   RMS_2         MOL.PDF     S_i\n",
      "------------------------------------------------------------------------------------------------------\n",
      " 1 Bond length potential              :      90       0      2   0.026   0.026      18.240       1.000\n",
      " 2 Bond angle potential               :     128       3     10   5.366   5.366      69.948       1.000\n",
      " 3 Stereochemical cosine torsion poten:      81       0      7  52.938  52.938      36.318       1.000\n",
      " 4 Stereochemical improper torsion pot:      40       1      2   4.463   4.463      15.325       1.000\n",
      " 5 Soft-sphere overlap restraints     :    3029       3      5   0.015   0.015      79.250       1.000\n",
      " 6 Lennard-Jones 6-12 potential       :       0       0      0   0.000   0.000      0.0000       1.000\n",
      " 7 Coulomb point-point electrostatic p:       0       0      0   0.000   0.000      0.0000       1.000\n",
      " 8 H-bonding potential                :       0       0      0   0.000   0.000      0.0000       1.000\n",
      " 9 Distance restraints 1 (CA-CA)      :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "10 Distance restraints 2 (N-O)        :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "11 Mainchain Phi dihedral restraints  :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "12 Mainchain Psi dihedral restraints  :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "13 Mainchain Omega dihedral restraints:      12       0      3  12.074  12.074      20.645       1.000\n",
      "14 Sidechain Chi_1 dihedral restraints:      11       0      0  76.412  76.412      4.5266       1.000\n",
      "15 Sidechain Chi_2 dihedral restraints:       7       0      1  98.422  98.422      9.2960       1.000\n",
      "16 Sidechain Chi_3 dihedral restraints:       3       0      0  84.355  84.355      1.5831       1.000\n",
      "17 Sidechain Chi_4 dihedral restraints:       0       0      0   0.000   0.000      0.0000       1.000\n",
      "18 Disulfide distance restraints      :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "19 Disulfide angle restraints         :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "20 Disulfide dihedral angle restraints:       0       0      0   0.000   0.000      0.0000       1.000\n",
      "21 Lower bound distance restraints    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "22 Upper bound distance restraints    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "23 Distance restraints 3 (SDCH-MNCH)  :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "24 Sidechain Chi_5 dihedral restraints:       0       0      0   0.000   0.000      0.0000       1.000\n",
      "25 Phi/Psi pair of dihedral restraints:      13       8      9  56.746 128.287      50.968       1.000\n",
      "26 Distance restraints 4 (SDCH-SDCH)  :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "27 Distance restraints 5 (X-Y)        :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "28 NMR distance restraints 6 (X-Y)    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "29 NMR distance restraints 7 (X-Y)    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "30 Minimal distance restraints        :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "31 Non-bonded restraints              :    3029       0     44   1.902   1.902      176.16       1.000\n",
      "32 Atomic accessibility restraints    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "33 Atomic density restraints          :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "34 Absolute position restraints       :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "35 Dihedral angle difference restraint:       0       0      0   0.000   0.000      0.0000       1.000\n",
      "36 GBSA implicit solvent potential    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "37 EM density fitting potential       :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "38 SAXS restraints                    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "39 Symmetry restraints                :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "\n",
      "\n",
      "\n",
      "List of the violated restraints:\n",
      "   A restraint is violated when the relative difference\n",
      "   from the best value (RVIOL) is larger than CUTOFF.\n",
      "\n",
      "   ICSR   ... index of a restraint in the current set.\n",
      "   RESNO  ... residue numbers of the first two atoms.\n",
      "   ATM    ... IUPAC atom names of the first two atoms.\n",
      "   FEAT   ... the value of the feature in the model.\n",
      "   restr  ... the mean of the basis restraint with the smallest\n",
      "              difference from the model (local minimum).\n",
      "   viol   ... difference from the local minimum.\n",
      "   rviol  ... relative difference from the local minimum.\n",
      "   RESTR  ... the best value (global minimum).\n",
      "   VIOL   ... difference from the best value.\n",
      "   RVIOL  ... relative difference from the best value.\n",
      "\n",
      "\n",
      "-------------------------------------------------------------------------------------------------\n",
      "\n",
      "Feature  2                           : Bond angle potential                    \n",
      "List of the RVIOL violations larger than   :       4.5000\n",
      "\n",
      "    #   ICSR  RESNO1/2 ATM1/2   INDATM1/2    FEAT   restr    viol   rviol   RESTR    VIOL   RVIOL\n",
      "    1    173 969S 969S N   CA   7896 7897  123.28  107.00   16.28    4.68  107.00   16.28    4.68\n",
      "    2    178 969S 970Y C   N    7900 7902  140.76  120.00   20.76    4.72  120.00   20.76    4.72\n",
      "    3    181 970Y 970Y N   CA   7902 7903  124.56  107.00   17.56    5.05  107.00   17.56    5.05\n",
      "\n",
      "-------------------------------------------------------------------------------------------------\n",
      "\n",
      "Feature 25                           : Phi/Psi pair of dihedral restraints     \n",
      "List of the RVIOL violations larger than   :       6.5000\n",
      "\n",
      "    #   ICSR  RESNO1/2 ATM1/2   INDATM1/2    FEAT   restr    viol   rviol   RESTR    VIOL   RVIOL\n",
      "    1    340 960R 961R C   N    7830 7832 -101.95 -125.20   26.57    0.80  -63.00  173.28   21.07\n",
      "    1        961R 961R N   CA   7832 7833  127.75  140.60                  -41.10\n",
      "    2    341 961R 962T C   N    7841 7843  -81.27  -78.10    9.98    0.53  -63.20  178.48   24.08\n",
      "    2        962T 962T N   CA   7843 7844  140.33  149.80                  -42.10\n",
      "    3    343 963V 964N C   N    7855 7857 -167.16 -119.90   66.31    1.93  -63.20  170.70   26.95\n",
      "    3        964N 964N N   CA   7857 7858 -176.49  137.00                  -41.10\n",
      "    4    345 965P 966Q C   N    7870 7872  -99.13 -121.10   65.98    2.71  -63.80  122.96   16.45\n",
      "    4        966Q 966Q N   CA   7872 7873   77.48  139.70                  -40.30\n",
      "    5    346 966Q 967D C   N    7879 7881 -126.84  -63.30   64.45    9.66  -63.30   64.45    9.66\n",
      "    5        967D 967D N   CA   7881 7882  -29.18  -40.00                  -40.00\n",
      "    6    348 968P 969S C   N    7894 7896 -146.54 -136.60   35.82    1.98  -64.10  172.73   11.19\n",
      "    6        969S 969S N   CA   7896 7897  116.78  151.20                  -35.00\n",
      "    7    349 969S 970Y C   N    7900 7902 -153.24  -63.50   97.06   18.10  -63.50   97.06   18.10\n",
      "    7        970Y 970Y N   CA   7902 7903  -80.37  -43.40                  -43.40\n",
      "    8    351 971V 972D C   N    7919 7921   44.54   54.50   37.40    2.04  -63.30  159.08   26.26\n",
      "    8        972D 972D N   CA   7921 7922   76.95   40.90                  -40.00\n",
      "\n",
      "\n",
      "report______> Distribution of short non-bonded contacts:\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "  serious non-bonded atom clash:  1055 1178   2.040\n",
      "  serious non-bonded atom clash:  1976 5131   1.952\n",
      "  serious non-bonded atom clash:  4241 4364   2.034\n",
      "  serious non-bonded atom clash:  7368 7491   2.036\n",
      "  serious non-bonded atom clash:  7778 8016   2.297\n",
      "  serious non-bonded atom clash: 1050010623   2.039\n",
      "  serious non-bonded atom clash: 1103111037   2.051\n",
      "  serious non-bonded atom clash: 1142114576   2.100\n",
      "  serious non-bonded atom clash: 1368613809   2.037\n",
      "  serious non-bonded atom clash: 1414614278   2.265\n",
      "\n",
      "DISTANCE1:  0.00 2.10 2.20 2.30 2.40 2.50 2.60 2.70 2.80 2.90 3.00 3.10 3.20 3.30 3.40\n",
      "DISTANCE2:  2.10 2.20 2.30 2.40 2.50 2.60 2.70 2.80 2.90 3.00 3.10 3.20 3.30 3.40 3.50\n",
      "FREQUENCY:     7    1    2   13   14   37   82  100  261  696  991  911 1814 2412 2315\n",
      "\n",
      "\n",
      "<< end of ENERGY.\n",
      "iupac_m_487_> NH1/2 swapped:     -179.8813       10    10\n",
      "iupac_m_487_> NH1/2 swapped:      179.9234       20    20\n",
      "iupac_m_487_> NH1/2 swapped:      179.8847      207   207\n",
      "iupac_m_487_> NH1/2 swapped:     -179.7626      327   327\n",
      "iupac_m_487_> NH1/2 swapped:      180.0000      405    18\n",
      "iupac_m_487_> NH1/2 swapped:     -179.7577      411    24\n",
      "iupac_m_487_> NH1/2 swapped:      179.8703      462    75\n",
      "iupac_m_487_> NH1/2 swapped:      179.7842      470    83\n",
      "iupac_m_487_> NH1/2 swapped:     -179.7762      475    88\n",
      "iupac_m_487_> NH1/2 swapped:      179.8481      502   115\n",
      "iupac_m_487_> NH1/2 swapped:     -179.9093      701   314\n",
      "iupac_m_487_> NH1/2 swapped:     -179.7168      715   328\n",
      "iupac_m_487_> NH1/2 swapped:      179.8615      716   400\n",
      "iupac_m_487_> NH1/2 swapped:     -179.9657      840    68\n",
      "iupac_m_487_> NH1/2 swapped:     -179.9209      847    75\n",
      "iupac_m_487_> NH1/2 swapped:      179.7175      860    88\n",
      "iupac_m_487_> NH1/2 swapped:     -180.0000      887   115\n",
      "iupac_m_487_> NH1/2 swapped:     -178.9152      923   151\n",
      "iupac_m_487_> NH1/2 swapped:      179.6326      960   188\n",
      "iupac_m_483_> CD1/2 CE1/2 swapped:     -138.6134      970   198\n",
      "iupac_m_397W> Atoms were not swapped because of the uncertainty of how to handle the H atom.\n",
      "iupac_m_485_> OD1/2 will be swapped:      119.8465      972   200\n",
      "iupac_m_487_> NH1/2 swapped:      179.9558     1086   314\n",
      "iupac_m_487_> NH1/2 swapped:     -179.5980     1100   328\n",
      "iupac_m_487_> NH1/2 swapped:     -179.6961     1170    10\n",
      "iupac_m_487_> NH1/2 swapped:     -179.8275     1180    20\n",
      "iupac_m_486_> OE1/2 will be swapped:     -109.6868     1332   172\n",
      "iupac_m_487_> NH1/2 swapped:      179.9477     1367   207\n",
      "iupac_m_487_> NH1/2 swapped:      179.9115     1487   327\n",
      "iupac_m_487_> NH1/2 swapped:     -179.5082     1571    24\n",
      "iupac_m_487_> NH1/2 swapped:      179.9720     1622    75\n",
      "iupac_m_487_> NH1/2 swapped:      180.0000     1630    83\n",
      "iupac_m_487_> NH1/2 swapped:     -179.8864     1635    88\n",
      "iupac_m_487_> NH1/2 swapped:      179.5804     1662   115\n",
      "iupac_m_487_> NH1/2 swapped:      179.8072     1861   314\n",
      "iupac_m_487_> NH1/2 swapped:      179.7521     1875   328\n",
      "iupac_m_487_> NH1/2 swapped:      179.8587     1876   400\n",
      "openf___224_> Open           muscle.BL00040001.pdb\n",
      "wrpdb___568_> Residues, atoms, selected atoms:     1935    15763    15763\n",
      "openf___224_> Open           muscle.IL00000001.pdb\n",
      "\n",
      "Dynamically allocated memory at       amaxmodel [B,KiB,MiB]:     45612318   44543.281    43.499\n",
      "read_mo_297_> Segments, residues, atoms:        5     1935    15763\n",
      "read_mo_298_> Segment:        1     1 A   469 A     3168\n",
      "read_mo_298_> Segment:        2     1 B   459 B     3150\n",
      "read_mo_298_> Segment:        3     4 C   459 C     3127\n",
      "read_mo_298_> Segment:        4     1 D   469 D     3168\n",
      "read_mo_298_> Segment:        5     1 E   459 E     3150\n",
      "randomi_498_> Atoms,selected atoms,random_seed,amplitude:    15763       86        1        5.0000\n",
      "randomi_496_> Amplitude is > 0; randomization is done.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "\n",
      "\n",
      ">> ENERGY; Differences between the model's features and restraints:\n",
      "Number of all residues in MODEL                   :     1935\n",
      "Number of all, selected real atoms                :    15763      86\n",
      "Number of all, selected pseudo atoms              :        0       0\n",
      "Number of all static, selected restraints         :      385     385\n",
      "COVALENT_CYS                                      :        F\n",
      "NONBONDED_SEL_ATOMS                               :        1\n",
      "Number of non-bonded pairs (excluding 1-2,1-3,1-4):     2793\n",
      "Dynamic pairs routine                             : 5, NATM_SEL x NATM double loop\n",
      "Atomic shift for contacts update (UPDATE_DYNAMIC) :    0.390\n",
      "LENNARD_JONES_SWITCH                              :    6.500   7.500\n",
      "COULOMB_JONES_SWITCH                              :    6.500   7.500\n",
      "RESIDUE_SPAN_RANGE                                :        0   99999\n",
      "NLOGN_USE                                         :       15\n",
      "CONTACT_SHELL                                     :    7.000\n",
      "DYNAMIC_PAIRS,_SPHERE,_COULOMB,_LENNARD,_MODELLER :        T       T       F       F       T\n",
      "SPHERE_STDV                                       :    0.050\n",
      "RADII_FACTOR                                      :    0.820\n",
      "Current energy                                    :         389.5705\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Summary of the restraint violations: \n",
      "\n",
      "   NUM     ... number of restraints.\n",
      "   NUMVI   ... number of restraints with RVIOL > VIOL_REPORT_CUT[i].\n",
      "   RVIOL   ... relative difference from the best value.\n",
      "   NUMVP   ... number of restraints with -Ln(pdf) > VIOL_REPORT_CUT2[i].\n",
      "   RMS_1   ... RMS(feature, minimally_violated_basis_restraint, NUMB).\n",
      "   RMS_2   ... RMS(feature, best_value, NUMB).\n",
      "   MOL.PDF ... scaled contribution to -Ln(Molecular pdf).\n",
      "\n",
      " #                     RESTRAINT_GROUP      NUM   NUMVI  NUMVP   RMS_1   RMS_2         MOL.PDF     S_i\n",
      "------------------------------------------------------------------------------------------------------\n",
      " 1 Bond length potential              :      90       1      3   0.030   0.030      22.606       1.000\n",
      " 2 Bond angle potential               :     128       4     10   6.269   6.269      95.309       1.000\n",
      " 3 Stereochemical cosine torsion poten:      81       0      2  41.325  41.325      22.578       1.000\n",
      " 4 Stereochemical improper torsion pot:      40       1      1   4.814   4.814      16.531       1.000\n",
      " 5 Soft-sphere overlap restraints     :    2793       2      5   0.014   0.014      60.821       1.000\n",
      " 6 Lennard-Jones 6-12 potential       :       0       0      0   0.000   0.000      0.0000       1.000\n",
      " 7 Coulomb point-point electrostatic p:       0       0      0   0.000   0.000      0.0000       1.000\n",
      " 8 H-bonding potential                :       0       0      0   0.000   0.000      0.0000       1.000\n",
      " 9 Distance restraints 1 (CA-CA)      :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "10 Distance restraints 2 (N-O)        :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "11 Mainchain Phi dihedral restraints  :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "12 Mainchain Psi dihedral restraints  :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "13 Mainchain Omega dihedral restraints:      12       2      3  14.028  14.028      27.871       1.000\n",
      "14 Sidechain Chi_1 dihedral restraints:      11       0      1  81.845  81.845      13.226       1.000\n",
      "15 Sidechain Chi_2 dihedral restraints:       7       0      2  99.238  99.238      7.0131       1.000\n",
      "16 Sidechain Chi_3 dihedral restraints:       3       0      0  36.425  36.425     0.45249       1.000\n",
      "17 Sidechain Chi_4 dihedral restraints:       0       0      0   0.000   0.000      0.0000       1.000\n",
      "18 Disulfide distance restraints      :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "19 Disulfide angle restraints         :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "20 Disulfide dihedral angle restraints:       0       0      0   0.000   0.000      0.0000       1.000\n",
      "21 Lower bound distance restraints    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "22 Upper bound distance restraints    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "23 Distance restraints 3 (SDCH-MNCH)  :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "24 Sidechain Chi_5 dihedral restraints:       0       0      0   0.000   0.000      0.0000       1.000\n",
      "25 Phi/Psi pair of dihedral restraints:      13       7      6  52.520 114.426      45.517       1.000\n",
      "26 Distance restraints 4 (SDCH-SDCH)  :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "27 Distance restraints 5 (X-Y)        :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "28 NMR distance restraints 6 (X-Y)    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "29 NMR distance restraints 7 (X-Y)    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "30 Minimal distance restraints        :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "31 Non-bonded restraints              :    2793       0     20   1.886   1.886      77.646       1.000\n",
      "32 Atomic accessibility restraints    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "33 Atomic density restraints          :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "34 Absolute position restraints       :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "35 Dihedral angle difference restraint:       0       0      0   0.000   0.000      0.0000       1.000\n",
      "36 GBSA implicit solvent potential    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "37 EM density fitting potential       :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "38 SAXS restraints                    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "39 Symmetry restraints                :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "\n",
      "\n",
      "\n",
      "List of the violated restraints:\n",
      "   A restraint is violated when the relative difference\n",
      "   from the best value (RVIOL) is larger than CUTOFF.\n",
      "\n",
      "   ICSR   ... index of a restraint in the current set.\n",
      "   RESNO  ... residue numbers of the first two atoms.\n",
      "   ATM    ... IUPAC atom names of the first two atoms.\n",
      "   FEAT   ... the value of the feature in the model.\n",
      "   restr  ... the mean of the basis restraint with the smallest\n",
      "              difference from the model (local minimum).\n",
      "   viol   ... difference from the local minimum.\n",
      "   rviol  ... relative difference from the local minimum.\n",
      "   RESTR  ... the best value (global minimum).\n",
      "   VIOL   ... difference from the best value.\n",
      "   RVIOL  ... relative difference from the best value.\n",
      "\n",
      "\n",
      "-------------------------------------------------------------------------------------------------\n",
      "\n",
      "Feature  1                           : Bond length potential                   \n",
      "List of the RVIOL violations larger than   :       4.5000\n",
      "\n",
      "    #   ICSR  RESNO1/2 ATM1/2   INDATM1/2    FEAT   restr    viol   rviol   RESTR    VIOL   RVIOL\n",
      "    1     47 967D 967D C   CA   7887 7882    1.68    1.49    0.19    5.56    1.49    0.19    5.56\n",
      "\n",
      "-------------------------------------------------------------------------------------------------\n",
      "\n",
      "Feature  2                           : Bond angle potential                    \n",
      "List of the RVIOL violations larger than   :       4.5000\n",
      "\n",
      "    #   ICSR  RESNO1/2 ATM1/2   INDATM1/2    FEAT   restr    viol   rviol   RESTR    VIOL   RVIOL\n",
      "    1    147 966Q 967D C   N    7879 7881  141.29  120.00   21.29    4.84  120.00   21.29    4.84\n",
      "    2    150 967D 967D N   CA   7881 7882  129.94  107.00   22.94    6.59  107.00   22.94    6.59\n",
      "    3    158 967D 968P C   N    7887 7889  135.60  117.00   18.60    4.63  117.00   18.60    4.63\n",
      "    4    161 968P 968P N   CA   7889 7890  133.84  108.20   25.64    5.83  108.20   25.64    5.83\n",
      "\n",
      "-------------------------------------------------------------------------------------------------\n",
      "\n",
      "Feature 13                           : Mainchain Omega dihedral restraints     \n",
      "List of the RVIOL violations larger than   :       4.5000\n",
      "\n",
      "    #   ICSR  RESNO1/2 ATM1/2   INDATM1/2    FEAT   restr    viol   rviol   RESTR    VIOL   RVIOL\n",
      "    1    358 966Q 966Q CA  C    7873 7879 -155.34  180.00   24.66    4.93  180.00   24.66    4.93\n",
      "    2    359 967D 967D CA  C    7882 7887 -148.44  180.00   31.56    6.31  180.00   31.56    6.31\n",
      "\n",
      "-------------------------------------------------------------------------------------------------\n",
      "\n",
      "Feature 25                           : Phi/Psi pair of dihedral restraints     \n",
      "List of the RVIOL violations larger than   :       6.5000\n",
      "\n",
      "    #   ICSR  RESNO1/2 ATM1/2   INDATM1/2    FEAT   restr    viol   rviol   RESTR    VIOL   RVIOL\n",
      "    1    340 960R 961R C   N    7830 7832 -101.95 -125.20   23.55    0.85  -63.00 -177.81   22.26\n",
      "    1        961R 961R N   CA   7832 7833  136.88  140.60                  -41.10\n",
      "    2    343 963V 964N C   N    7855 7857  -63.88  -71.20   24.23    1.75  -63.20  153.00   19.16\n",
      "    2        964N 964N N   CA   7857 7858  165.90  142.80                  -41.10\n",
      "    3    344 964N 965P C   N    7863 7865  -57.14  -58.70    6.93    0.70  -64.50  171.11   12.39\n",
      "    3        965P 965P N   CA   7865 7866  -23.75  -30.50                  147.20\n",
      "    4    345 965P 966Q C   N    7870 7872  -97.73 -121.10   27.58    0.85  -63.80  168.79   23.17\n",
      "    4        966Q 966Q N   CA   7872 7873  125.04  139.70                  -40.30\n",
      "    5    346 966Q 967D C   N    7879 7881  -56.38  -70.90   39.84    1.38  -63.30  153.36   19.36\n",
      "    5        967D 967D N   CA   7881 7882  113.21  150.30                  -40.00\n",
      "    6    348 968P 969S C   N    7894 7896 -131.96  -64.10   98.42   12.00  -64.10   98.42   12.00\n",
      "    6        969S 969S N   CA   7896 7897 -106.28  -35.00                  -35.00\n",
      "    7    351 971V 972D C   N    7919 7921   -7.39  -63.30   83.72    9.28  -63.30   83.72    9.28\n",
      "    7        972D 972D N   CA   7921 7922 -102.32  -40.00                  -40.00\n",
      "\n",
      "\n",
      "report______> Distribution of short non-bonded contacts:\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "  serious non-bonded atom clash:  1055 1178   2.040\n",
      "  serious non-bonded atom clash:  1976 5131   1.952\n",
      "  serious non-bonded atom clash:  4241 4364   2.034\n",
      "  serious non-bonded atom clash:  7368 7491   2.036\n",
      "  serious non-bonded atom clash:  7778 8016   2.297\n",
      "  serious non-bonded atom clash: 1050010623   2.039\n",
      "  serious non-bonded atom clash: 1103111037   2.051\n",
      "  serious non-bonded atom clash: 1142114576   2.100\n",
      "  serious non-bonded atom clash: 1368613809   2.037\n",
      "  serious non-bonded atom clash: 1414614278   2.265\n",
      "\n",
      "DISTANCE1:  0.00 2.10 2.20 2.30 2.40 2.50 2.60 2.70 2.80 2.90 3.00 3.10 3.20 3.30 3.40\n",
      "DISTANCE2:  2.10 2.20 2.30 2.40 2.50 2.60 2.70 2.80 2.90 3.00 3.10 3.20 3.30 3.40 3.50\n",
      "FREQUENCY:     7    1    2   14   13   38   77   93  260  695  996  914 1801 2419 2317\n",
      "\n",
      "\n",
      "<< end of ENERGY.\n",
      "iupac_m_487_> NH1/2 swapped:     -179.8813       10    10\n",
      "iupac_m_487_> NH1/2 swapped:      179.9234       20    20\n",
      "iupac_m_487_> NH1/2 swapped:      179.8847      207   207\n",
      "iupac_m_487_> NH1/2 swapped:     -179.7626      327   327\n",
      "iupac_m_487_> NH1/2 swapped:      180.0000      405    18\n",
      "iupac_m_487_> NH1/2 swapped:     -179.7577      411    24\n",
      "iupac_m_487_> NH1/2 swapped:      179.8703      462    75\n",
      "iupac_m_487_> NH1/2 swapped:      179.7842      470    83\n",
      "iupac_m_487_> NH1/2 swapped:     -179.7762      475    88\n",
      "iupac_m_487_> NH1/2 swapped:      179.8481      502   115\n",
      "iupac_m_487_> NH1/2 swapped:     -179.9093      701   314\n",
      "iupac_m_487_> NH1/2 swapped:     -179.7168      715   328\n",
      "iupac_m_487_> NH1/2 swapped:      179.8615      716   400\n",
      "iupac_m_487_> NH1/2 swapped:     -179.9657      840    68\n",
      "iupac_m_487_> NH1/2 swapped:     -179.9209      847    75\n",
      "iupac_m_487_> NH1/2 swapped:      179.7175      860    88\n",
      "iupac_m_487_> NH1/2 swapped:     -180.0000      887   115\n",
      "iupac_m_487_> NH1/2 swapped:     -178.9152      923   151\n",
      "iupac_m_487_> NH1/2 swapped:      179.6326      960   188\n",
      "iupac_m_485_> OD1/2 will be swapped:      -92.0115      967   195\n",
      "iupac_m_484W> Dihedral still outside +-90:       92.1411\n",
      "iupac_m_485_> OD1/2 will be swapped:      167.5403      972   200\n",
      "iupac_m_487_> NH1/2 swapped:      179.9558     1086   314\n",
      "iupac_m_487_> NH1/2 swapped:     -179.5980     1100   328\n",
      "iupac_m_487_> NH1/2 swapped:     -179.6961     1170    10\n",
      "iupac_m_487_> NH1/2 swapped:     -179.8275     1180    20\n",
      "iupac_m_486_> OE1/2 will be swapped:     -109.6868     1332   172\n",
      "iupac_m_487_> NH1/2 swapped:      179.9477     1367   207\n",
      "iupac_m_487_> NH1/2 swapped:      179.9115     1487   327\n",
      "iupac_m_487_> NH1/2 swapped:     -179.5082     1571    24\n",
      "iupac_m_487_> NH1/2 swapped:      179.9720     1622    75\n",
      "iupac_m_487_> NH1/2 swapped:      180.0000     1630    83\n",
      "iupac_m_487_> NH1/2 swapped:     -179.8864     1635    88\n",
      "iupac_m_487_> NH1/2 swapped:      179.5804     1662   115\n",
      "iupac_m_487_> NH1/2 swapped:      179.8072     1861   314\n",
      "iupac_m_487_> NH1/2 swapped:      179.7521     1875   328\n",
      "iupac_m_487_> NH1/2 swapped:      179.8587     1876   400\n",
      "openf___224_> Open           muscle.BL00050001.pdb\n",
      "wrpdb___568_> Residues, atoms, selected atoms:     1935    15763    15763\n",
      "openf___224_> Open           muscle.IL00000001.pdb\n",
      "\n",
      "Dynamically allocated memory at       amaxmodel [B,KiB,MiB]:     45612318   44543.281    43.499\n",
      "read_mo_297_> Segments, residues, atoms:        5     1935    15763\n",
      "read_mo_298_> Segment:        1     1 A   469 A     3168\n",
      "read_mo_298_> Segment:        2     1 B   459 B     3150\n",
      "read_mo_298_> Segment:        3     4 C   459 C     3127\n",
      "read_mo_298_> Segment:        4     1 D   469 D     3168\n",
      "read_mo_298_> Segment:        5     1 E   459 E     3150\n",
      "randomi_498_> Atoms,selected atoms,random_seed,amplitude:    15763       86        1        5.0000\n",
      "randomi_496_> Amplitude is > 0; randomization is done.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "\n",
      "\n",
      ">> ENERGY; Differences between the model's features and restraints:\n",
      "Number of all residues in MODEL                   :     1935\n",
      "Number of all, selected real atoms                :    15763      86\n",
      "Number of all, selected pseudo atoms              :        0       0\n",
      "Number of all static, selected restraints         :      385     385\n",
      "COVALENT_CYS                                      :        F\n",
      "NONBONDED_SEL_ATOMS                               :        1\n",
      "Number of non-bonded pairs (excluding 1-2,1-3,1-4):     2035\n",
      "Dynamic pairs routine                             : 5, NATM_SEL x NATM double loop\n",
      "Atomic shift for contacts update (UPDATE_DYNAMIC) :    0.390\n",
      "LENNARD_JONES_SWITCH                              :    6.500   7.500\n",
      "COULOMB_JONES_SWITCH                              :    6.500   7.500\n",
      "RESIDUE_SPAN_RANGE                                :        0   99999\n",
      "NLOGN_USE                                         :       15\n",
      "CONTACT_SHELL                                     :    7.000\n",
      "DYNAMIC_PAIRS,_SPHERE,_COULOMB,_LENNARD,_MODELLER :        T       T       F       F       T\n",
      "SPHERE_STDV                                       :    0.050\n",
      "RADII_FACTOR                                      :    0.820\n",
      "Current energy                                    :          70.8709\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Summary of the restraint violations: \n",
      "\n",
      "   NUM     ... number of restraints.\n",
      "   NUMVI   ... number of restraints with RVIOL > VIOL_REPORT_CUT[i].\n",
      "   RVIOL   ... relative difference from the best value.\n",
      "   NUMVP   ... number of restraints with -Ln(pdf) > VIOL_REPORT_CUT2[i].\n",
      "   RMS_1   ... RMS(feature, minimally_violated_basis_restraint, NUMB).\n",
      "   RMS_2   ... RMS(feature, best_value, NUMB).\n",
      "   MOL.PDF ... scaled contribution to -Ln(Molecular pdf).\n",
      "\n",
      " #                     RESTRAINT_GROUP      NUM   NUMVI  NUMVP   RMS_1   RMS_2         MOL.PDF     S_i\n",
      "------------------------------------------------------------------------------------------------------\n",
      " 1 Bond length potential              :      90       0      0   0.010   0.010      2.6948       1.000\n",
      " 2 Bond angle potential               :     128       0      0   3.314   3.314      25.233       1.000\n",
      " 3 Stereochemical cosine torsion poten:      81       0      2  44.681  44.681      25.969       1.000\n",
      " 4 Stereochemical improper torsion pot:      40       0      1   2.243   2.243      5.7413       1.000\n",
      " 5 Soft-sphere overlap restraints     :    2035       0      0   0.001   0.001     0.69261E-01   1.000\n",
      " 6 Lennard-Jones 6-12 potential       :       0       0      0   0.000   0.000      0.0000       1.000\n",
      " 7 Coulomb point-point electrostatic p:       0       0      0   0.000   0.000      0.0000       1.000\n",
      " 8 H-bonding potential                :       0       0      0   0.000   0.000      0.0000       1.000\n",
      " 9 Distance restraints 1 (CA-CA)      :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "10 Distance restraints 2 (N-O)        :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "11 Mainchain Phi dihedral restraints  :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "12 Mainchain Psi dihedral restraints  :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "13 Mainchain Omega dihedral restraints:      12       0      0   4.516   4.516      2.8877       1.000\n",
      "14 Sidechain Chi_1 dihedral restraints:      11       0      0  70.172  70.172      3.4414       1.000\n",
      "15 Sidechain Chi_2 dihedral restraints:       7       0      0 110.697 110.697      5.1776       1.000\n",
      "16 Sidechain Chi_3 dihedral restraints:       3       0      0  94.171  94.171      1.3062       1.000\n",
      "17 Sidechain Chi_4 dihedral restraints:       0       0      0   0.000   0.000      0.0000       1.000\n",
      "18 Disulfide distance restraints      :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "19 Disulfide angle restraints         :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "20 Disulfide dihedral angle restraints:       0       0      0   0.000   0.000      0.0000       1.000\n",
      "21 Lower bound distance restraints    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "22 Upper bound distance restraints    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "23 Distance restraints 3 (SDCH-MNCH)  :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "24 Sidechain Chi_5 dihedral restraints:       0       0      0   0.000   0.000      0.0000       1.000\n",
      "25 Phi/Psi pair of dihedral restraints:      13       7      3  39.160 113.596      27.447       1.000\n",
      "26 Distance restraints 4 (SDCH-SDCH)  :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "27 Distance restraints 5 (X-Y)        :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "28 NMR distance restraints 6 (X-Y)    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "29 NMR distance restraints 7 (X-Y)    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "30 Minimal distance restraints        :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "31 Non-bonded restraints              :    2035       0      4   1.908   1.908     -29.096       1.000\n",
      "32 Atomic accessibility restraints    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "33 Atomic density restraints          :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "34 Absolute position restraints       :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "35 Dihedral angle difference restraint:       0       0      0   0.000   0.000      0.0000       1.000\n",
      "36 GBSA implicit solvent potential    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "37 EM density fitting potential       :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "38 SAXS restraints                    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "39 Symmetry restraints                :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "\n",
      "\n",
      "\n",
      "List of the violated restraints:\n",
      "   A restraint is violated when the relative difference\n",
      "   from the best value (RVIOL) is larger than CUTOFF.\n",
      "\n",
      "   ICSR   ... index of a restraint in the current set.\n",
      "   RESNO  ... residue numbers of the first two atoms.\n",
      "   ATM    ... IUPAC atom names of the first two atoms.\n",
      "   FEAT   ... the value of the feature in the model.\n",
      "   restr  ... the mean of the basis restraint with the smallest\n",
      "              difference from the model (local minimum).\n",
      "   viol   ... difference from the local minimum.\n",
      "   rviol  ... relative difference from the local minimum.\n",
      "   RESTR  ... the best value (global minimum).\n",
      "   VIOL   ... difference from the best value.\n",
      "   RVIOL  ... relative difference from the best value.\n",
      "\n",
      "\n",
      "-------------------------------------------------------------------------------------------------\n",
      "\n",
      "Feature 25                           : Phi/Psi pair of dihedral restraints     \n",
      "List of the RVIOL violations larger than   :       6.5000\n",
      "\n",
      "    #   ICSR  RESNO1/2 ATM1/2   INDATM1/2    FEAT   restr    viol   rviol   RESTR    VIOL   RVIOL\n",
      "    1    340 960R 961R C   N    7830 7832 -101.95 -125.20   25.91    1.32  -63.00  171.35   25.85\n",
      "    1        961R 961R N   CA   7832 7833  152.04  140.60                  -41.10\n",
      "    2    342 962T 963V C   N    7848 7850  -60.18  -62.40   16.88    2.27 -125.40 -178.88    7.09\n",
      "    2        963V 963V N   CA   7850 7851  -25.67  -42.40                  143.30\n",
      "    3    343 963V 964N C   N    7855 7857  -59.22  -71.20   15.49    1.25  -63.20  166.34   20.58\n",
      "    3        964N 964N N   CA   7857 7858  152.61  142.80                  -41.10\n",
      "    4    345 965P 966Q C   N    7870 7872 -155.70 -121.10   48.95    1.59  -63.80  171.98   29.95\n",
      "    4        966Q 966Q N   CA   7872 7873  174.33  139.70                  -40.30\n",
      "    5    346 966Q 967D C   N    7879 7881 -125.20  -63.30   73.84    9.17  -63.30   73.84    9.17\n",
      "    5        967D 967D N   CA   7881 7882    0.25  -40.00                  -40.00\n",
      "    6    349 969S 970Y C   N    7900 7902 -115.17 -124.30   12.70    0.57  -63.50  177.66   25.78\n",
      "    6        970Y 970Y N   CA   7902 7903  126.58  135.40                  -43.40\n",
      "    7    351 971V 972D C   N    7919 7921 -108.66  -63.30   77.78   12.47  -63.30   77.78   12.47\n",
      "    7        972D 972D N   CA   7921 7922 -103.19  -40.00                  -40.00\n",
      "\n",
      "\n",
      "report______> Distribution of short non-bonded contacts:\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "  serious non-bonded atom clash:  1055 1178   2.040\n",
      "  serious non-bonded atom clash:  1976 5131   1.952\n",
      "  serious non-bonded atom clash:  4241 4364   2.034\n",
      "  serious non-bonded atom clash:  7368 7491   2.036\n",
      "  serious non-bonded atom clash:  7778 8016   2.297\n",
      "  serious non-bonded atom clash: 1050010623   2.039\n",
      "  serious non-bonded atom clash: 1103111037   2.051\n",
      "  serious non-bonded atom clash: 1142114576   2.100\n",
      "  serious non-bonded atom clash: 1368613809   2.037\n",
      "  serious non-bonded atom clash: 1414614278   2.265\n",
      "\n",
      "DISTANCE1:  0.00 2.10 2.20 2.30 2.40 2.50 2.60 2.70 2.80 2.90 3.00 3.10 3.20 3.30 3.40\n",
      "DISTANCE2:  2.10 2.20 2.30 2.40 2.50 2.60 2.70 2.80 2.90 3.00 3.10 3.20 3.30 3.40 3.50\n",
      "FREQUENCY:     7    1    2   13   11   36   76   86  254  688  990  900 1798 2405 2301\n",
      "\n",
      "\n",
      "<< end of ENERGY.\n",
      "iupac_m_487_> NH1/2 swapped:     -179.8813       10    10\n",
      "iupac_m_487_> NH1/2 swapped:      179.9234       20    20\n",
      "iupac_m_487_> NH1/2 swapped:      179.8847      207   207\n",
      "iupac_m_487_> NH1/2 swapped:     -179.7626      327   327\n",
      "iupac_m_487_> NH1/2 swapped:      180.0000      405    18\n",
      "iupac_m_487_> NH1/2 swapped:     -179.7577      411    24\n",
      "iupac_m_487_> NH1/2 swapped:      179.8703      462    75\n",
      "iupac_m_487_> NH1/2 swapped:      179.7842      470    83\n",
      "iupac_m_487_> NH1/2 swapped:     -179.7762      475    88\n",
      "iupac_m_487_> NH1/2 swapped:      179.8481      502   115\n",
      "iupac_m_487_> NH1/2 swapped:     -179.9093      701   314\n",
      "iupac_m_487_> NH1/2 swapped:     -179.7168      715   328\n",
      "iupac_m_487_> NH1/2 swapped:      179.8615      716   400\n",
      "iupac_m_487_> NH1/2 swapped:     -179.9657      840    68\n",
      "iupac_m_487_> NH1/2 swapped:     -179.9209      847    75\n",
      "iupac_m_487_> NH1/2 swapped:      179.7175      860    88\n",
      "iupac_m_487_> NH1/2 swapped:     -180.0000      887   115\n",
      "iupac_m_487_> NH1/2 swapped:     -178.9152      923   151\n",
      "iupac_m_487_> NH1/2 swapped:      179.6326      960   188\n",
      "iupac_m_485_> OD1/2 will be swapped:      126.6878      967   195\n",
      "iupac_m_483_> CD1/2 CE1/2 swapped:      -96.9521      970   198\n",
      "iupac_m_397W> Atoms were not swapped because of the uncertainty of how to handle the H atom.\n",
      "iupac_m_485_> OD1/2 will be swapped:     -159.4678      972   200\n",
      "iupac_m_487_> NH1/2 swapped:      179.9558     1086   314\n",
      "iupac_m_487_> NH1/2 swapped:     -179.5980     1100   328\n",
      "iupac_m_487_> NH1/2 swapped:     -179.6961     1170    10\n",
      "iupac_m_487_> NH1/2 swapped:     -179.8275     1180    20\n",
      "iupac_m_486_> OE1/2 will be swapped:     -109.6868     1332   172\n",
      "iupac_m_487_> NH1/2 swapped:      179.9477     1367   207\n",
      "iupac_m_487_> NH1/2 swapped:      179.9115     1487   327\n",
      "iupac_m_487_> NH1/2 swapped:     -179.5082     1571    24\n",
      "iupac_m_487_> NH1/2 swapped:      179.9720     1622    75\n",
      "iupac_m_487_> NH1/2 swapped:      180.0000     1630    83\n",
      "iupac_m_487_> NH1/2 swapped:     -179.8864     1635    88\n",
      "iupac_m_487_> NH1/2 swapped:      179.5804     1662   115\n",
      "iupac_m_487_> NH1/2 swapped:      179.8072     1861   314\n",
      "iupac_m_487_> NH1/2 swapped:      179.7521     1875   328\n",
      "iupac_m_487_> NH1/2 swapped:      179.8587     1876   400\n",
      "openf___224_> Open           muscle.BL00060001.pdb\n",
      "wrpdb___568_> Residues, atoms, selected atoms:     1935    15763    15763\n",
      "openf___224_> Open           muscle.IL00000001.pdb\n",
      "\n",
      "Dynamically allocated memory at       amaxmodel [B,KiB,MiB]:     45612318   44543.281    43.499\n",
      "read_mo_297_> Segments, residues, atoms:        5     1935    15763\n",
      "read_mo_298_> Segment:        1     1 A   469 A     3168\n",
      "read_mo_298_> Segment:        2     1 B   459 B     3150\n",
      "read_mo_298_> Segment:        3     4 C   459 C     3127\n",
      "read_mo_298_> Segment:        4     1 D   469 D     3168\n",
      "read_mo_298_> Segment:        5     1 E   459 E     3150\n",
      "randomi_498_> Atoms,selected atoms,random_seed,amplitude:    15763       86        1        5.0000\n",
      "randomi_496_> Amplitude is > 0; randomization is done.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "\n",
      "\n",
      ">> ENERGY; Differences between the model's features and restraints:\n",
      "Number of all residues in MODEL                   :     1935\n",
      "Number of all, selected real atoms                :    15763      86\n",
      "Number of all, selected pseudo atoms              :        0       0\n",
      "Number of all static, selected restraints         :      385     385\n",
      "COVALENT_CYS                                      :        F\n",
      "NONBONDED_SEL_ATOMS                               :        1\n",
      "Number of non-bonded pairs (excluding 1-2,1-3,1-4):     2891\n",
      "Dynamic pairs routine                             : 5, NATM_SEL x NATM double loop\n",
      "Atomic shift for contacts update (UPDATE_DYNAMIC) :    0.390\n",
      "LENNARD_JONES_SWITCH                              :    6.500   7.500\n",
      "COULOMB_JONES_SWITCH                              :    6.500   7.500\n",
      "RESIDUE_SPAN_RANGE                                :        0   99999\n",
      "NLOGN_USE                                         :       15\n",
      "CONTACT_SHELL                                     :    7.000\n",
      "DYNAMIC_PAIRS,_SPHERE,_COULOMB,_LENNARD,_MODELLER :        T       T       F       F       T\n",
      "SPHERE_STDV                                       :    0.050\n",
      "RADII_FACTOR                                      :    0.820\n",
      "Current energy                                    :        1874.2494\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Summary of the restraint violations: \n",
      "\n",
      "   NUM     ... number of restraints.\n",
      "   NUMVI   ... number of restraints with RVIOL > VIOL_REPORT_CUT[i].\n",
      "   RVIOL   ... relative difference from the best value.\n",
      "   NUMVP   ... number of restraints with -Ln(pdf) > VIOL_REPORT_CUT2[i].\n",
      "   RMS_1   ... RMS(feature, minimally_violated_basis_restraint, NUMB).\n",
      "   RMS_2   ... RMS(feature, best_value, NUMB).\n",
      "   MOL.PDF ... scaled contribution to -Ln(Molecular pdf).\n",
      "\n",
      " #                     RESTRAINT_GROUP      NUM   NUMVI  NUMVP   RMS_1   RMS_2         MOL.PDF     S_i\n",
      "------------------------------------------------------------------------------------------------------\n",
      " 1 Bond length potential              :      90       5      5   0.127   0.127      363.11       1.000\n",
      " 2 Bond angle potential               :     128      11     19   8.740   8.740      167.67       1.000\n",
      " 3 Stereochemical cosine torsion poten:      81       0      3  48.117  48.117      36.807       1.000\n",
      " 4 Stereochemical improper torsion pot:      40       4      9   9.508   9.508      85.960       1.000\n",
      " 5 Soft-sphere overlap restraints     :    2891      27     35   0.047   0.047      765.54       1.000\n",
      " 6 Lennard-Jones 6-12 potential       :       0       0      0   0.000   0.000      0.0000       1.000\n",
      " 7 Coulomb point-point electrostatic p:       0       0      0   0.000   0.000      0.0000       1.000\n",
      " 8 H-bonding potential                :       0       0      0   0.000   0.000      0.0000       1.000\n",
      " 9 Distance restraints 1 (CA-CA)      :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "10 Distance restraints 2 (N-O)        :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "11 Mainchain Phi dihedral restraints  :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "12 Mainchain Psi dihedral restraints  :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "13 Mainchain Omega dihedral restraints:      12       1      2  14.644  14.644      30.369       1.000\n",
      "14 Sidechain Chi_1 dihedral restraints:      11       0      1  80.587  80.587      7.6351       1.000\n",
      "15 Sidechain Chi_2 dihedral restraints:       7       0      2  86.785  86.785      15.806       1.000\n",
      "16 Sidechain Chi_3 dihedral restraints:       3       0      0  85.882  85.882      1.9592       1.000\n",
      "17 Sidechain Chi_4 dihedral restraints:       0       0      0   0.000   0.000      0.0000       1.000\n",
      "18 Disulfide distance restraints      :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "19 Disulfide angle restraints         :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "20 Disulfide dihedral angle restraints:       0       0      0   0.000   0.000      0.0000       1.000\n",
      "21 Lower bound distance restraints    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "22 Upper bound distance restraints    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "23 Distance restraints 3 (SDCH-MNCH)  :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "24 Sidechain Chi_5 dihedral restraints:       0       0      0   0.000   0.000      0.0000       1.000\n",
      "25 Phi/Psi pair of dihedral restraints:      13       9      4  59.110 139.576      43.301       1.000\n",
      "26 Distance restraints 4 (SDCH-SDCH)  :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "27 Distance restraints 5 (X-Y)        :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "28 NMR distance restraints 6 (X-Y)    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "29 NMR distance restraints 7 (X-Y)    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "30 Minimal distance restraints        :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "31 Non-bonded restraints              :    2891       0     84   1.912   1.912      356.09       1.000\n",
      "32 Atomic accessibility restraints    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "33 Atomic density restraints          :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "34 Absolute position restraints       :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "35 Dihedral angle difference restraint:       0       0      0   0.000   0.000      0.0000       1.000\n",
      "36 GBSA implicit solvent potential    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "37 EM density fitting potential       :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "38 SAXS restraints                    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "39 Symmetry restraints                :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "\n",
      "\n",
      "\n",
      "List of the violated restraints:\n",
      "   A restraint is violated when the relative difference\n",
      "   from the best value (RVIOL) is larger than CUTOFF.\n",
      "\n",
      "   ICSR   ... index of a restraint in the current set.\n",
      "   RESNO  ... residue numbers of the first two atoms.\n",
      "   ATM    ... IUPAC atom names of the first two atoms.\n",
      "   FEAT   ... the value of the feature in the model.\n",
      "   restr  ... the mean of the basis restraint with the smallest\n",
      "              difference from the model (local minimum).\n",
      "   viol   ... difference from the local minimum.\n",
      "   rviol  ... relative difference from the local minimum.\n",
      "   RESTR  ... the best value (global minimum).\n",
      "   VIOL   ... difference from the best value.\n",
      "   RVIOL  ... relative difference from the best value.\n",
      "\n",
      "\n",
      "-------------------------------------------------------------------------------------------------\n",
      "\n",
      "Feature  1                           : Bond length potential                   \n",
      "List of the RVIOL violations larger than   :       4.5000\n",
      "\n",
      "    #   ICSR  RESNO1/2 ATM1/2   INDATM1/2    FEAT   restr    viol   rviol   RESTR    VIOL   RVIOL\n",
      "    1     62 969S 970Y C   N    7900 7902    1.64    1.35    0.30   10.58    1.35    0.30   10.58\n",
      "    2     63 970Y 970Y CB  CA   7904 7903    2.41    1.54    0.87   24.02    1.54    0.87   24.02\n",
      "    3     64 970Y 970Y CG  CB   7905 7904    1.96    1.49    0.47   13.19    1.49    0.47   13.19\n",
      "    4     72 970Y 970Y N   CA   7902 7903    1.90    1.43    0.47   15.34    1.43    0.47   15.34\n",
      "    5     74 970Y 970Y C   CA   7912 7903    1.83    1.49    0.34    9.99    1.49    0.34    9.99\n",
      "\n",
      "-------------------------------------------------------------------------------------------------\n",
      "\n",
      "Feature  2                           : Bond angle potential                    \n",
      "List of the RVIOL violations larger than   :       4.5000\n",
      "\n",
      "    #   ICSR  RESNO1/2 ATM1/2   INDATM1/2    FEAT   restr    viol   rviol   RESTR    VIOL   RVIOL\n",
      "    1    176 969S 969S CA  C    7897 7900  132.43  116.50   15.93    4.58  116.50   15.93    4.58\n",
      "    2    178 969S 970Y C   N    7900 7902  150.07  120.00   30.07    6.83  120.00   30.07    6.83\n",
      "    3    180 970Y 970Y N   CA   7902 7903  131.32  113.50   17.82    4.79  113.50   17.82    4.79\n",
      "    4    182 970Y 970Y CA  CB   7903 7904  131.97  107.50   24.47    5.66  107.50   24.47    5.66\n",
      "    5    184 970Y 970Y CA  C    7903 7912  136.02  116.50   19.52    5.61  116.50   19.52    5.61\n",
      "    6    188 970Y 970Y CG  CD2  7905 7907  151.46  120.00   31.46    6.39  120.00   31.46    6.39\n",
      "    7    195 970Y 970Y C   CA   7912 7903  127.98  108.00   19.98    4.63  108.00   19.98    4.63\n",
      "\n",
      "-------------------------------------------------------------------------------------------------\n",
      "\n",
      "Feature  4                           : Stereochemical improper torsion potentia\n",
      "List of the RVIOL violations larger than   :       4.5000\n",
      "\n",
      "    #   ICSR  RESNO1/2 ATM1/2   INDATM1/2    FEAT   restr    viol   rviol   RESTR    VIOL   RVIOL\n",
      "    1    245 970Y 970Y CB  CG   7904 7905 -153.34  180.00   26.66    8.13  180.00   26.66    8.13\n",
      "    2    248 970Y 970Y CE1 CZ   7908 7910   18.50    0.00   18.50    5.64    0.00   18.50    5.64\n",
      "\n",
      "-------------------------------------------------------------------------------------------------\n",
      "\n",
      "Feature 13                           : Mainchain Omega dihedral restraints     \n",
      "List of the RVIOL violations larger than   :       4.5000\n",
      "\n",
      "    #   ICSR  RESNO1/2 ATM1/2   INDATM1/2    FEAT   restr    viol   rviol   RESTR    VIOL   RVIOL\n",
      "    1    361 969S 969S CA  C    7897 7900 -133.90  180.00   46.10    9.22  180.00   46.10    9.22\n",
      "\n",
      "-------------------------------------------------------------------------------------------------\n",
      "\n",
      "Feature 25                           : Phi/Psi pair of dihedral restraints     \n",
      "List of the RVIOL violations larger than   :       6.5000\n",
      "\n",
      "    #   ICSR  RESNO1/2 ATM1/2   INDATM1/2    FEAT   restr    viol   rviol   RESTR    VIOL   RVIOL\n",
      "    1    340 960R 961R C   N    7830 7832 -101.95 -125.20   23.74    0.83  -63.00 -178.85   22.12\n",
      "    1        961R 961R N   CA   7832 7833  135.82  140.60                  -41.10\n",
      "    2    341 961R 962T C   N    7841 7843  -73.65  -78.10   37.14    1.45  -63.20  155.38   19.25\n",
      "    2        962T 962T N   CA   7843 7844  112.93  149.80                  -42.10\n",
      "    3    342 962T 963V C   N    7848 7850  -77.65  -62.40   17.61    3.01 -125.40  172.25    9.51\n",
      "    3        963V 963V N   CA   7850 7851  -51.20  -42.40                  143.30\n",
      "    4    343 963V 964N C   N    7855 7857 -144.35 -119.90   49.98    1.65  -63.20  160.36   24.56\n",
      "    4        964N 964N N   CA   7857 7858 -179.41  137.00                  -41.10\n",
      "    5    345 965P 966Q C   N    7870 7872 -139.16 -121.10   71.85    3.08  -63.80  133.72   23.44\n",
      "    5        966Q 966Q N   CA   7872 7873 -150.76  139.70                  -40.30\n",
      "    6    346 966Q 967D C   N    7879 7881 -115.44  -96.50   23.15    0.95  -63.30  150.22   16.17\n",
      "    6        967D 967D N   CA   7881 7882  100.88  114.20                  -40.00\n",
      "    7    348 968P 969S C   N    7894 7896 -144.95 -136.60   33.61    1.84  -64.10  173.62   11.17\n",
      "    7        969S 969S N   CA   7896 7897  118.65  151.20                  -35.00\n",
      "    8    349 969S 970Y C   N    7900 7902   68.77 -124.30  168.62   11.88  -63.50 -154.41   28.39\n",
      "    8        970Y 970Y N   CA   7902 7903  159.21  135.40                  -43.40\n",
      "    9    350 970Y 971V C   N    7912 7914 -116.59  -62.40   55.97    9.39 -125.40  160.56    8.20\n",
      "    9        971V 971V N   CA   7914 7915  -56.38  -42.40                  143.30\n",
      "\n",
      "\n",
      "report______> Distribution of short non-bonded contacts:\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "  serious non-bonded atom clash:  1055 1178   2.040\n",
      "  serious non-bonded atom clash:  1976 5131   1.952\n",
      "  serious non-bonded atom clash:  4241 4364   2.034\n",
      "  serious non-bonded atom clash:  7368 7491   2.036\n",
      "  serious non-bonded atom clash:  7508 7903   2.183\n",
      "  serious non-bonded atom clash:  7508 7904   2.295\n",
      "  serious non-bonded atom clash:  7510 7903   2.160\n",
      "  serious non-bonded atom clash:  7512 7902   2.296\n",
      "  serious non-bonded atom clash:  7778 8016   2.297\n",
      "  serious non-bonded atom clash:  7903 7935   2.257\n",
      "  serious non-bonded atom clash:  7904 7935   1.847\n",
      "  serious non-bonded atom clash:  7905 7935   2.271\n",
      "  serious non-bonded atom clash: 1050010623   2.039\n",
      "  serious non-bonded atom clash: 1103111037   2.051\n",
      "  serious non-bonded atom clash: 1142114576   2.100\n",
      "  serious non-bonded atom clash: 1368613809   2.037\n",
      "  serious non-bonded atom clash: 1414614278   2.265\n",
      "\n",
      "DISTANCE1:  0.00 2.10 2.20 2.30 2.40 2.50 2.60 2.70 2.80 2.90 3.00 3.10 3.20 3.30 3.40\n",
      "DISTANCE2:  2.10 2.20 2.30 2.40 2.50 2.60 2.70 2.80 2.90 3.00 3.10 3.20 3.30 3.40 3.50\n",
      "FREQUENCY:     8    3    6   17   16   40   90   94  263  704  994  914 1808 2413 2319\n",
      "\n",
      "\n",
      "<< end of ENERGY.\n",
      "iupac_m_487_> NH1/2 swapped:     -179.8813       10    10\n",
      "iupac_m_487_> NH1/2 swapped:      179.9234       20    20\n",
      "iupac_m_487_> NH1/2 swapped:      179.8847      207   207\n",
      "iupac_m_487_> NH1/2 swapped:     -179.7626      327   327\n",
      "iupac_m_487_> NH1/2 swapped:      180.0000      405    18\n",
      "iupac_m_487_> NH1/2 swapped:     -179.7577      411    24\n",
      "iupac_m_487_> NH1/2 swapped:      179.8703      462    75\n",
      "iupac_m_487_> NH1/2 swapped:      179.7842      470    83\n",
      "iupac_m_487_> NH1/2 swapped:     -179.7762      475    88\n",
      "iupac_m_487_> NH1/2 swapped:      179.8481      502   115\n",
      "iupac_m_487_> NH1/2 swapped:     -179.9093      701   314\n",
      "iupac_m_487_> NH1/2 swapped:     -179.7168      715   328\n",
      "iupac_m_487_> NH1/2 swapped:      179.8615      716   400\n",
      "iupac_m_487_> NH1/2 swapped:     -179.9657      840    68\n",
      "iupac_m_487_> NH1/2 swapped:     -179.9209      847    75\n",
      "iupac_m_487_> NH1/2 swapped:      179.7175      860    88\n",
      "iupac_m_487_> NH1/2 swapped:     -180.0000      887   115\n",
      "iupac_m_487_> NH1/2 swapped:     -178.9152      923   151\n",
      "iupac_m_487_> NH1/2 swapped:      179.6326      960   188\n",
      "iupac_m_485_> OD1/2 will be swapped:     -142.6391      967   195\n",
      "iupac_m_483_> CD1/2 CE1/2 swapped:      168.2475      970   198\n",
      "iupac_m_397W> Atoms were not swapped because of the uncertainty of how to handle the H atom.\n",
      "iupac_m_487_> NH1/2 swapped:      179.9558     1086   314\n",
      "iupac_m_487_> NH1/2 swapped:     -179.5980     1100   328\n",
      "iupac_m_487_> NH1/2 swapped:     -179.6961     1170    10\n",
      "iupac_m_487_> NH1/2 swapped:     -179.8275     1180    20\n",
      "iupac_m_486_> OE1/2 will be swapped:     -109.6868     1332   172\n",
      "iupac_m_487_> NH1/2 swapped:      179.9477     1367   207\n",
      "iupac_m_487_> NH1/2 swapped:      179.9115     1487   327\n",
      "iupac_m_487_> NH1/2 swapped:     -179.5082     1571    24\n",
      "iupac_m_487_> NH1/2 swapped:      179.9720     1622    75\n",
      "iupac_m_487_> NH1/2 swapped:      180.0000     1630    83\n",
      "iupac_m_487_> NH1/2 swapped:     -179.8864     1635    88\n",
      "iupac_m_487_> NH1/2 swapped:      179.5804     1662   115\n",
      "iupac_m_487_> NH1/2 swapped:      179.8072     1861   314\n",
      "iupac_m_487_> NH1/2 swapped:      179.7521     1875   328\n",
      "iupac_m_487_> NH1/2 swapped:      179.8587     1876   400\n",
      "openf___224_> Open           muscle.BL00070001.pdb\n",
      "wrpdb___568_> Residues, atoms, selected atoms:     1935    15763    15763\n",
      "openf___224_> Open           muscle.IL00000001.pdb\n",
      "\n",
      "Dynamically allocated memory at       amaxmodel [B,KiB,MiB]:     45612318   44543.281    43.499\n",
      "read_mo_297_> Segments, residues, atoms:        5     1935    15763\n",
      "read_mo_298_> Segment:        1     1 A   469 A     3168\n",
      "read_mo_298_> Segment:        2     1 B   459 B     3150\n",
      "read_mo_298_> Segment:        3     4 C   459 C     3127\n",
      "read_mo_298_> Segment:        4     1 D   469 D     3168\n",
      "read_mo_298_> Segment:        5     1 E   459 E     3150\n",
      "randomi_498_> Atoms,selected atoms,random_seed,amplitude:    15763       86        1        5.0000\n",
      "randomi_496_> Amplitude is > 0; randomization is done.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "\n",
      "\n",
      ">> ENERGY; Differences between the model's features and restraints:\n",
      "Number of all residues in MODEL                   :     1935\n",
      "Number of all, selected real atoms                :    15763      86\n",
      "Number of all, selected pseudo atoms              :        0       0\n",
      "Number of all static, selected restraints         :      385     385\n",
      "COVALENT_CYS                                      :        F\n",
      "NONBONDED_SEL_ATOMS                               :        1\n",
      "Number of non-bonded pairs (excluding 1-2,1-3,1-4):     2601\n",
      "Dynamic pairs routine                             : 5, NATM_SEL x NATM double loop\n",
      "Atomic shift for contacts update (UPDATE_DYNAMIC) :    0.390\n",
      "LENNARD_JONES_SWITCH                              :    6.500   7.500\n",
      "COULOMB_JONES_SWITCH                              :    6.500   7.500\n",
      "RESIDUE_SPAN_RANGE                                :        0   99999\n",
      "NLOGN_USE                                         :       15\n",
      "CONTACT_SHELL                                     :    7.000\n",
      "DYNAMIC_PAIRS,_SPHERE,_COULOMB,_LENNARD,_MODELLER :        T       T       F       F       T\n",
      "SPHERE_STDV                                       :    0.050\n",
      "RADII_FACTOR                                      :    0.820\n",
      "Current energy                                    :         299.5407\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Summary of the restraint violations: \n",
      "\n",
      "   NUM     ... number of restraints.\n",
      "   NUMVI   ... number of restraints with RVIOL > VIOL_REPORT_CUT[i].\n",
      "   RVIOL   ... relative difference from the best value.\n",
      "   NUMVP   ... number of restraints with -Ln(pdf) > VIOL_REPORT_CUT2[i].\n",
      "   RMS_1   ... RMS(feature, minimally_violated_basis_restraint, NUMB).\n",
      "   RMS_2   ... RMS(feature, best_value, NUMB).\n",
      "   MOL.PDF ... scaled contribution to -Ln(Molecular pdf).\n",
      "\n",
      " #                     RESTRAINT_GROUP      NUM   NUMVI  NUMVP   RMS_1   RMS_2         MOL.PDF     S_i\n",
      "------------------------------------------------------------------------------------------------------\n",
      " 1 Bond length potential              :      90       0      2   0.027   0.027      16.921       1.000\n",
      " 2 Bond angle potential               :     128       2      6   5.096   5.096      63.630       1.000\n",
      " 3 Stereochemical cosine torsion poten:      81       0      4  47.866  47.866      31.106       1.000\n",
      " 4 Stereochemical improper torsion pot:      40       0      1   2.726   2.726      5.8387       1.000\n",
      " 5 Soft-sphere overlap restraints     :    2601       3      5   0.012   0.012      47.602       1.000\n",
      " 6 Lennard-Jones 6-12 potential       :       0       0      0   0.000   0.000      0.0000       1.000\n",
      " 7 Coulomb point-point electrostatic p:       0       0      0   0.000   0.000      0.0000       1.000\n",
      " 8 H-bonding potential                :       0       0      0   0.000   0.000      0.0000       1.000\n",
      " 9 Distance restraints 1 (CA-CA)      :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "10 Distance restraints 2 (N-O)        :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "11 Mainchain Phi dihedral restraints  :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "12 Mainchain Psi dihedral restraints  :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "13 Mainchain Omega dihedral restraints:      12       1      2  11.444  11.444      18.548       1.000\n",
      "14 Sidechain Chi_1 dihedral restraints:      11       0      1  76.244  76.244      10.170       1.000\n",
      "15 Sidechain Chi_2 dihedral restraints:       7       0      0  23.672  23.672      4.0935       1.000\n",
      "16 Sidechain Chi_3 dihedral restraints:       3       0      0  29.841  29.841     0.12643       1.000\n",
      "17 Sidechain Chi_4 dihedral restraints:       0       0      0   0.000   0.000      0.0000       1.000\n",
      "18 Disulfide distance restraints      :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "19 Disulfide angle restraints         :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "20 Disulfide dihedral angle restraints:       0       0      0   0.000   0.000      0.0000       1.000\n",
      "21 Lower bound distance restraints    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "22 Upper bound distance restraints    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "23 Distance restraints 3 (SDCH-MNCH)  :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "24 Sidechain Chi_5 dihedral restraints:       0       0      0   0.000   0.000      0.0000       1.000\n",
      "25 Phi/Psi pair of dihedral restraints:      13       9      5  59.032 151.333      35.066       1.000\n",
      "26 Distance restraints 4 (SDCH-SDCH)  :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "27 Distance restraints 5 (X-Y)        :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "28 NMR distance restraints 6 (X-Y)    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "29 NMR distance restraints 7 (X-Y)    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "30 Minimal distance restraints        :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "31 Non-bonded restraints              :    2601       0     23   1.875   1.875      66.439       1.000\n",
      "32 Atomic accessibility restraints    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "33 Atomic density restraints          :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "34 Absolute position restraints       :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "35 Dihedral angle difference restraint:       0       0      0   0.000   0.000      0.0000       1.000\n",
      "36 GBSA implicit solvent potential    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "37 EM density fitting potential       :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "38 SAXS restraints                    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "39 Symmetry restraints                :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "\n",
      "\n",
      "\n",
      "List of the violated restraints:\n",
      "   A restraint is violated when the relative difference\n",
      "   from the best value (RVIOL) is larger than CUTOFF.\n",
      "\n",
      "   ICSR   ... index of a restraint in the current set.\n",
      "   RESNO  ... residue numbers of the first two atoms.\n",
      "   ATM    ... IUPAC atom names of the first two atoms.\n",
      "   FEAT   ... the value of the feature in the model.\n",
      "   restr  ... the mean of the basis restraint with the smallest\n",
      "              difference from the model (local minimum).\n",
      "   viol   ... difference from the local minimum.\n",
      "   rviol  ... relative difference from the local minimum.\n",
      "   RESTR  ... the best value (global minimum).\n",
      "   VIOL   ... difference from the best value.\n",
      "   RVIOL  ... relative difference from the best value.\n",
      "\n",
      "\n",
      "-------------------------------------------------------------------------------------------------\n",
      "\n",
      "Feature  2                           : Bond angle potential                    \n",
      "List of the RVIOL violations larger than   :       4.5000\n",
      "\n",
      "    #   ICSR  RESNO1/2 ATM1/2   INDATM1/2    FEAT   restr    viol   rviol   RESTR    VIOL   RVIOL\n",
      "    1    173 969S 969S N   CA   7896 7897  129.16  107.00   22.16    6.37  107.00   22.16    6.37\n",
      "    2    182 970Y 970Y CA  CB   7903 7904  128.77  107.50   21.27    4.92  107.50   21.27    4.92\n",
      "\n",
      "-------------------------------------------------------------------------------------------------\n",
      "\n",
      "Feature 13                           : Mainchain Omega dihedral restraints     \n",
      "List of the RVIOL violations larger than   :       4.5000\n",
      "\n",
      "    #   ICSR  RESNO1/2 ATM1/2   INDATM1/2    FEAT   restr    viol   rviol   RESTR    VIOL   RVIOL\n",
      "    1    361 969S 969S CA  C    7897 7900 -152.03  180.00   27.97    5.59  180.00   27.97    5.59\n",
      "\n",
      "-------------------------------------------------------------------------------------------------\n",
      "\n",
      "Feature 25                           : Phi/Psi pair of dihedral restraints     \n",
      "List of the RVIOL violations larger than   :       6.5000\n",
      "\n",
      "    #   ICSR  RESNO1/2 ATM1/2   INDATM1/2    FEAT   restr    viol   rviol   RESTR    VIOL   RVIOL\n",
      "    1    340 960R 961R C   N    7830 7832 -101.95 -125.20   24.08    0.81  -63.00  179.73   21.93\n",
      "    1        961R 961R N   CA   7832 7833  134.36  140.60                  -41.10\n",
      "    2    343 963V 964N C   N    7855 7857 -162.89 -119.90   43.90    2.04  -63.20 -163.61   21.10\n",
      "    2        964N 964N N   CA   7857 7858  128.11  137.00                  -41.10\n",
      "    3    344 964N 965P C   N    7863 7865  -75.20  -58.70   20.46    1.67  -64.50  165.95   12.96\n",
      "    3        965P 965P N   CA   7865 7866  -18.40  -30.50                  147.20\n",
      "    4    345 965P 966Q C   N    7870 7872 -149.05 -121.10   58.43    2.20  -63.80  154.36   26.99\n",
      "    4        966Q 966Q N   CA   7872 7873 -168.99  139.70                  -40.30\n",
      "    5    346 966Q 967D C   N    7879 7881   85.70   54.50  156.70   14.05  -63.30 -165.17   22.82\n",
      "    5        967D 967D N   CA   7881 7882 -165.54   40.90                  -40.00\n",
      "    6    348 968P 969S C   N    7894 7896  -68.38  -72.40    8.12    0.37  -64.10  179.71   13.28\n",
      "    6        969S 969S N   CA   7896 7897  145.34  152.40                  -35.00\n",
      "    7    349 969S 970Y C   N    7900 7902 -137.17 -124.30   32.41    1.30  -63.50  168.43   30.34\n",
      "    7        970Y 970Y N   CA   7902 7903  165.14  135.40                  -43.40\n",
      "    8    350 970Y 971V C   N    7912 7914  -50.93  -62.40   11.59    1.75 -125.40 -172.00   10.66\n",
      "    8        971V 971V N   CA   7914 7915  -44.08  -42.40                  143.30\n",
      "    9    351 971V 972D C   N    7919 7921 -114.85  -96.50   18.39    0.75  -63.30  161.47   17.50\n",
      "    9        972D 972D N   CA   7921 7922  113.02  114.20                  -40.00\n",
      "\n",
      "\n",
      "report______> Distribution of short non-bonded contacts:\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "  serious non-bonded atom clash:  1055 1178   2.040\n",
      "  serious non-bonded atom clash:  1976 5131   1.952\n",
      "  serious non-bonded atom clash:  4241 4364   2.034\n",
      "  serious non-bonded atom clash:  7368 7491   2.036\n",
      "  serious non-bonded atom clash:  7778 8016   2.297\n",
      "  serious non-bonded atom clash: 1050010623   2.039\n",
      "  serious non-bonded atom clash: 1103111037   2.051\n",
      "  serious non-bonded atom clash: 1142114576   2.100\n",
      "  serious non-bonded atom clash: 1368613809   2.037\n",
      "  serious non-bonded atom clash: 1414614278   2.265\n",
      "\n",
      "DISTANCE1:  0.00 2.10 2.20 2.30 2.40 2.50 2.60 2.70 2.80 2.90 3.00 3.10 3.20 3.30 3.40\n",
      "DISTANCE2:  2.10 2.20 2.30 2.40 2.50 2.60 2.70 2.80 2.90 3.00 3.10 3.20 3.30 3.40 3.50\n",
      "FREQUENCY:     7    1    2   13   12   38   82   85  262  696  992  906 1801 2411 2320\n",
      "\n",
      "\n",
      "<< end of ENERGY.\n",
      "iupac_m_487_> NH1/2 swapped:     -179.8813       10    10\n",
      "iupac_m_487_> NH1/2 swapped:      179.9234       20    20\n",
      "iupac_m_487_> NH1/2 swapped:      179.8847      207   207\n",
      "iupac_m_487_> NH1/2 swapped:     -179.7626      327   327\n",
      "iupac_m_487_> NH1/2 swapped:      180.0000      405    18\n",
      "iupac_m_487_> NH1/2 swapped:     -179.7577      411    24\n",
      "iupac_m_487_> NH1/2 swapped:      179.8703      462    75\n",
      "iupac_m_487_> NH1/2 swapped:      179.7842      470    83\n",
      "iupac_m_487_> NH1/2 swapped:     -179.7762      475    88\n",
      "iupac_m_487_> NH1/2 swapped:      179.8481      502   115\n",
      "iupac_m_487_> NH1/2 swapped:     -179.9093      701   314\n",
      "iupac_m_487_> NH1/2 swapped:     -179.7168      715   328\n",
      "iupac_m_487_> NH1/2 swapped:      179.8615      716   400\n",
      "iupac_m_487_> NH1/2 swapped:     -179.9657      840    68\n",
      "iupac_m_487_> NH1/2 swapped:     -179.9209      847    75\n",
      "iupac_m_487_> NH1/2 swapped:      179.7175      860    88\n",
      "iupac_m_487_> NH1/2 swapped:     -180.0000      887   115\n",
      "iupac_m_487_> NH1/2 swapped:     -178.9152      923   151\n",
      "iupac_m_487_> NH1/2 swapped:      179.6326      960   188\n",
      "iupac_m_487_> NH1/2 swapped:      179.9558     1086   314\n",
      "iupac_m_487_> NH1/2 swapped:     -179.5980     1100   328\n",
      "iupac_m_487_> NH1/2 swapped:     -179.6961     1170    10\n",
      "iupac_m_487_> NH1/2 swapped:     -179.8275     1180    20\n",
      "iupac_m_486_> OE1/2 will be swapped:     -109.6868     1332   172\n",
      "iupac_m_487_> NH1/2 swapped:      179.9477     1367   207\n",
      "iupac_m_487_> NH1/2 swapped:      179.9115     1487   327\n",
      "iupac_m_487_> NH1/2 swapped:     -179.5082     1571    24\n",
      "iupac_m_487_> NH1/2 swapped:      179.9720     1622    75\n",
      "iupac_m_487_> NH1/2 swapped:      180.0000     1630    83\n",
      "iupac_m_487_> NH1/2 swapped:     -179.8864     1635    88\n",
      "iupac_m_487_> NH1/2 swapped:      179.5804     1662   115\n",
      "iupac_m_487_> NH1/2 swapped:      179.8072     1861   314\n",
      "iupac_m_487_> NH1/2 swapped:      179.7521     1875   328\n",
      "iupac_m_487_> NH1/2 swapped:      179.8587     1876   400\n",
      "openf___224_> Open           muscle.BL00080001.pdb\n",
      "wrpdb___568_> Residues, atoms, selected atoms:     1935    15763    15763\n",
      "openf___224_> Open           muscle.IL00000001.pdb\n",
      "\n",
      "Dynamically allocated memory at       amaxmodel [B,KiB,MiB]:     45612318   44543.281    43.499\n",
      "read_mo_297_> Segments, residues, atoms:        5     1935    15763\n",
      "read_mo_298_> Segment:        1     1 A   469 A     3168\n",
      "read_mo_298_> Segment:        2     1 B   459 B     3150\n",
      "read_mo_298_> Segment:        3     4 C   459 C     3127\n",
      "read_mo_298_> Segment:        4     1 D   469 D     3168\n",
      "read_mo_298_> Segment:        5     1 E   459 E     3150\n",
      "randomi_498_> Atoms,selected atoms,random_seed,amplitude:    15763       86        1        5.0000\n",
      "randomi_496_> Amplitude is > 0; randomization is done.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "\n",
      "\n",
      ">> ENERGY; Differences between the model's features and restraints:\n",
      "Number of all residues in MODEL                   :     1935\n",
      "Number of all, selected real atoms                :    15763      86\n",
      "Number of all, selected pseudo atoms              :        0       0\n",
      "Number of all static, selected restraints         :      385     385\n",
      "COVALENT_CYS                                      :        F\n",
      "NONBONDED_SEL_ATOMS                               :        1\n",
      "Number of non-bonded pairs (excluding 1-2,1-3,1-4):     2434\n",
      "Dynamic pairs routine                             : 5, NATM_SEL x NATM double loop\n",
      "Atomic shift for contacts update (UPDATE_DYNAMIC) :    0.390\n",
      "LENNARD_JONES_SWITCH                              :    6.500   7.500\n",
      "COULOMB_JONES_SWITCH                              :    6.500   7.500\n",
      "RESIDUE_SPAN_RANGE                                :        0   99999\n",
      "NLOGN_USE                                         :       15\n",
      "CONTACT_SHELL                                     :    7.000\n",
      "DYNAMIC_PAIRS,_SPHERE,_COULOMB,_LENNARD,_MODELLER :        T       T       F       F       T\n",
      "SPHERE_STDV                                       :    0.050\n",
      "RADII_FACTOR                                      :    0.820\n",
      "Current energy                                    :         509.5998\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Summary of the restraint violations: \n",
      "\n",
      "   NUM     ... number of restraints.\n",
      "   NUMVI   ... number of restraints with RVIOL > VIOL_REPORT_CUT[i].\n",
      "   RVIOL   ... relative difference from the best value.\n",
      "   NUMVP   ... number of restraints with -Ln(pdf) > VIOL_REPORT_CUT2[i].\n",
      "   RMS_1   ... RMS(feature, minimally_violated_basis_restraint, NUMB).\n",
      "   RMS_2   ... RMS(feature, best_value, NUMB).\n",
      "   MOL.PDF ... scaled contribution to -Ln(Molecular pdf).\n",
      "\n",
      " #                     RESTRAINT_GROUP      NUM   NUMVI  NUMVP   RMS_1   RMS_2         MOL.PDF     S_i\n",
      "------------------------------------------------------------------------------------------------------\n",
      " 1 Bond length potential              :      90       3      5   0.047   0.047      61.026       1.000\n",
      " 2 Bond angle potential               :     128       5     12   6.495   6.495      98.735       1.000\n",
      " 3 Stereochemical cosine torsion poten:      81       0      5  49.340  49.340      31.694       1.000\n",
      " 4 Stereochemical improper torsion pot:      40       0      2   3.604   3.604      9.3701       1.000\n",
      " 5 Soft-sphere overlap restraints     :    2434       7      9   0.020   0.020      114.08       1.000\n",
      " 6 Lennard-Jones 6-12 potential       :       0       0      0   0.000   0.000      0.0000       1.000\n",
      " 7 Coulomb point-point electrostatic p:       0       0      0   0.000   0.000      0.0000       1.000\n",
      " 8 H-bonding potential                :       0       0      0   0.000   0.000      0.0000       1.000\n",
      " 9 Distance restraints 1 (CA-CA)      :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "10 Distance restraints 2 (N-O)        :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "11 Mainchain Phi dihedral restraints  :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "12 Mainchain Psi dihedral restraints  :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "13 Mainchain Omega dihedral restraints:      12       1      1   8.124   8.124      9.3470       1.000\n",
      "14 Sidechain Chi_1 dihedral restraints:      11       0      0  73.483  73.483      4.3733       1.000\n",
      "15 Sidechain Chi_2 dihedral restraints:       7       0      0  75.353  75.353      4.6228       1.000\n",
      "16 Sidechain Chi_3 dihedral restraints:       3       0      0  74.862  74.862      2.1188       1.000\n",
      "17 Sidechain Chi_4 dihedral restraints:       0       0      0   0.000   0.000      0.0000       1.000\n",
      "18 Disulfide distance restraints      :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "19 Disulfide angle restraints         :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "20 Disulfide dihedral angle restraints:       0       0      0   0.000   0.000      0.0000       1.000\n",
      "21 Lower bound distance restraints    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "22 Upper bound distance restraints    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "23 Distance restraints 3 (SDCH-MNCH)  :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "24 Sidechain Chi_5 dihedral restraints:       0       0      0   0.000   0.000      0.0000       1.000\n",
      "25 Phi/Psi pair of dihedral restraints:      13      10      6  67.270 154.094      52.062       1.000\n",
      "26 Distance restraints 4 (SDCH-SDCH)  :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "27 Distance restraints 5 (X-Y)        :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "28 NMR distance restraints 6 (X-Y)    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "29 NMR distance restraints 7 (X-Y)    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "30 Minimal distance restraints        :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "31 Non-bonded restraints              :    2434       0     30   1.946   1.946      122.17       1.000\n",
      "32 Atomic accessibility restraints    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "33 Atomic density restraints          :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "34 Absolute position restraints       :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "35 Dihedral angle difference restraint:       0       0      0   0.000   0.000      0.0000       1.000\n",
      "36 GBSA implicit solvent potential    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "37 EM density fitting potential       :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "38 SAXS restraints                    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "39 Symmetry restraints                :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "\n",
      "\n",
      "\n",
      "List of the violated restraints:\n",
      "   A restraint is violated when the relative difference\n",
      "   from the best value (RVIOL) is larger than CUTOFF.\n",
      "\n",
      "   ICSR   ... index of a restraint in the current set.\n",
      "   RESNO  ... residue numbers of the first two atoms.\n",
      "   ATM    ... IUPAC atom names of the first two atoms.\n",
      "   FEAT   ... the value of the feature in the model.\n",
      "   restr  ... the mean of the basis restraint with the smallest\n",
      "              difference from the model (local minimum).\n",
      "   viol   ... difference from the local minimum.\n",
      "   rviol  ... relative difference from the local minimum.\n",
      "   RESTR  ... the best value (global minimum).\n",
      "   VIOL   ... difference from the best value.\n",
      "   RVIOL  ... relative difference from the best value.\n",
      "\n",
      "\n",
      "-------------------------------------------------------------------------------------------------\n",
      "\n",
      "Feature  1                           : Bond length potential                   \n",
      "List of the RVIOL violations larger than   :       4.5000\n",
      "\n",
      "    #   ICSR  RESNO1/2 ATM1/2   INDATM1/2    FEAT   restr    viol   rviol   RESTR    VIOL   RVIOL\n",
      "    1     74 970Y 970Y C   CA   7912 7903    1.66    1.49    0.17    4.95    1.49    0.17    4.95\n",
      "    2     75 970Y 971V C   N    7912 7914    1.59    1.35    0.25    8.72    1.35    0.25    8.72\n",
      "    3     79 971V 971V N   CA   7914 7915    1.68    1.43    0.25    8.13    1.43    0.25    8.13\n",
      "\n",
      "-------------------------------------------------------------------------------------------------\n",
      "\n",
      "Feature  2                           : Bond angle potential                    \n",
      "List of the RVIOL violations larger than   :       4.5000\n",
      "\n",
      "    #   ICSR  RESNO1/2 ATM1/2   INDATM1/2    FEAT   restr    viol   rviol   RESTR    VIOL   RVIOL\n",
      "    1    182 970Y 970Y CA  CB   7903 7904  130.20  107.50   22.70    5.25  107.50   22.70    5.25\n",
      "    2    184 970Y 970Y CA  C    7903 7912  134.13  116.50   17.63    5.07  116.50   17.63    5.07\n",
      "    3    195 970Y 970Y C   CA   7912 7903  136.21  108.00   28.21    6.54  108.00   28.21    6.54\n",
      "    4    196 970Y 971V C   N    7912 7914  151.07  120.00   31.07    7.06  120.00   31.07    7.06\n",
      "    5    199 971V 971V N   CA   7914 7915  131.72  107.00   24.72    7.11  107.00   24.72    7.11\n",
      "\n",
      "-------------------------------------------------------------------------------------------------\n",
      "\n",
      "Feature 13                           : Mainchain Omega dihedral restraints     \n",
      "List of the RVIOL violations larger than   :       4.5000\n",
      "\n",
      "    #   ICSR  RESNO1/2 ATM1/2   INDATM1/2    FEAT   restr    viol   rviol   RESTR    VIOL   RVIOL\n",
      "    1    364 972D 972D CA  C    7922 7927 -156.52  180.00   23.48    4.70  180.00   23.48    4.70\n",
      "\n",
      "-------------------------------------------------------------------------------------------------\n",
      "\n",
      "Feature 25                           : Phi/Psi pair of dihedral restraints     \n",
      "List of the RVIOL violations larger than   :       6.5000\n",
      "\n",
      "    #   ICSR  RESNO1/2 ATM1/2   INDATM1/2    FEAT   restr    viol   rviol   RESTR    VIOL   RVIOL\n",
      "    1    340 960R 961R C   N    7830 7832 -101.95 -125.20   24.54    0.80  -63.00  178.17   21.72\n",
      "    1        961R 961R N   CA   7832 7833  132.77  140.60                  -41.10\n",
      "    2    341 961R 962T C   N    7841 7843   62.73  -78.10  140.90    8.43  -63.20 -146.31   22.65\n",
      "    2        962T 962T N   CA   7843 7844  145.26  149.80                  -42.10\n",
      "    3    342 962T 963V C   N    7848 7850  -71.73  -62.40   17.98    1.97 -125.40  178.59    7.34\n",
      "    3        963V 963V N   CA   7850 7851  -27.03  -42.40                  143.30\n",
      "    4    343 963V 964N C   N    7855 7857  -92.64  -71.20   28.67    1.56  -63.20  159.80   21.64\n",
      "    4        964N 964N N   CA   7857 7858  161.84  142.80                  -41.10\n",
      "    5    345 965P 966Q C   N    7870 7872 -151.25 -121.10  109.37    4.62  -63.80  115.12   20.84\n",
      "    5        966Q 966Q N   CA   7872 7873 -115.17  139.70                  -40.30\n",
      "    6    346 966Q 967D C   N    7879 7881  -80.82  -70.90   24.95    0.85  -63.30  147.84   19.35\n",
      "    6        967D 967D N   CA   7881 7882  173.20  150.30                  -40.00\n",
      "    7    347 967D 968P C   N    7887 7889  -64.83  -58.70   21.90    2.32  -64.50  161.27   12.02\n",
      "    7        968P 968P N   CA   7889 7890  -51.53  -30.50                  147.20\n",
      "    8    348 968P 969S C   N    7894 7896 -170.70 -136.60   50.53    2.84  -64.10 -176.86   12.92\n",
      "    8        969S 969S N   CA   7896 7897  113.92  151.20                  -35.00\n",
      "    9    349 969S 970Y C   N    7900 7902 -166.79 -124.30   96.72    3.82  -63.50  139.87   26.45\n",
      "    9        970Y 970Y N   CA   7902 7903 -137.71  135.40                  -43.40\n",
      "   10    351 971V 972D C   N    7919 7921   77.07   54.50   97.90    9.02  -63.30 -134.76   36.58\n",
      "   10        972D 972D N   CA   7921 7922  136.16   40.90                  -40.00\n",
      "\n",
      "\n",
      "report______> Distribution of short non-bonded contacts:\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "  serious non-bonded atom clash:  1055 1178   2.040\n",
      "  serious non-bonded atom clash:  1976 5131   1.952\n",
      "  serious non-bonded atom clash:  4241 4364   2.034\n",
      "  serious non-bonded atom clash:  7368 7491   2.036\n",
      "  serious non-bonded atom clash:  7636 7914   2.087\n",
      "  serious non-bonded atom clash:  7778 8016   2.297\n",
      "  serious non-bonded atom clash: 1050010623   2.039\n",
      "  serious non-bonded atom clash: 1103111037   2.051\n",
      "  serious non-bonded atom clash: 1142114576   2.100\n",
      "  serious non-bonded atom clash: 1368613809   2.037\n",
      "  serious non-bonded atom clash: 1414614278   2.265\n",
      "\n",
      "DISTANCE1:  0.00 2.10 2.20 2.30 2.40 2.50 2.60 2.70 2.80 2.90 3.00 3.10 3.20 3.30 3.40\n",
      "DISTANCE2:  2.10 2.20 2.30 2.40 2.50 2.60 2.70 2.80 2.90 3.00 3.10 3.20 3.30 3.40 3.50\n",
      "FREQUENCY:     8    1    2   15   12   41   77   88  259  695  994  900 1803 2406 2318\n",
      "\n",
      "\n",
      "<< end of ENERGY.\n",
      "iupac_m_487_> NH1/2 swapped:     -179.8813       10    10\n",
      "iupac_m_487_> NH1/2 swapped:      179.9234       20    20\n",
      "iupac_m_487_> NH1/2 swapped:      179.8847      207   207\n",
      "iupac_m_487_> NH1/2 swapped:     -179.7626      327   327\n",
      "iupac_m_487_> NH1/2 swapped:      180.0000      405    18\n",
      "iupac_m_487_> NH1/2 swapped:     -179.7577      411    24\n",
      "iupac_m_487_> NH1/2 swapped:      179.8703      462    75\n",
      "iupac_m_487_> NH1/2 swapped:      179.7842      470    83\n",
      "iupac_m_487_> NH1/2 swapped:     -179.7762      475    88\n",
      "iupac_m_487_> NH1/2 swapped:      179.8481      502   115\n",
      "iupac_m_487_> NH1/2 swapped:     -179.9093      701   314\n",
      "iupac_m_487_> NH1/2 swapped:     -179.7168      715   328\n",
      "iupac_m_487_> NH1/2 swapped:      179.8615      716   400\n",
      "iupac_m_487_> NH1/2 swapped:     -179.9657      840    68\n",
      "iupac_m_487_> NH1/2 swapped:     -179.9209      847    75\n",
      "iupac_m_487_> NH1/2 swapped:      179.7175      860    88\n",
      "iupac_m_487_> NH1/2 swapped:     -180.0000      887   115\n",
      "iupac_m_487_> NH1/2 swapped:     -178.9152      923   151\n",
      "iupac_m_487_> NH1/2 swapped:      179.6326      960   188\n",
      "iupac_m_485_> OD1/2 will be swapped:      174.3780      967   195\n",
      "iupac_m_483_> CD1/2 CE1/2 swapped:      114.8879      970   198\n",
      "iupac_m_397W> Atoms were not swapped because of the uncertainty of how to handle the H atom.\n",
      "iupac_m_487_> NH1/2 swapped:      179.9558     1086   314\n",
      "iupac_m_487_> NH1/2 swapped:     -179.5980     1100   328\n",
      "iupac_m_487_> NH1/2 swapped:     -179.6961     1170    10\n",
      "iupac_m_487_> NH1/2 swapped:     -179.8275     1180    20\n",
      "iupac_m_486_> OE1/2 will be swapped:     -109.6868     1332   172\n",
      "iupac_m_487_> NH1/2 swapped:      179.9477     1367   207\n",
      "iupac_m_487_> NH1/2 swapped:      179.9115     1487   327\n",
      "iupac_m_487_> NH1/2 swapped:     -179.5082     1571    24\n",
      "iupac_m_487_> NH1/2 swapped:      179.9720     1622    75\n",
      "iupac_m_487_> NH1/2 swapped:      180.0000     1630    83\n",
      "iupac_m_487_> NH1/2 swapped:     -179.8864     1635    88\n",
      "iupac_m_487_> NH1/2 swapped:      179.5804     1662   115\n",
      "iupac_m_487_> NH1/2 swapped:      179.8072     1861   314\n",
      "iupac_m_487_> NH1/2 swapped:      179.7521     1875   328\n",
      "iupac_m_487_> NH1/2 swapped:      179.8587     1876   400\n",
      "openf___224_> Open           muscle.BL00090001.pdb\n",
      "wrpdb___568_> Residues, atoms, selected atoms:     1935    15763    15763\n",
      "openf___224_> Open           muscle.IL00000001.pdb\n",
      "\n",
      "Dynamically allocated memory at       amaxmodel [B,KiB,MiB]:     45612318   44543.281    43.499\n",
      "read_mo_297_> Segments, residues, atoms:        5     1935    15763\n",
      "read_mo_298_> Segment:        1     1 A   469 A     3168\n",
      "read_mo_298_> Segment:        2     1 B   459 B     3150\n",
      "read_mo_298_> Segment:        3     4 C   459 C     3127\n",
      "read_mo_298_> Segment:        4     1 D   469 D     3168\n",
      "read_mo_298_> Segment:        5     1 E   459 E     3150\n",
      "randomi_498_> Atoms,selected atoms,random_seed,amplitude:    15763       86        1        5.0000\n",
      "randomi_496_> Amplitude is > 0; randomization is done.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "preppdf_457W> Both soft sphere and statistical potential terms selected.\n",
      "\n",
      "\n",
      ">> ENERGY; Differences between the model's features and restraints:\n",
      "Number of all residues in MODEL                   :     1935\n",
      "Number of all, selected real atoms                :    15763      86\n",
      "Number of all, selected pseudo atoms              :        0       0\n",
      "Number of all static, selected restraints         :      385     385\n",
      "COVALENT_CYS                                      :        F\n",
      "NONBONDED_SEL_ATOMS                               :        1\n",
      "Number of non-bonded pairs (excluding 1-2,1-3,1-4):     3612\n",
      "Dynamic pairs routine                             : 5, NATM_SEL x NATM double loop\n",
      "Atomic shift for contacts update (UPDATE_DYNAMIC) :    0.390\n",
      "LENNARD_JONES_SWITCH                              :    6.500   7.500\n",
      "COULOMB_JONES_SWITCH                              :    6.500   7.500\n",
      "RESIDUE_SPAN_RANGE                                :        0   99999\n",
      "NLOGN_USE                                         :       15\n",
      "CONTACT_SHELL                                     :    7.000\n",
      "DYNAMIC_PAIRS,_SPHERE,_COULOMB,_LENNARD,_MODELLER :        T       T       F       F       T\n",
      "SPHERE_STDV                                       :    0.050\n",
      "RADII_FACTOR                                      :    0.820\n",
      "Current energy                                    :        1373.4894\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Summary of the restraint violations: \n",
      "\n",
      "   NUM     ... number of restraints.\n",
      "   NUMVI   ... number of restraints with RVIOL > VIOL_REPORT_CUT[i].\n",
      "   RVIOL   ... relative difference from the best value.\n",
      "   NUMVP   ... number of restraints with -Ln(pdf) > VIOL_REPORT_CUT2[i].\n",
      "   RMS_1   ... RMS(feature, minimally_violated_basis_restraint, NUMB).\n",
      "   RMS_2   ... RMS(feature, best_value, NUMB).\n",
      "   MOL.PDF ... scaled contribution to -Ln(Molecular pdf).\n",
      "\n",
      " #                     RESTRAINT_GROUP      NUM   NUMVI  NUMVP   RMS_1   RMS_2         MOL.PDF     S_i\n",
      "------------------------------------------------------------------------------------------------------\n",
      " 1 Bond length potential              :      90       5     11   0.074   0.074      131.53       1.000\n",
      " 2 Bond angle potential               :     128       8     21   8.375   8.375      171.94       1.000\n",
      " 3 Stereochemical cosine torsion poten:      81       0      3  45.096  45.096      26.468       1.000\n",
      " 4 Stereochemical improper torsion pot:      40       2      4   7.402   7.402      39.358       1.000\n",
      " 5 Soft-sphere overlap restraints     :    3612      20     30   0.034   0.034      486.14       1.000\n",
      " 6 Lennard-Jones 6-12 potential       :       0       0      0   0.000   0.000      0.0000       1.000\n",
      " 7 Coulomb point-point electrostatic p:       0       0      0   0.000   0.000      0.0000       1.000\n",
      " 8 H-bonding potential                :       0       0      0   0.000   0.000      0.0000       1.000\n",
      " 9 Distance restraints 1 (CA-CA)      :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "10 Distance restraints 2 (N-O)        :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "11 Mainchain Phi dihedral restraints  :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "12 Mainchain Psi dihedral restraints  :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "13 Mainchain Omega dihedral restraints:      12       1      4  13.836  13.836      27.110       1.000\n",
      "14 Sidechain Chi_1 dihedral restraints:      11       0      2  73.020  73.020      8.8597       1.000\n",
      "15 Sidechain Chi_2 dihedral restraints:       7       0      0  94.155  94.155      5.6361       1.000\n",
      "16 Sidechain Chi_3 dihedral restraints:       3       0      0  75.023  75.023      3.6760       1.000\n",
      "17 Sidechain Chi_4 dihedral restraints:       0       0      0   0.000   0.000      0.0000       1.000\n",
      "18 Disulfide distance restraints      :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "19 Disulfide angle restraints         :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "20 Disulfide dihedral angle restraints:       0       0      0   0.000   0.000      0.0000       1.000\n",
      "21 Lower bound distance restraints    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "22 Upper bound distance restraints    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "23 Distance restraints 3 (SDCH-MNCH)  :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "24 Sidechain Chi_5 dihedral restraints:       0       0      0   0.000   0.000      0.0000       1.000\n",
      "25 Phi/Psi pair of dihedral restraints:      13       9      4  32.361 137.616      25.111       1.000\n",
      "26 Distance restraints 4 (SDCH-SDCH)  :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "27 Distance restraints 5 (X-Y)        :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "28 NMR distance restraints 6 (X-Y)    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "29 NMR distance restraints 7 (X-Y)    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "30 Minimal distance restraints        :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "31 Non-bonded restraints              :    3612       0    109   1.884   1.884      447.66       1.000\n",
      "32 Atomic accessibility restraints    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "33 Atomic density restraints          :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "34 Absolute position restraints       :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "35 Dihedral angle difference restraint:       0       0      0   0.000   0.000      0.0000       1.000\n",
      "36 GBSA implicit solvent potential    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "37 EM density fitting potential       :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "38 SAXS restraints                    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "39 Symmetry restraints                :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "\n",
      "\n",
      "\n",
      "List of the violated restraints:\n",
      "   A restraint is violated when the relative difference\n",
      "   from the best value (RVIOL) is larger than CUTOFF.\n",
      "\n",
      "   ICSR   ... index of a restraint in the current set.\n",
      "   RESNO  ... residue numbers of the first two atoms.\n",
      "   ATM    ... IUPAC atom names of the first two atoms.\n",
      "   FEAT   ... the value of the feature in the model.\n",
      "   restr  ... the mean of the basis restraint with the smallest\n",
      "              difference from the model (local minimum).\n",
      "   viol   ... difference from the local minimum.\n",
      "   rviol  ... relative difference from the local minimum.\n",
      "   RESTR  ... the best value (global minimum).\n",
      "   VIOL   ... difference from the best value.\n",
      "   RVIOL  ... relative difference from the best value.\n",
      "\n",
      "\n",
      "-------------------------------------------------------------------------------------------------\n",
      "\n",
      "Feature  1                           : Bond length potential                   \n",
      "List of the RVIOL violations larger than   :       4.5000\n",
      "\n",
      "    #   ICSR  RESNO1/2 ATM1/2   INDATM1/2    FEAT   restr    viol   rviol   RESTR    VIOL   RVIOL\n",
      "    1     48 967D 968P C   N    7887 7889    1.70    1.30    0.40   11.97    1.30    0.40   11.97\n",
      "    2     56 968P 968P N   CD   7889 7891    1.65    1.46    0.20    6.57    1.46    0.20    6.57\n",
      "    3     63 970Y 970Y CB  CA   7904 7903    1.72    1.54    0.18    5.02    1.54    0.18    5.02\n",
      "    4     74 970Y 970Y C   CA   7912 7903    1.76    1.49    0.27    7.83    1.49    0.27    7.83\n",
      "\n",
      "-------------------------------------------------------------------------------------------------\n",
      "\n",
      "Feature  2                           : Bond angle potential                    \n",
      "List of the RVIOL violations larger than   :       4.5000\n",
      "\n",
      "    #   ICSR  RESNO1/2 ATM1/2   INDATM1/2    FEAT   restr    viol   rviol   RESTR    VIOL   RVIOL\n",
      "    1    159 967D 968P C   N    7887 7889  147.55  117.00   30.55    7.60  117.00   30.55    7.60\n",
      "    2    169 968P 968P C   CA   7894 7890  132.63  112.30   20.33    4.71  112.30   20.33    4.71\n",
      "    3    170 968P 969S C   N    7894 7896  144.84  120.00   24.84    5.64  120.00   24.84    5.64\n",
      "    4    178 969S 970Y C   N    7900 7902  141.12  120.00   21.12    4.80  120.00   21.12    4.80\n",
      "    5    182 970Y 970Y CA  CB   7903 7904  144.10  107.50   36.60    8.46  107.50   36.60    8.46\n",
      "    6    195 970Y 970Y C   CA   7912 7903  131.52  108.00   23.52    5.45  108.00   23.52    5.45\n",
      "\n",
      "-------------------------------------------------------------------------------------------------\n",
      "\n",
      "Feature 13                           : Mainchain Omega dihedral restraints     \n",
      "List of the RVIOL violations larger than   :       4.5000\n",
      "\n",
      "    #   ICSR  RESNO1/2 ATM1/2   INDATM1/2    FEAT   restr    viol   rviol   RESTR    VIOL   RVIOL\n",
      "    1    360 968P 968P CA  C    7890 7894 -151.17  180.00   28.83    5.77  180.00   28.83    5.77\n",
      "\n",
      "-------------------------------------------------------------------------------------------------\n",
      "\n",
      "Feature 25                           : Phi/Psi pair of dihedral restraints     \n",
      "List of the RVIOL violations larger than   :       6.5000\n",
      "\n",
      "    #   ICSR  RESNO1/2 ATM1/2   INDATM1/2    FEAT   restr    viol   rviol   RESTR    VIOL   RVIOL\n",
      "    1    340 960R 961R C   N    7830 7832 -101.95 -125.20   26.83    0.80  -63.00  172.75   21.00\n",
      "    1        961R 961R N   CA   7832 7833  127.21  140.60                  -41.10\n",
      "    2    342 962T 963V C   N    7848 7850  -61.88  -62.40    3.71    0.50 -125.40 -171.03   10.59\n",
      "    2        963V 963V N   CA   7850 7851  -38.73  -42.40                  143.30\n",
      "    3    343 963V 964N C   N    7855 7857 -100.71 -119.90   31.32    1.70  -63.20  161.56   22.33\n",
      "    3        964N 964N N   CA   7857 7858  161.76  137.00                  -41.10\n",
      "    4    344 964N 965P C   N    7863 7865  -71.62  -58.70   31.21    1.83  -64.50  149.46   11.53\n",
      "    4        965P 965P N   CA   7865 7866   -2.09  -30.50                  147.20\n",
      "    5    345 965P 966Q C   N    7870 7872  -88.74  -73.00   49.04    2.92  -63.80  134.87   21.17\n",
      "    5        966Q 966Q N   CA   7872 7873 -172.85  140.70                  -40.30\n",
      "    6    346 966Q 967D C   N    7879 7881  -65.32  -70.90    5.60    0.41  -63.30  170.24   21.14\n",
      "    6        967D 967D N   CA   7881 7882  149.78  150.30                  -40.00\n",
      "    7    347 967D 968P C   N    7887 7889  -42.40  -58.70   54.23    3.37  -64.50  132.44   11.08\n",
      "    7        968P 968P N   CA   7889 7890  -82.22  -30.50                  147.20\n",
      "    8    349 969S 970Y C   N    7900 7902  -74.30  -98.40   24.18    0.98  -63.50  170.22   26.47\n",
      "    8        970Y 970Y N   CA   7902 7903  126.48  128.40                  -43.40\n",
      "    9    351 971V 972D C   N    7919 7921   51.09   54.50   18.08    1.07  -63.30  151.05   25.42\n",
      "    9        972D 972D N   CA   7921 7922   58.65   40.90                  -40.00\n",
      "\n",
      "\n",
      "report______> Distribution of short non-bonded contacts:\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "  serious non-bonded atom clash:  1055 1178   2.040\n",
      "  serious non-bonded atom clash:  1976 5131   1.952\n",
      "  serious non-bonded atom clash:  4241 4364   2.034\n",
      "  serious non-bonded atom clash:  7368 7491   2.036\n",
      "  serious non-bonded atom clash:  7636 7903   2.175\n",
      "  serious non-bonded atom clash:  7649 7889   2.057\n",
      "  serious non-bonded atom clash:  7649 7891   2.171\n",
      "  serious non-bonded atom clash:  7778 8016   2.297\n",
      "  serious non-bonded atom clash: 1050010623   2.039\n",
      "  serious non-bonded atom clash: 1103111037   2.051\n",
      "  serious non-bonded atom clash: 1142114576   2.100\n",
      "  serious non-bonded atom clash: 1368613809   2.037\n",
      "  serious non-bonded atom clash: 1414614278   2.265\n",
      "\n",
      "DISTANCE1:  0.00 2.10 2.20 2.30 2.40 2.50 2.60 2.70 2.80 2.90 3.00 3.10 3.20 3.30 3.40\n",
      "DISTANCE2:  2.10 2.20 2.30 2.40 2.50 2.60 2.70 2.80 2.90 3.00 3.10 3.20 3.30 3.40 3.50\n",
      "FREQUENCY:     8    3    2   15   13   49   91  112  268  705 1007  914 1824 2416 2331\n",
      "\n",
      "\n",
      "<< end of ENERGY.\n",
      "iupac_m_487_> NH1/2 swapped:     -179.8813       10    10\n",
      "iupac_m_487_> NH1/2 swapped:      179.9234       20    20\n",
      "iupac_m_487_> NH1/2 swapped:      179.8847      207   207\n",
      "iupac_m_487_> NH1/2 swapped:     -179.7626      327   327\n",
      "iupac_m_487_> NH1/2 swapped:      180.0000      405    18\n",
      "iupac_m_487_> NH1/2 swapped:     -179.7577      411    24\n",
      "iupac_m_487_> NH1/2 swapped:      179.8703      462    75\n",
      "iupac_m_487_> NH1/2 swapped:      179.7842      470    83\n",
      "iupac_m_487_> NH1/2 swapped:     -179.7762      475    88\n",
      "iupac_m_487_> NH1/2 swapped:      179.8481      502   115\n",
      "iupac_m_487_> NH1/2 swapped:     -179.9093      701   314\n",
      "iupac_m_487_> NH1/2 swapped:     -179.7168      715   328\n",
      "iupac_m_487_> NH1/2 swapped:      179.8615      716   400\n",
      "iupac_m_487_> NH1/2 swapped:     -179.9657      840    68\n",
      "iupac_m_487_> NH1/2 swapped:     -179.9209      847    75\n",
      "iupac_m_487_> NH1/2 swapped:      179.7175      860    88\n",
      "iupac_m_487_> NH1/2 swapped:     -180.0000      887   115\n",
      "iupac_m_487_> NH1/2 swapped:     -178.9152      923   151\n",
      "iupac_m_487_> NH1/2 swapped:      179.6326      960   188\n",
      "iupac_m_485_> OD1/2 will be swapped:      151.2609      967   195\n",
      "iupac_m_487_> NH1/2 swapped:      179.9558     1086   314\n",
      "iupac_m_487_> NH1/2 swapped:     -179.5980     1100   328\n",
      "iupac_m_487_> NH1/2 swapped:     -179.6961     1170    10\n",
      "iupac_m_487_> NH1/2 swapped:     -179.8275     1180    20\n",
      "iupac_m_486_> OE1/2 will be swapped:     -109.6868     1332   172\n",
      "iupac_m_487_> NH1/2 swapped:      179.9477     1367   207\n",
      "iupac_m_487_> NH1/2 swapped:      179.9115     1487   327\n",
      "iupac_m_487_> NH1/2 swapped:     -179.5082     1571    24\n",
      "iupac_m_487_> NH1/2 swapped:      179.9720     1622    75\n",
      "iupac_m_487_> NH1/2 swapped:      180.0000     1630    83\n",
      "iupac_m_487_> NH1/2 swapped:     -179.8864     1635    88\n",
      "iupac_m_487_> NH1/2 swapped:      179.5804     1662   115\n",
      "iupac_m_487_> NH1/2 swapped:      179.8072     1861   314\n",
      "iupac_m_487_> NH1/2 swapped:      179.7521     1875   328\n",
      "iupac_m_487_> NH1/2 swapped:      179.8587     1876   400\n",
      "openf___224_> Open           muscle.BL00100001.pdb\n",
      "wrpdb___568_> Residues, atoms, selected atoms:     1935    15763    15763\n",
      "\n",
      ">> Summary of successfully produced loop models:\n",
      "Filename                          molpdf\n",
      "----------------------------------------\n",
      "muscle.BL00010001.pdb           76.53861\n",
      "muscle.BL00020001.pdb           40.19336\n",
      "muscle.BL00030001.pdb           61.41253\n",
      "muscle.BL00040001.pdb          482.25879\n",
      "muscle.BL00050001.pdb          389.57047\n",
      "muscle.BL00060001.pdb           70.87090\n",
      "muscle.BL00070001.pdb         1874.24939\n",
      "muscle.BL00080001.pdb          299.54074\n",
      "muscle.BL00090001.pdb          509.59979\n",
      "muscle.BL00100001.pdb         1373.48938\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# Loop refinement of an existing model\n",
    "from modeller import *\n",
    "from modeller.automodel import *\n",
    "\n",
    "log.verbose()\n",
    "env = environ()\n",
    "\n",
    "# directories for input atom files\n",
    "env.io.atom_files_directory = ['.']  # needed if you put your coordinate files somewhere\n",
    "\n",
    "# Create a new class based on 'loopmodel' so that we can redefine\n",
    "# select_loop_atoms (necessary)\n",
    "class MyLoop(loopmodel):\n",
    "    # This routine picks the residues to be refined by loop modeling\n",
    "    def select_loop_atoms(self):\n",
    "        # 10 residue insertion \n",
    "        return selection(self.residue_range('190:C', '200:C'))\n",
    "\n",
    "m = MyLoop(env,\n",
    "           inimodel='muscle_nachr.pdb', # initial model of the target\n",
    "           sequence='muscle')          # code of the target\n",
    "\n",
    "m.loop.starting_model= 1           # index of the first loop model \n",
    "m.loop.ending_model  = 10          # index of the last loop model\n",
    "m.loop.md_level = refine.very_fast # loop refinement method; this yields\n",
    "                                   # models quickly but of low quality;\n",
    "                                   # use refine.slow for better models\n",
    "\n",
    "m.make()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### 6. Scoring your models"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "- A typical strategy in obtaining accurate models is to generate multiple homology models and assess their quality based on various scoring functions.\n",
    "\n",
    "- Contained in the example_models directory are 100 previously generated homology models.\n",
    "\n",
    "- You can find the Modeller objective function score on the second line of each .pdb file.\n",
    "\n",
    "- This function is not an absolute measure and only pertains to models generated from the same alignment. Essentailly it is a relative scoring function for the set of models you have just built.\n",
    "\n",
    "- Lets extract the scores for these models...\n",
    "\n",
    "\t"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "! grep OBJECTIVE example_model/*pdb >> example_model/scores.txt"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "- Open this file in vim and type:\n",
    "\n",
    "\t`Shift :` to enter command input\n",
    "    \n",
    "\n",
    "- Then type:\n",
    "\n",
    "\t`:%!column -t`  which will ensure the data is column readable the type:\n",
    "    \n",
    "    `:%!sort -k7nr` which should sort on the last column which is the objective function\n",
    "    \n",
    "- Use this to select the top 10 models based on the Modeller objective function.\n",
    "\n",
    "- Copy these files into a new directory.\n",
    "\n",
    "- Next we will upload these models to the QMEAN server in order to select a single model.\n",
    "\n",
    "- The QMEAN (Qualitative Model Energy ANalysis) scoring function derives a quality estimate for both local and global (per-residue) quality estimates. For more information on the scoring function see: [https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2703985/](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2703985/)\n",
    "\n",
    "- Go to the QMEAN server website via:\n",
    "\n",
    "\t[https://swissmodel.expasy.org/qmean/](https://swissmodel.expasy.org/qmean)\n",
    "    \n",
    "\n",
    "- Select just 'QMEANBrane' from the three method options available (as we have a membrane protein)\n",
    "\n",
    "- Select the top 10 models you saved in the new directory for analysis by clicking the 'select coordinate file' option and click submit once these have been uploaded. Feel free to add your email if you'd like a copy of the results sent to you.\n",
    "\n",
    "- In the output you should see QMEAN scores for each model as well as a breakdown of the local quality.\n",
    "\n",
    "- By clicking Project Archive download we can obtain a .zip with a breakdown of the results.\n",
    "\n",
    "- For each model you should be able to scroll over the local quality score which will also simultaneously display amino acid residues on a 3D projection of your model on the panel to the right.\n",
    "\n",
    "- Select the model with the highest QMEAN4 value for the next step, keeping the QMEAN tab open.\n",
    "\n",
    "- We will now check the model's backbone torsions by generating a Ramachandran plot.\n",
    "\n",
    "- Go to the following website and upload your structure:\n",
    "\n",
    "\t[http://mordred.bioc.cam.ac.uk/~rapper/rampage.php](http://mordred.bioc.cam.ac.uk/~rapper/rampage.php)\n",
    "    \n",
    "\n",
    "- Download the postscript file to obtain a pdf.\n",
    "\n",
    "- Are there any violations and if so, why might they be occuring there?\n",
    "- Open up your .pdb file in pymol or VMD to check.\n",
    "\n",
    "- Do these regions also correspond to low local quality scores in QMEAN?"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### END!  Congratulations you are now ready to build your own homology models!"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 1
}
